{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prediction of the most effective treatment for patients with breast cancer\n",
    "\n",
    "This repository is the fruit of the collaborative work of **Eloi Ancellin**, **Maxime Bouthors**, **Joël Garde**, **Mathieu Juttet**, and **Louis-Oscar Morel**.\n",
    "It was created as an assignement of the Data Camp course of the Master 2 Data Science of the Polytechnic Institute of Paris during the year 2021.\n",
    "\n",
    "## Breast Cancer ; An introduction\n",
    "\n",
    "Breast cancer is the most common cancer in women with 2 million new cases annually worldwide according to the WHO. It alone is responsible for more than 500,000 deaths per year and nearly one in seven women will develop this disease in her lifetime. Breast cancer is detected by mammography, but it is the pathological examination of the tissues taken from the patient that makes it possible to establish the precise diagnosis of breast cancer.\n",
    "\n",
    "Pathology analysis is a complex process, **the challenge of which is to be able to offer the best treatment to each patient, thereby improving her chances of survival**. Therefore, the pathologist's role is to characterize the tumor so that the oncologist can give the best medicine. The diagnosis is based on the microscopic analysis of the lesion. \n",
    "\n",
    "The characterization of the tumor is based on (i) the analysis of its morphology which makes it possible to define its histological type (more than 25 types) and on (ii) its molecular characteristics.\n",
    "\n",
    "The morphological analysis corresponds to the traditional anatomopathological examination, based on the analysis of cell types, their relationships, and whether or not the overall architecture of the tissue is respected.\n",
    "\n",
    "Molecular analysis refine the type of tumor. Indeed, the development of therapies specifically targeting certain tumor molecular alterations present only in subgroups of patients, currently allows an increasingly important personalization of cancer treatment. The detection of these molecular abnormalities relies on the quantification of specific proteins in tumor tissue (Immunohistochemistry), the fluorescence quantification of specific genetic elements in tumor cells (FISH), the identification of tumor genomic abnormalities by sequencing of the DNA (NGS) [1 to 8]. Taking these characteristics into account when choosing the therapeutic approach has shown strong impacts on reducing the risk of disease progression (eg 35% for the treatment of metastatic breast cancer with Alpelisib, a molecule prescribed for female patients presenting RH + cancers (immunohistochemical characterization) with the presence of activating genomic mutations of the PIK3CA gene (NGS characterization)).\n",
    "\n",
    "**It is the tumor precise characterization that makes it possible to better understand the biology of a given patient's cancer and thus to prescribe the most suitable treatment.**\n",
    "\n",
    "\n",
    "However,**the complexity and number of examinations to perform for the precise characterization of the tumor increases the risk of errors.** For example, the proposal for innovative targeted chemotherapies is based on the combined interpretation by the physician of (i) morphological examination, HE biopsy, (ii) of the Extended Evaluation, (iii) of immunohistochemistry, (iv) of genomic analysis by DNA sequencing. The use of more and more parameters tends to make the diagnosis impracticable by the physician alone.\n",
    "\n",
    "The development of statistical learning algorithms and Data Science has opened up opportunities for improvement in this field, but are still very limited. Indeed, these developments require access to data that are difficult to obtain, are based on complex interdisciplinary know-how (mathematics, medicine and Computer Science). Recent advances in AI are promising, but operate in a black box, thus slowing the adoption of these solutions by pathologists in the medical community.\n",
    "\n",
    "### Conclusion:\n",
    "\n",
    "The complexity of tumor diagnosis requires the development of new solutions allowing the automatic integration of multiparametric morphological and molecular analyzes allowing physician support in their diagnostic process, thus improving their analytical capacities and their ability to offer effective personalized treatments.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Molecular Biology ; Key concepts to understand the work that needs to be done\n",
    "\n",
    "**The aim of this project is to find profiles of patients with characteristics associated with a good or bad response to a specific treatment. Depending on the data we have, our goal will be to assign the right treatment for each patient, thus improving their chances of survival.**\n",
    "\n",
    "The biology of cancer is complex. In this challenge, we will focus on (i) clinical data, such as age, ethnicity etc ...; (ii) the anatomopathological characteristics, ie the grade of the lesion, its stage and its histological type and; (iii) molecular biology, respectively genomic alteration (DNA mutation) and genomic expression (RNAseq) data.\n",
    "\n",
    "\n",
    "The proposal for highly personalized treatments must take into account this very large amount of information. We will approach here very quickly the fundamental notions of biology so that the reader can understand the data and can work on the dataframe to reduce its dimensions and create relevant composite criteria for his prediction work.\n",
    "\n",
    "\n",
    "DNA, or deoxyribonucleic acid, is a series of four chemical units called nucleotides (A, T, C, G), the combination of which supports information from living things. In humans, this “hardware” support is located in the nuclei of cells and represents a combination of 3 billion nucleotides grouped together in 23 pairs of chromosomes. The majority of this DNA (70%) is said to be non-coding, ie it does not allow the production of protein. The remaining 30% is divided into 20k-30k pairs of genes. A gene is a portion of DNA that produces messenger RNA (mRNA).\n",
    "\n",
    "\n",
    "RNA is a sequence of nucleotides (A, U, C, G) which is copied from the DNA model by an enzyme (protein) during the process of \"transcription\". This mRNA leaves the nucleus of the cell to be then \"translated\" into protein by a ribosome (RNA/proteic complex). The ribosome matches each RNA nucleotide triplet with an amino acid. The sequence of these amino acids constitutes a protein. Proteins have chemical groups that allow them to influence their environment and are therefore one of the supports of life.\n",
    "\n",
    "\n",
    "Cancer is characterized by an accumulation of mutations in DNA, causing changes in the function of proteins and therefore in the behavior of cells, making them:\n",
    "- self-sufficient in growth signals;\n",
    "- Insensitive to growth inhibiting signals;\n",
    "- Capable of avoiding apoptosis (cell immortality);\n",
    "- Capable of replicating indefinitely (infinite growth)\n",
    "- Capable of inducing angiogenesis and forming metastases.\n",
    "\n",
    "\n",
    "\n",
    "# Getting started\n",
    "The starting kit notebook can be found [here](https://github.com/MathieuJuttet/Cancer/blob/main/Starting_kit.ipynb). It provides detailed information on the project and explanations on the data.\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data Loading and Processing\n",
    "\n",
    "Collecting large and complete data for tumor characterization is a complex task. We have chosen to base ourselves on the GDC Portal, which aggregates many databases specializing in cancer. The GDC Data Portal is a robust data-driven platform that allows cancer researchers and bioinformaticians to search and download cancer data for analysis.\n",
    "\n",
    "The GDC Data Portal combines 67 projects focusing on 68 different cancers for a total of 84,392 patients. We were interested in patients with primary breast cancer (9,115 cases). Data retrieval for these 9115 patients was achieved by sending a JSON through the GDC API.\n",
    "(insert code).\n",
    "\n",
    "expliquer le travail qui a été fait sur la récuperation des données, son organisation, pourquoi on a viré certaines colonnes etc...\n",
    "(insert code)\n",
    "\n",
    "\n",
    "In order to get data, we used the API provided by the GDC website. From this load, we get 4 datasets that we load in separate dataframes. First, we will show how we performed the API query, then how we have cleaned them up to get a single dataframe. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import json\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os.path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "##### Folders\n",
    "pload_p = \"payload/\"\n",
    "data_p = \"data/\"\n",
    "\n",
    "headers = {'Content-Type': 'application/json',}\n",
    "root = 'https://api.gdc.cancer.gov/'\n",
    "endpt_l = {\"case\":\"cases\",\n",
    "           \"ssm\":\"ssm_occurrences\",\n",
    "           \"cnv\":\"cnv_occurrences\",\n",
    "           \"gene\":\"analysis/top_mutated_genes_by_project\"}\n",
    "\n",
    "pload_l = [\"case\", \"ssm\", \"cnv\", \"gene\"]\n",
    "\n",
    "def dl(endpt, data):\n",
    "    \"\"\" simple wrapper around request.post with correct arguments filled in.\n",
    "    \"\"\"\n",
    "    response = requests.post(root + endpt, timeout=30, headers=headers, data=data)\n",
    "    return response.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fill_filter_gene():\n",
    "    \"\"\" Builds the filter for cvn/ssm to ensure only motation on top genes are reported.\n",
    "    \"\"\"\n",
    "    top_genes = pd.read_csv(data_p + \"gene.tsv\", \"\\t\")[\"symbol\"].to_numpy()\n",
    "    for req in (\"ssm\", \"cnv\"):\n",
    "        with open(pload_p + req, \"r+\") as f:\n",
    "            payload = json.load(f)\n",
    "            payload[\"filters\"][\"content\"][0][\"content\"][\"value\"] = list(top_genes)\n",
    "            f.seek(0)\n",
    "            json.dump(payload, f, separators=(\",\",\":\"))\n",
    "            f.truncate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load(fname, force_dl=False):\n",
    "    \"\"\" wrapper utility to avoid re-downloading data.\n",
    "    \"\"\"\n",
    "    fp = data_p + fname + \".tsv\"\n",
    "    if not force_dl and os.path.exists(fp):\n",
    "        return pd.read_csv(fp, sep=\"\\t\", low_memory=False)\n",
    "    #load request params\n",
    "    print(f\"file {fp} not found... Downloading...\")\n",
    "    data = open(pload_p + fname)\n",
    "    text = dl(endpt_l[fname], data)\n",
    "    with open(fp, \"w\") as f:\n",
    "        f.write(text)\n",
    "        \n",
    "    return pd.read_csv(fp, sep=\"\\t\", low_memory=False)       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "casedf = load(\"case\")\n",
    "genedf = load(\"gene\")\n",
    "fill_filter_gene() #update filters to only cater top 50genes\n",
    "ssmdf = load(\"ssm\")\n",
    "cnvdf = load(\"cnv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After executing the above code, you have four .tsv files in a data folder.\n",
    "Now, we will load the downloaded data cleaning we have done."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/macbookprodeloi/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (0,1,2,3,4,5,6,7,8,9,10,12,13,14,15,16,17,18,19,20,21,22,23,25,28,29,51,53,54,55,56,57,60,61,62,87,97,119,139,140,148,151,154,156,157,161,162,163,165,166,168,171,175,176,177,185,186,187,189,190,192,199,200,201,209,210,211,213,214,216,219,223,224,225,233,234,235,237,238,239,241,242,243,244,245,246,247,248,249,251,258,259,260,268,269,270,272,273,275,282,283,284,292,293,294,296,297,299,302,306,307,316,317,318,320,321,323,326,330,331,340,342,344,345,347,353,354,355,364,365,366,368,369,371,374,378,379,388,389,390,392,393,410,416,419,428,429,437,440,441,444,446,448,449,450,452,455,467,470,472,473,488,491,492,503,506,508,509,510,511,514,515,519,523,524,527,528,539,542,544,545,546,547,550,551,555,559,560,563,564,575,578,580,581,582,583,586,587,591,595,596,599,600,611,614,616,618,619,622,623,627,631,632,635,636,647,654,655,658,659,663,667,668,672,683,690,691,694,695,699,703,704,707,708,719,726,727,730,731,735,739,740,743,744,755,762,763,766,767,771,775,777,778,779,780,781) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "df_case = pd.read_csv('data/case.tsv', sep='\\t')\n",
    "df_cnv = pd.read_csv('data/cnv.tsv', sep='\\t')\n",
    "df_gene = pd.read_csv('data/gene.tsv', sep='\\t')\n",
    "df_ssm = pd.read_csv('data/ssm.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Clinical and pathological Data\n",
    "\n",
    "The clinical and pathological data is provided through the df_case dataframe. It contains general informations about the patient, age, sexe, ethnicity, multiple risk factors, diagnosis informations, what treatment has been given, the stage of the tumor, caracteristics of the tumor...etc\n",
    "\n",
    "Below is the display of all the columns and their filling purcentage.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotations.0.annotation_id \t 0.31815688425671973\n",
      "annotations.0.case_id \t 0.31815688425671973\n",
      "annotations.0.category \t 0.31815688425671973\n",
      "annotations.0.classification \t 0.31815688425671973\n",
      "annotations.0.created_datetime \t 0.31815688425671973\n",
      "annotations.0.entity_id \t 0.31815688425671973\n",
      "annotations.0.entity_submitter_id \t 0.31815688425671973\n",
      "annotations.0.entity_type \t 0.31815688425671973\n",
      "annotations.0.notes \t 0.31815688425671973\n",
      "annotations.0.state \t 0.31815688425671973\n",
      "annotations.0.status \t 0.31815688425671973\n",
      "annotations.0.submitter_id \t 0.31815688425671973\n",
      "annotations.0.updated_datetime \t 0.31815688425671973\n",
      "annotations.1.annotation_id \t 0.06582556226001096\n",
      "annotations.1.case_id \t 0.06582556226001096\n",
      "annotations.1.category \t 0.06582556226001096\n",
      "annotations.1.classification \t 0.06582556226001096\n",
      "annotations.1.created_datetime \t 0.06582556226001096\n",
      "annotations.1.entity_id \t 0.06582556226001096\n",
      "annotations.1.entity_submitter_id \t 0.06582556226001096\n",
      "annotations.1.entity_type \t 0.06582556226001096\n",
      "annotations.1.notes \t 0.06582556226001096\n",
      "annotations.1.state \t 0.06582556226001096\n",
      "annotations.1.status \t 0.06582556226001096\n",
      "annotations.1.submitter_id \t 0.06582556226001096\n",
      "annotations.1.updated_datetime \t 0.06582556226001096\n",
      "case_id \t 100.0\n",
      "demographic.age_at_index \t 67.60285244103127\n",
      "demographic.age_is_obfuscated \t 0.010970927043335162\n",
      "demographic.cause_of_death \t 0.010970927043335162\n",
      "demographic.cause_of_death_source \t 0.0\n",
      "demographic.country_of_residence_at_enrollment \t 0.0\n",
      "demographic.created_datetime \t 87.94295117937466\n",
      "demographic.days_to_birth \t 67.44925946242458\n",
      "demographic.days_to_death \t 1.6675809105869446\n",
      "demographic.demographic_id \t 99.98902907295667\n",
      "demographic.ethnicity \t 99.98902907295667\n",
      "demographic.gender \t 99.98902907295667\n",
      "demographic.occupation_duration_years \t 0.0\n",
      "demographic.premature_at_birth \t 0.0\n",
      "demographic.race \t 99.98902907295667\n",
      "demographic.state \t 99.98902907295667\n",
      "demographic.submitter_id \t 99.98902907295667\n",
      "demographic.updated_datetime \t 99.98902907295667\n",
      "demographic.vital_status \t 99.98902907295667\n",
      "demographic.weeks_gestation_at_birth \t 0.0\n",
      "demographic.year_of_birth \t 12.024136039495337\n",
      "demographic.year_of_death \t 1.1409764125068567\n",
      "diagnoses.0.age_at_diagnosis \t 43.65331870543061\n",
      "diagnoses.0.ajcc_clinical_m \t 0.0\n",
      "diagnoses.0.ajcc_clinical_n \t 0.0\n",
      "diagnoses.0.ajcc_clinical_stage \t 0.010970927043335162\n",
      "diagnoses.0.ajcc_clinical_t \t 0.0\n",
      "diagnoses.0.ajcc_pathologic_m \t 12.046077893582009\n",
      "diagnoses.0.ajcc_pathologic_n \t 12.046077893582009\n",
      "diagnoses.0.ajcc_pathologic_stage \t 13.395501919912235\n",
      "diagnoses.0.ajcc_pathologic_t \t 12.046077893582009\n",
      "diagnoses.0.ajcc_staging_system_edition \t 10.499177180471749\n",
      "diagnoses.0.anaplasia_present \t 0.0\n",
      "diagnoses.0.anaplasia_present_type \t 0.0\n",
      "diagnoses.0.ann_arbor_b_symptoms \t 0.010970927043335162\n",
      "diagnoses.0.ann_arbor_clinical_stage \t 0.010970927043335162\n",
      "diagnoses.0.ann_arbor_extranodal_involvement \t 0.010970927043335162\n",
      "diagnoses.0.ann_arbor_pathologic_stage \t 0.0\n",
      "diagnoses.0.best_overall_response \t 0.0\n",
      "diagnoses.0.breslow_thickness \t 0.0\n",
      "diagnoses.0.burkitt_lymphoma_clinical_variant \t 0.0\n",
      "diagnoses.0.child_pugh_classification \t 0.0\n",
      "diagnoses.0.circumferential_resection_margin \t 0.0\n",
      "diagnoses.0.classification_of_tumor \t 41.85408667032364\n",
      "diagnoses.0.cog_liver_stage \t 0.0\n",
      "diagnoses.0.cog_neuroblastoma_risk_group \t 0.0\n",
      "diagnoses.0.cog_renal_stage \t 0.0\n",
      "diagnoses.0.cog_rhabdomyosarcoma_risk_group \t 0.0\n",
      "diagnoses.0.created_datetime \t 87.94295117937466\n",
      "diagnoses.0.days_to_best_overall_response \t 0.0\n",
      "diagnoses.0.days_to_diagnosis \t 12.057048820625344\n",
      "diagnoses.0.days_to_last_follow_up \t 10.905101481075151\n",
      "diagnoses.0.days_to_last_known_disease_status \t 0.0\n",
      "diagnoses.0.days_to_recurrence \t 0.0\n",
      "diagnoses.0.diagnosis_id \t 99.98902907295667\n",
      "diagnoses.0.enneking_msts_grade \t 0.0\n",
      "diagnoses.0.enneking_msts_metastasis \t 0.0\n",
      "diagnoses.0.enneking_msts_stage \t 0.0\n",
      "diagnoses.0.enneking_msts_tumor_site \t 0.0\n",
      "diagnoses.0.esophageal_columnar_dysplasia_degree \t 0.0\n",
      "diagnoses.0.esophageal_columnar_metaplasia_present \t 0.0\n",
      "diagnoses.0.figo_stage \t 1.4701042238069117\n",
      "diagnoses.0.figo_staging_edition_year \t 0.0\n",
      "diagnoses.0.first_symptom_prior_to_diagnosis \t 0.0\n",
      "diagnoses.0.gastric_esophageal_junction_involvement \t 0.0\n",
      "diagnoses.0.gleason_grade_group \t 0.0\n",
      "diagnoses.0.gleason_grade_tertiary \t 0.0\n",
      "diagnoses.0.gleason_patterns_percent \t 0.0\n",
      "diagnoses.0.goblet_cells_columnar_mucosa_present \t 0.0\n",
      "diagnoses.0.greatest_tumor_dimension \t 0.0\n",
      "diagnoses.0.gross_tumor_weight \t 0.0\n",
      "diagnoses.0.icd_10_code \t 12.057048820625344\n",
      "diagnoses.0.igcccg_stage \t 0.0\n",
      "diagnoses.0.inpc_grade \t 0.0\n",
      "diagnoses.0.inpc_histologic_group \t 0.0\n",
      "diagnoses.0.inrg_stage \t 0.0\n",
      "diagnoses.0.inss_stage \t 0.0\n",
      "diagnoses.0.international_prognostic_index \t 0.0\n",
      "diagnoses.0.irs_group \t 0.0\n",
      "diagnoses.0.irs_stage \t 0.0\n",
      "diagnoses.0.ishak_fibrosis_score \t 0.0\n",
      "diagnoses.0.iss_stage \t 0.0\n",
      "diagnoses.0.largest_extrapelvic_peritoneal_focus \t 0.0\n",
      "diagnoses.0.last_known_disease_status \t 97.59736697750961\n",
      "diagnoses.0.laterality \t 0.0\n",
      "diagnoses.0.lymph_node_involved_site \t 0.0\n",
      "diagnoses.0.lymph_nodes_positive \t 0.0\n",
      "diagnoses.0.lymph_nodes_tested \t 0.0\n",
      "diagnoses.0.lymphatic_invasion_present \t 0.0\n",
      "diagnoses.0.margin_distance \t 0.0\n",
      "diagnoses.0.margins_involved_site \t 0.0\n",
      "diagnoses.0.masaoka_stage \t 0.0\n",
      "diagnoses.0.medulloblastoma_molecular_classification \t 0.0\n",
      "diagnoses.0.metastasis_at_diagnosis \t 0.010970927043335162\n",
      "diagnoses.0.metastasis_at_diagnosis_site \t 0.0\n",
      "diagnoses.0.method_of_diagnosis \t 0.0\n",
      "diagnoses.0.micropapillary_features \t 0.0\n",
      "diagnoses.0.mitosis_karyorrhexis_index \t 0.0\n",
      "diagnoses.0.mitotic_count \t 0.0\n",
      "diagnoses.0.morphology \t 99.98902907295667\n",
      "diagnoses.0.non_nodal_regional_disease \t 0.0\n",
      "diagnoses.0.non_nodal_tumor_deposits \t 0.0\n",
      "diagnoses.0.ovarian_specimen_status \t 0.0\n",
      "diagnoses.0.ovarian_surface_involvement \t 0.0\n",
      "diagnoses.0.papillary_renal_cell_type \t 0.0\n",
      "diagnoses.0.percent_tumor_invasion \t 0.0\n",
      "diagnoses.0.perineural_invasion_present \t 0.0\n",
      "diagnoses.0.peripancreatic_lymph_nodes_positive \t 0.0\n",
      "diagnoses.0.peripancreatic_lymph_nodes_tested \t 0.0\n",
      "diagnoses.0.peritoneal_fluid_cytological_status \t 0.0\n",
      "diagnoses.0.pregnant_at_diagnosis \t 0.0\n",
      "diagnoses.0.primary_diagnosis \t 99.98902907295667\n",
      "diagnoses.0.primary_gleason_grade \t 0.0\n",
      "diagnoses.0.prior_malignancy \t 13.395501919912235\n",
      "diagnoses.0.prior_treatment \t 12.057048820625344\n",
      "diagnoses.0.progression_or_recurrence \t 97.59736697750961\n",
      "diagnoses.0.residual_disease \t 0.0\n",
      "diagnoses.0.secondary_gleason_grade \t 0.0\n",
      "diagnoses.0.site_of_resection_or_biopsy \t 99.98902907295667\n",
      "diagnoses.0.state \t 99.98902907295667\n",
      "diagnoses.0.submitter_id \t 99.98902907295667\n",
      "diagnoses.0.supratentorial_localization \t 0.0\n",
      "diagnoses.0.synchronous_malignancy \t 12.046077893582009\n",
      "diagnoses.0.tissue_or_organ_of_origin \t 99.98902907295667\n",
      "diagnoses.0.transglottic_extension \t 0.0\n",
      "diagnoses.0.treatments.0.created_datetime \t 6.023038946791004\n",
      "diagnoses.0.treatments.0.days_to_treatment_end \t 0.010970927043335162\n",
      "diagnoses.0.treatments.0.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.0.initial_disease_status \t 0.010970927043335162\n",
      "diagnoses.0.treatments.0.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.0.state \t 12.057048820625344\n",
      "diagnoses.0.treatments.0.submitter_id \t 12.057048820625344\n",
      "diagnoses.0.treatments.0.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_id \t 12.057048820625344\n",
      "diagnoses.0.treatments.0.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.0.treatment_or_therapy \t 12.057048820625344\n",
      "diagnoses.0.treatments.0.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_type \t 12.057048820625344\n",
      "diagnoses.0.treatments.0.updated_datetime \t 12.057048820625344\n",
      "diagnoses.0.treatments.1.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.1.created_datetime \t 6.044980800877674\n",
      "diagnoses.0.treatments.1.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.1.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.1.initial_disease_status \t 0.010970927043335162\n",
      "diagnoses.0.treatments.1.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.1.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.1.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.1.state \t 12.057048820625344\n",
      "diagnoses.0.treatments.1.submitter_id \t 12.057048820625344\n",
      "diagnoses.0.treatments.1.therapeutic_agents \t 0.010970927043335162\n",
      "diagnoses.0.treatments.1.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_id \t 12.057048820625344\n",
      "diagnoses.0.treatments.1.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.1.treatment_or_therapy \t 12.057048820625344\n",
      "diagnoses.0.treatments.1.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_type \t 12.057048820625344\n",
      "diagnoses.0.treatments.1.updated_datetime \t 12.057048820625344\n",
      "diagnoses.0.treatments.10.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.10.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.10.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.10.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.10.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.10.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.10.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.therapeutic_agents \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.10.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.2.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.2.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.2.initial_disease_status \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.2.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.2.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.2.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.therapeutic_agents \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.2.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.initial_disease_status \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.therapeutic_agents \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.3.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.4.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.4.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.4.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.4.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.4.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.4.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.therapeutic_agents \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.4.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.5.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.5.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.5.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.5.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.5.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.5.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.therapeutic_agents \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.5.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.6.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.6.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.6.initial_disease_status \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.6.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.6.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.6.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.6.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.7.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.7.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.initial_disease_status \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.7.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.7.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.7.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.treatment_intent_type \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.7.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.8.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.8.days_to_treatment_start \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.8.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.8.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.8.regimen_or_line_of_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.8.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.9.created_datetime \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.9.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.9.initial_disease_status \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.9.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.9.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.9.state \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.submitter_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_id \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_intent_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_or_therapy \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_type \t 0.010970927043335162\n",
      "diagnoses.0.treatments.9.updated_datetime \t 0.010970927043335162\n",
      "diagnoses.0.tumor_confined_to_organ_of_origin \t 0.0\n",
      "diagnoses.0.tumor_depth \t 0.0\n",
      "diagnoses.0.tumor_focality \t 0.0\n",
      "diagnoses.0.tumor_grade \t 97.60833790455294\n",
      "diagnoses.0.tumor_largest_dimension_diameter \t 0.0\n",
      "diagnoses.0.tumor_regression_grade \t 0.0\n",
      "diagnoses.0.tumor_stage \t 97.59736697750961\n",
      "diagnoses.0.updated_datetime \t 99.98902907295667\n",
      "diagnoses.0.vascular_invasion_present \t 0.0\n",
      "diagnoses.0.vascular_invasion_type \t 0.0\n",
      "diagnoses.0.weiss_assessment_score \t 0.0\n",
      "diagnoses.0.wilms_tumor_histologic_subtype \t 0.0\n",
      "diagnoses.0.year_of_diagnosis \t 12.024136039495337\n",
      "exposures.0.age_at_onset \t 0.0\n",
      "exposures.0.alcohol_days_per_week \t 0.0\n",
      "exposures.0.alcohol_drinks_per_day \t 0.0\n",
      "exposures.0.alcohol_history \t 12.046077893582009\n",
      "exposures.0.alcohol_intensity \t 0.0\n",
      "exposures.0.asbestos_exposure \t 0.0\n",
      "exposures.0.bmi \t 0.010970927043335162\n",
      "exposures.0.cigarettes_per_day \t 0.0\n",
      "exposures.0.coal_dust_exposure \t 0.0\n",
      "exposures.0.created_datetime \t 0.010970927043335162\n",
      "exposures.0.environmental_tobacco_smoke_exposure \t 0.0\n",
      "exposures.0.exposure_duration \t 0.0\n",
      "exposures.0.exposure_id \t 12.057048820625344\n",
      "exposures.0.exposure_type \t 0.0\n",
      "exposures.0.height \t 0.010970927043335162\n",
      "exposures.0.marijuana_use_per_week \t 0.0\n",
      "exposures.0.pack_years_smoked \t 0.0\n",
      "exposures.0.radon_exposure \t 0.0\n",
      "exposures.0.respirable_crystalline_silica_exposure \t 0.0\n",
      "exposures.0.secondhand_smoke_as_child \t 0.0\n",
      "exposures.0.smoking_frequency \t 0.0\n",
      "exposures.0.state \t 12.057048820625344\n",
      "exposures.0.submitter_id \t 12.057048820625344\n",
      "exposures.0.time_between_waking_and_first_smoke \t 0.0\n",
      "exposures.0.tobacco_smoking_onset_year \t 0.0\n",
      "exposures.0.tobacco_smoking_quit_year \t 0.0\n",
      "exposures.0.tobacco_smoking_status \t 0.010970927043335162\n",
      "exposures.0.tobacco_use_per_day \t 0.0\n",
      "exposures.0.type_of_smoke_exposure \t 0.0\n",
      "exposures.0.type_of_tobacco_used \t 0.0\n",
      "exposures.0.updated_datetime \t 12.057048820625344\n",
      "exposures.0.weight \t 0.010970927043335162\n",
      "exposures.0.years_smoked \t 0.0\n",
      "family_histories.0.created_datetime \t 0.010970927043335162\n",
      "family_histories.0.family_history_id \t 0.010970927043335162\n",
      "family_histories.0.relationship_age_at_diagnosis \t 0.0\n",
      "family_histories.0.relationship_gender \t 0.0\n",
      "family_histories.0.relationship_primary_diagnosis \t 0.010970927043335162\n",
      "family_histories.0.relationship_type \t 0.0\n",
      "family_histories.0.relative_with_cancer_history \t 0.010970927043335162\n",
      "family_histories.0.relatives_with_cancer_history_count \t 0.0\n",
      "family_histories.0.state \t 0.010970927043335162\n",
      "family_histories.0.submitter_id \t 0.010970927043335162\n",
      "family_histories.0.updated_datetime \t 0.010970927043335162\n",
      "id \t 100.0\n",
      "samples.0.biospecimen_anatomic_site \t 2.38069116840373\n",
      "samples.0.biospecimen_laterality \t 0.0\n",
      "samples.0.catalog_reference \t 0.0\n",
      "samples.0.composition \t 0.7021393307734504\n",
      "samples.0.created_datetime \t 91.90345584201866\n",
      "samples.0.current_weight \t 0.0\n",
      "samples.0.days_to_collection \t 8.107515085024684\n",
      "samples.0.days_to_sample_procurement \t 3.9824465167306635\n",
      "samples.0.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.0.distance_normal_to_tumor \t 0.0\n",
      "samples.0.distributor_reference \t 0.0\n",
      "samples.0.freezing_method \t 0.0\n",
      "samples.0.growth_rate \t 0.0\n",
      "samples.0.initial_weight \t 4.256719692814043\n",
      "samples.0.intermediate_dimension \t 0.0\n",
      "samples.0.is_ffpe \t 14.448710916072407\n",
      "samples.0.longest_dimension \t 0.0\n",
      "samples.0.method_of_sample_procurement \t 0.0\n",
      "samples.0.oct_embedded \t 12.068019747668677\n",
      "samples.0.passage_count \t 0.0\n",
      "samples.0.pathology_report_uuid \t 3.8178826110806363\n",
      "samples.0.preservation_method \t 3.9824465167306635\n",
      "samples.0.sample_id \t 100.0\n",
      "samples.0.sample_type \t 100.0\n",
      "samples.0.sample_type_id \t 13.53812397147559\n",
      "samples.0.shortest_dimension \t 0.0\n",
      "samples.0.state \t 100.0\n",
      "samples.0.submitter_id \t 100.0\n",
      "samples.0.time_between_clamping_and_freezing \t 0.0\n",
      "samples.0.time_between_excision_and_freezing \t 0.0\n",
      "samples.0.tissue_collection_type \t 0.0\n",
      "samples.0.tissue_type \t 100.0\n",
      "samples.0.tumor_code \t 0.0\n",
      "samples.0.tumor_code_id \t 0.0\n",
      "samples.0.tumor_descriptor \t 28.337904552934724\n",
      "samples.0.updated_datetime \t 100.0\n",
      "samples.1.biospecimen_anatomic_site \t 2.227098189797038\n",
      "samples.1.biospecimen_laterality \t 0.0\n",
      "samples.1.catalog_reference \t 0.0\n",
      "samples.1.composition \t 0.7679648930334613\n",
      "samples.1.created_datetime \t 7.383433900164564\n",
      "samples.1.current_weight \t 0.0\n",
      "samples.1.days_to_collection \t 8.381788261108063\n",
      "samples.1.days_to_sample_procurement \t 3.6642896324739436\n",
      "samples.1.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.1.distance_normal_to_tumor \t 0.0\n",
      "samples.1.distributor_reference \t 0.0\n",
      "samples.1.freezing_method \t 0.0\n",
      "samples.1.growth_rate \t 0.0\n",
      "samples.1.initial_weight \t 5.013713658804169\n",
      "samples.1.intermediate_dimension \t 0.0\n",
      "samples.1.is_ffpe \t 14.28414701042238\n",
      "samples.1.longest_dimension \t 0.0\n",
      "samples.1.method_of_sample_procurement \t 0.0\n",
      "samples.1.oct_embedded \t 12.057048820625344\n",
      "samples.1.passage_count \t 0.0\n",
      "samples.1.pathology_report_uuid \t 4.509051014810751\n",
      "samples.1.preservation_method \t 3.6642896324739436\n",
      "samples.1.sample_id \t 15.765222161272627\n",
      "samples.1.sample_type \t 15.765222161272627\n",
      "samples.1.sample_type_id \t 13.53812397147559\n",
      "samples.1.shortest_dimension \t 0.0\n",
      "samples.1.state \t 15.765222161272627\n",
      "samples.1.submitter_id \t 15.765222161272627\n",
      "samples.1.time_between_clamping_and_freezing \t 0.0\n",
      "samples.1.time_between_excision_and_freezing \t 0.0\n",
      "samples.1.tissue_collection_type \t 0.0\n",
      "samples.1.tissue_type \t 15.765222161272627\n",
      "samples.1.tumor_code \t 0.0\n",
      "samples.1.tumor_code_id \t 0.0\n",
      "samples.1.tumor_descriptor \t 0.0\n",
      "samples.1.updated_datetime \t 15.765222161272627\n",
      "samples.2.biospecimen_anatomic_site \t 1.6346681294569392\n",
      "samples.2.biospecimen_laterality \t 0.0\n",
      "samples.2.catalog_reference \t 0.0\n",
      "samples.2.composition \t 0.4388370817334065\n",
      "samples.2.created_datetime \t 5.573230938014262\n",
      "samples.2.current_weight \t 0.0\n",
      "samples.2.days_to_collection \t 7.931980252331322\n",
      "samples.2.days_to_sample_procurement \t 3.6752605595172794\n",
      "samples.2.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.2.distance_normal_to_tumor \t 0.0\n",
      "samples.2.distributor_reference \t 0.0\n",
      "samples.2.freezing_method \t 0.0\n",
      "samples.2.growth_rate \t 0.0\n",
      "samples.2.initial_weight \t 4.201865057597367\n",
      "samples.2.intermediate_dimension \t 0.0\n",
      "samples.2.is_ffpe \t 13.274821722435545\n",
      "samples.2.longest_dimension \t 0.0\n",
      "samples.2.method_of_sample_procurement \t 0.0\n",
      "samples.2.oct_embedded \t 11.640153592978606\n",
      "samples.2.passage_count \t 0.0\n",
      "samples.2.pathology_report_uuid \t 3.7081733406472845\n",
      "samples.2.preservation_method \t 3.6752605595172794\n",
      "samples.2.sample_id \t 13.527153044432255\n",
      "samples.2.sample_type \t 13.527153044432255\n",
      "samples.2.sample_type_id \t 11.892484914975315\n",
      "samples.2.shortest_dimension \t 0.0\n",
      "samples.2.state \t 13.527153044432255\n",
      "samples.2.submitter_id \t 13.527153044432255\n",
      "samples.2.time_between_clamping_and_freezing \t 0.0\n",
      "samples.2.time_between_excision_and_freezing \t 0.0\n",
      "samples.2.tissue_collection_type \t 0.0\n",
      "samples.2.tissue_type \t 13.527153044432255\n",
      "samples.2.tumor_code \t 0.0\n",
      "samples.2.tumor_code_id \t 0.0\n",
      "samples.2.tumor_descriptor \t 0.0\n",
      "samples.2.updated_datetime \t 13.527153044432255\n",
      "samples.3.biospecimen_anatomic_site \t 0.6253428414701042\n",
      "samples.3.biospecimen_laterality \t 0.0\n",
      "samples.3.catalog_reference \t 0.0\n",
      "samples.3.composition \t 0.12068019747668679\n",
      "samples.3.created_datetime \t 0.9654415798134943\n",
      "samples.3.current_weight \t 0.0\n",
      "samples.3.days_to_collection \t 0.7240811848601206\n",
      "samples.3.days_to_sample_procurement \t 0.34009873834339005\n",
      "samples.3.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.3.distance_normal_to_tumor \t 0.0\n",
      "samples.3.distributor_reference \t 0.0\n",
      "samples.3.freezing_method \t 0.0\n",
      "samples.3.growth_rate \t 0.0\n",
      "samples.3.initial_weight \t 0.4388370817334065\n",
      "samples.3.intermediate_dimension \t 0.0\n",
      "samples.3.is_ffpe \t 1.6895227646736148\n",
      "samples.3.longest_dimension \t 0.0\n",
      "samples.3.method_of_sample_procurement \t 0.0\n",
      "samples.3.oct_embedded \t 1.0641799232035107\n",
      "samples.3.passage_count \t 0.0\n",
      "samples.3.pathology_report_uuid \t 0.2523313219967087\n",
      "samples.3.preservation_method \t 0.34009873834339005\n",
      "samples.3.sample_id \t 1.6895227646736148\n",
      "samples.3.sample_type \t 1.6895227646736148\n",
      "samples.3.sample_type_id \t 1.0641799232035107\n",
      "samples.3.shortest_dimension \t 0.0\n",
      "samples.3.state \t 1.6895227646736148\n",
      "samples.3.submitter_id \t 1.6895227646736148\n",
      "samples.3.time_between_clamping_and_freezing \t 0.0\n",
      "samples.3.time_between_excision_and_freezing \t 0.0\n",
      "samples.3.tissue_collection_type \t 0.0\n",
      "samples.3.tissue_type \t 1.6895227646736148\n",
      "samples.3.tumor_code \t 0.0\n",
      "samples.3.tumor_code_id \t 0.0\n",
      "samples.3.tumor_descriptor \t 0.0\n",
      "samples.3.updated_datetime \t 1.6895227646736148\n",
      "samples.4.biospecimen_anatomic_site \t 0.41689522764673614\n",
      "samples.4.biospecimen_laterality \t 0.0\n",
      "samples.4.catalog_reference \t 0.0\n",
      "samples.4.composition \t 0.07679648930334614\n",
      "samples.4.created_datetime \t 0.41689522764673614\n",
      "samples.4.current_weight \t 0.0\n",
      "samples.4.days_to_collection \t 0.04388370817334065\n",
      "samples.4.days_to_sample_procurement \t 0.0\n",
      "samples.4.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.4.distance_normal_to_tumor \t 0.0\n",
      "samples.4.distributor_reference \t 0.0\n",
      "samples.4.freezing_method \t 0.0\n",
      "samples.4.growth_rate \t 0.0\n",
      "samples.4.initial_weight \t 0.021941854086670324\n",
      "samples.4.intermediate_dimension \t 0.0\n",
      "samples.4.is_ffpe \t 0.4607789358200768\n",
      "samples.4.longest_dimension \t 0.0\n",
      "samples.4.method_of_sample_procurement \t 0.0\n",
      "samples.4.oct_embedded \t 0.04388370817334065\n",
      "samples.4.passage_count \t 0.0\n",
      "samples.4.pathology_report_uuid \t 0.021941854086670324\n",
      "samples.4.preservation_method \t 0.0\n",
      "samples.4.sample_id \t 0.4607789358200768\n",
      "samples.4.sample_type \t 0.4607789358200768\n",
      "samples.4.sample_type_id \t 0.04388370817334065\n",
      "samples.4.shortest_dimension \t 0.0\n",
      "samples.4.state \t 0.4607789358200768\n",
      "samples.4.submitter_id \t 0.4607789358200768\n",
      "samples.4.time_between_clamping_and_freezing \t 0.0\n",
      "samples.4.time_between_excision_and_freezing \t 0.0\n",
      "samples.4.tissue_collection_type \t 0.0\n",
      "samples.4.tissue_type \t 0.4607789358200768\n",
      "samples.4.tumor_code \t 0.0\n",
      "samples.4.tumor_code_id \t 0.0\n",
      "samples.4.tumor_descriptor \t 0.0\n",
      "samples.4.updated_datetime \t 0.4607789358200768\n",
      "samples.5.biospecimen_anatomic_site \t 0.15359297860669227\n",
      "samples.5.biospecimen_laterality \t 0.0\n",
      "samples.5.catalog_reference \t 0.0\n",
      "samples.5.composition \t 0.06582556226001096\n",
      "samples.5.created_datetime \t 0.15359297860669227\n",
      "samples.5.current_weight \t 0.0\n",
      "samples.5.days_to_collection \t 0.0\n",
      "samples.5.days_to_sample_procurement \t 0.0\n",
      "samples.5.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.5.distance_normal_to_tumor \t 0.0\n",
      "samples.5.distributor_reference \t 0.0\n",
      "samples.5.freezing_method \t 0.0\n",
      "samples.5.growth_rate \t 0.0\n",
      "samples.5.initial_weight \t 0.0\n",
      "samples.5.intermediate_dimension \t 0.0\n",
      "samples.5.is_ffpe \t 0.15359297860669227\n",
      "samples.5.longest_dimension \t 0.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "samples.5.method_of_sample_procurement \t 0.0\n",
      "samples.5.oct_embedded \t 0.0\n",
      "samples.5.passage_count \t 0.0\n",
      "samples.5.pathology_report_uuid \t 0.0\n",
      "samples.5.preservation_method \t 0.0\n",
      "samples.5.sample_id \t 0.15359297860669227\n",
      "samples.5.sample_type \t 0.15359297860669227\n",
      "samples.5.sample_type_id \t 0.0\n",
      "samples.5.shortest_dimension \t 0.0\n",
      "samples.5.state \t 0.15359297860669227\n",
      "samples.5.submitter_id \t 0.15359297860669227\n",
      "samples.5.time_between_clamping_and_freezing \t 0.0\n",
      "samples.5.time_between_excision_and_freezing \t 0.0\n",
      "samples.5.tissue_collection_type \t 0.0\n",
      "samples.5.tissue_type \t 0.15359297860669227\n",
      "samples.5.tumor_code \t 0.0\n",
      "samples.5.tumor_code_id \t 0.0\n",
      "samples.5.tumor_descriptor \t 0.0\n",
      "samples.5.updated_datetime \t 0.15359297860669227\n",
      "samples.6.biospecimen_anatomic_site \t 0.09873834339001646\n",
      "samples.6.biospecimen_laterality \t 0.0\n",
      "samples.6.catalog_reference \t 0.0\n",
      "samples.6.composition \t 0.0\n",
      "samples.6.created_datetime \t 0.09873834339001646\n",
      "samples.6.current_weight \t 0.0\n",
      "samples.6.days_to_collection \t 0.0\n",
      "samples.6.days_to_sample_procurement \t 0.0\n",
      "samples.6.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.6.distance_normal_to_tumor \t 0.0\n",
      "samples.6.distributor_reference \t 0.0\n",
      "samples.6.freezing_method \t 0.0\n",
      "samples.6.growth_rate \t 0.0\n",
      "samples.6.initial_weight \t 0.0\n",
      "samples.6.intermediate_dimension \t 0.0\n",
      "samples.6.is_ffpe \t 0.09873834339001646\n",
      "samples.6.longest_dimension \t 0.0\n",
      "samples.6.method_of_sample_procurement \t 0.0\n",
      "samples.6.oct_embedded \t 0.0\n",
      "samples.6.passage_count \t 0.0\n",
      "samples.6.pathology_report_uuid \t 0.0\n",
      "samples.6.preservation_method \t 0.0\n",
      "samples.6.sample_id \t 0.09873834339001646\n",
      "samples.6.sample_type \t 0.09873834339001646\n",
      "samples.6.sample_type_id \t 0.0\n",
      "samples.6.shortest_dimension \t 0.0\n",
      "samples.6.state \t 0.09873834339001646\n",
      "samples.6.submitter_id \t 0.09873834339001646\n",
      "samples.6.time_between_clamping_and_freezing \t 0.0\n",
      "samples.6.time_between_excision_and_freezing \t 0.0\n",
      "samples.6.tissue_collection_type \t 0.0\n",
      "samples.6.tissue_type \t 0.09873834339001646\n",
      "samples.6.tumor_code \t 0.0\n",
      "samples.6.tumor_code_id \t 0.0\n",
      "samples.6.tumor_descriptor \t 0.0\n",
      "samples.6.updated_datetime \t 0.09873834339001646\n",
      "samples.7.biospecimen_anatomic_site \t 0.03291278113000548\n",
      "samples.7.biospecimen_laterality \t 0.0\n",
      "samples.7.catalog_reference \t 0.0\n",
      "samples.7.composition \t 0.010970927043335162\n",
      "samples.7.created_datetime \t 0.03291278113000548\n",
      "samples.7.current_weight \t 0.0\n",
      "samples.7.days_to_collection \t 0.0\n",
      "samples.7.days_to_sample_procurement \t 0.0\n",
      "samples.7.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.7.distance_normal_to_tumor \t 0.0\n",
      "samples.7.distributor_reference \t 0.0\n",
      "samples.7.freezing_method \t 0.0\n",
      "samples.7.growth_rate \t 0.0\n",
      "samples.7.initial_weight \t 0.0\n",
      "samples.7.intermediate_dimension \t 0.0\n",
      "samples.7.is_ffpe \t 0.03291278113000548\n",
      "samples.7.longest_dimension \t 0.0\n",
      "samples.7.method_of_sample_procurement \t 0.0\n",
      "samples.7.oct_embedded \t 0.0\n",
      "samples.7.passage_count \t 0.0\n",
      "samples.7.pathology_report_uuid \t 0.0\n",
      "samples.7.preservation_method \t 0.0\n",
      "samples.7.sample_id \t 0.03291278113000548\n",
      "samples.7.sample_type \t 0.03291278113000548\n",
      "samples.7.sample_type_id \t 0.0\n",
      "samples.7.shortest_dimension \t 0.0\n",
      "samples.7.state \t 0.03291278113000548\n",
      "samples.7.submitter_id \t 0.03291278113000548\n",
      "samples.7.time_between_clamping_and_freezing \t 0.0\n",
      "samples.7.time_between_excision_and_freezing \t 0.0\n",
      "samples.7.tissue_collection_type \t 0.0\n",
      "samples.7.tissue_type \t 0.03291278113000548\n",
      "samples.7.tumor_code \t 0.0\n",
      "samples.7.tumor_code_id \t 0.0\n",
      "samples.7.tumor_descriptor \t 0.0\n",
      "samples.7.updated_datetime \t 0.03291278113000548\n",
      "samples.8.biospecimen_anatomic_site \t 0.010970927043335162\n",
      "samples.8.biospecimen_laterality \t 0.0\n",
      "samples.8.catalog_reference \t 0.0\n",
      "samples.8.composition \t 0.010970927043335162\n",
      "samples.8.created_datetime \t 0.010970927043335162\n",
      "samples.8.current_weight \t 0.0\n",
      "samples.8.days_to_collection \t 0.0\n",
      "samples.8.days_to_sample_procurement \t 0.0\n",
      "samples.8.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.8.distance_normal_to_tumor \t 0.0\n",
      "samples.8.distributor_reference \t 0.0\n",
      "samples.8.freezing_method \t 0.0\n",
      "samples.8.growth_rate \t 0.0\n",
      "samples.8.initial_weight \t 0.0\n",
      "samples.8.intermediate_dimension \t 0.0\n",
      "samples.8.is_ffpe \t 0.010970927043335162\n",
      "samples.8.longest_dimension \t 0.0\n",
      "samples.8.method_of_sample_procurement \t 0.0\n",
      "samples.8.oct_embedded \t 0.0\n",
      "samples.8.passage_count \t 0.0\n",
      "samples.8.pathology_report_uuid \t 0.0\n",
      "samples.8.preservation_method \t 0.0\n",
      "samples.8.sample_id \t 0.010970927043335162\n",
      "samples.8.sample_type \t 0.010970927043335162\n",
      "samples.8.sample_type_id \t 0.0\n",
      "samples.8.shortest_dimension \t 0.0\n",
      "samples.8.state \t 0.010970927043335162\n",
      "samples.8.submitter_id \t 0.010970927043335162\n",
      "samples.8.time_between_clamping_and_freezing \t 0.0\n",
      "samples.8.time_between_excision_and_freezing \t 0.0\n",
      "samples.8.tissue_collection_type \t 0.0\n",
      "samples.8.tissue_type \t 0.010970927043335162\n",
      "samples.8.tumor_code \t 0.0\n",
      "samples.8.tumor_code_id \t 0.0\n",
      "samples.8.tumor_descriptor \t 0.0\n",
      "samples.8.updated_datetime \t 0.010970927043335162\n",
      "submitter_id \t 100.0\n",
      "tissue_source_site.bcr_id \t 12.057048820625344\n",
      "tissue_source_site.code \t 12.057048820625344\n",
      "tissue_source_site.name \t 12.057048820625344\n",
      "tissue_source_site.project \t 12.057048820625344\n",
      "tissue_source_site.tissue_source_site_id \t 12.057048820625344\n"
     ]
    }
   ],
   "source": [
    "serie = df_case.notnull().mean()*100\n",
    "for name,val in zip(serie.index,serie): \n",
    "    print(name,'\\t',val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(9115, 782)\n"
     ]
    }
   ],
   "source": [
    "print(df_case.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We observe two things: \n",
    "* There are 782 columns and a lot of them a not useful to tackle our problem\n",
    "* There are many empty columns\n",
    "\n",
    "Over those 782 columns, we have decided to eliminate many of them. \n",
    "First, many of them are empty or almost empty, thus, we set a threshold corresponding to 10% of fill rate. Below this trigger, the column will be dropped.\n",
    "Moreover, some columns will resist but with no useful informations, such as the id's. Indeed, we are interested in having information about a certain patient, in one dataframe. Then, having the case_id, which is the id of the patient is enough to link all the dataframes together. Other columns are of no use for preduction such as the ones containing data on the tissue source site ( the hospital it has been taken from the patient), the sample state (which is to say if the data has been released of not)...etc. We also drop them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_case = df_case.drop(df_case.columns[df_case.notnull().mean()<0.1], axis=1)\n",
    "columns_to_drop = ['tissue_source_site.code', 'tissue_source_site.project','tissue_source_site.name', 'diagnoses.0.treatments.1.state', 'diagnoses.0.state', 'diagnoses.0.treatments.0.state', 'exposures.0.state', 'samples.0.state', 'samples.1.state', 'samples.2.state', 'demographic.demographic_id', 'demographic.state', 'demographic.submitter_id', 'demographic.updated_datetime', 'diagnoses.0.diagnosis_id', 'diagnoses.0.icd_10_code', 'diagnoses.0.submitter_id', 'diagnoses.0.treatments.1.updated_datetime', 'diagnoses.0.treatments.1.submitter_id', 'diagnoses.0.treatments.0.updated_datetime', 'diagnoses.0.updated_datetime', 'exposures.0.exposure_id', 'exposures.0.submitter_id', 'exposures.0.updated_datetime', 'samples.0.created_datetime', 'samples.0.sample_id', 'samples.0.sample_type_id', 'samples.0.submitter_id', 'samples.0.updated_datetime', 'samples.1.sample_id', 'samples.1.sample_type_id', 'samples.1.submitter_id', 'samples.2.sample_id', 'samples.2.sample_type_id', 'samples.2.submitter_id', 'tissue_source_site.bcr_id', 'submitter_id', 'diagnoses.0.treatments.0.submitter_id', 'diagnoses.0.treatments.1.treatment_id', 'samples.2.updated_datetime', 'samples.1.updated_datetime', 'diagnoses.0.treatments.0.treatment_id', 'demographic.created_datetime', 'diagnoses.0.created_datetime', 'tissue_source_site.tissue_source_site_id']\n",
    "df_case = df_case.drop(columns_to_drop, axis=1)\n",
    "serie = df_case.notnull().mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>demographic.age_at_index</th>\n",
       "      <th>demographic.days_to_birth</th>\n",
       "      <th>demographic.ethnicity</th>\n",
       "      <th>demographic.gender</th>\n",
       "      <th>demographic.race</th>\n",
       "      <th>demographic.vital_status</th>\n",
       "      <th>demographic.year_of_birth</th>\n",
       "      <th>diagnoses.0.age_at_diagnosis</th>\n",
       "      <th>diagnoses.0.ajcc_pathologic_m</th>\n",
       "      <th>...</th>\n",
       "      <th>samples.0.tissue_type</th>\n",
       "      <th>samples.0.tumor_descriptor</th>\n",
       "      <th>samples.1.is_ffpe</th>\n",
       "      <th>samples.1.oct_embedded</th>\n",
       "      <th>samples.1.sample_type</th>\n",
       "      <th>samples.1.tissue_type</th>\n",
       "      <th>samples.2.is_ffpe</th>\n",
       "      <th>samples.2.oct_embedded</th>\n",
       "      <th>samples.2.sample_type</th>\n",
       "      <th>samples.2.tissue_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>e479200e-a6dc-4973-b326-8ae7eab5bc88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not reported</td>\n",
       "      <td>female</td>\n",
       "      <td>not reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26671.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0532931f-a727-42d8-877f-3e4a20e2ee44</td>\n",
       "      <td>18993.0</td>\n",
       "      <td>-18993.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b30a070d-e090-40cc-8277-33fdc66a7700</td>\n",
       "      <td>27393.0</td>\n",
       "      <td>-27393.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>017a3a78-249c-48e0-a758-f48351ba5734</td>\n",
       "      <td>25567.0</td>\n",
       "      <td>-25567.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1a38613b-e8f8-4ac9-b86d-cae0738d6625</td>\n",
       "      <td>18262.0</td>\n",
       "      <td>-18262.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                case_id  demographic.age_at_index  \\\n",
       "0  e479200e-a6dc-4973-b326-8ae7eab5bc88                       NaN   \n",
       "1  0532931f-a727-42d8-877f-3e4a20e2ee44                   18993.0   \n",
       "2  b30a070d-e090-40cc-8277-33fdc66a7700                   27393.0   \n",
       "3  017a3a78-249c-48e0-a758-f48351ba5734                   25567.0   \n",
       "4  1a38613b-e8f8-4ac9-b86d-cae0738d6625                   18262.0   \n",
       "\n",
       "   demographic.days_to_birth   demographic.ethnicity demographic.gender  \\\n",
       "0                        NaN            not reported             female   \n",
       "1                   -18993.0  not hispanic or latino             female   \n",
       "2                   -27393.0  not hispanic or latino             female   \n",
       "3                   -25567.0  not hispanic or latino             female   \n",
       "4                   -18262.0  not hispanic or latino             female   \n",
       "\n",
       "  demographic.race demographic.vital_status  demographic.year_of_birth  \\\n",
       "0     not reported             Not Reported                        NaN   \n",
       "1            white             Not Reported                        NaN   \n",
       "2            white             Not Reported                        NaN   \n",
       "3            white             Not Reported                        NaN   \n",
       "4            white             Not Reported                        NaN   \n",
       "\n",
       "   diagnoses.0.age_at_diagnosis diagnoses.0.ajcc_pathologic_m  ...  \\\n",
       "0                       26671.0                           NaN  ...   \n",
       "1                           NaN                           NaN  ...   \n",
       "2                           NaN                           NaN  ...   \n",
       "3                           NaN                           NaN  ...   \n",
       "4                           NaN                           NaN  ...   \n",
       "\n",
       "  samples.0.tissue_type samples.0.tumor_descriptor samples.1.is_ffpe  \\\n",
       "0          Not Reported                    Primary               NaN   \n",
       "1          Not Reported                        NaN               NaN   \n",
       "2          Not Reported                        NaN               NaN   \n",
       "3          Not Reported                        NaN               NaN   \n",
       "4          Not Reported                        NaN               NaN   \n",
       "\n",
       "  samples.1.oct_embedded samples.1.sample_type  samples.1.tissue_type  \\\n",
       "0                    NaN                   NaN                    NaN   \n",
       "1                    NaN                   NaN                    NaN   \n",
       "2                    NaN                   NaN                    NaN   \n",
       "3                    NaN                   NaN                    NaN   \n",
       "4                    NaN                   NaN                    NaN   \n",
       "\n",
       "   samples.2.is_ffpe samples.2.oct_embedded samples.2.sample_type  \\\n",
       "0                NaN                    NaN                   NaN   \n",
       "1                NaN                    NaN                   NaN   \n",
       "2                NaN                    NaN                   NaN   \n",
       "3                NaN                    NaN                   NaN   \n",
       "4                NaN                    NaN                   NaN   \n",
       "\n",
       "  samples.2.tissue_type  \n",
       "0                   NaN  \n",
       "1                   NaN  \n",
       "2                   NaN  \n",
       "3                   NaN  \n",
       "4                   NaN  \n",
       "\n",
       "[5 rows x 48 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_case.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Molecular Biology Data\n",
    "\n",
    "In the era of post-genomic living (Lander, E. et al. Initial sequencing and analysis of the human genome. Nature (2001)), our ability to sequence DNA and integrate numerous biological data allows us to develop increasingly personalized approaches to understanding and treating cancer. In this challenge, in combination with \"traditional\" criteria for characterizing cancer (ie: clinical and pathological data), we will also be interested in genetic data.\n",
    "\n",
    "Nous allons nous interesser pour chaque patient aux 50 gènes les plus mutés dans cette population.\n",
    "\n",
    "connus comme étant impliqués dans les cancers d’une façon générale (gènes les plus mutés dans la population tumorale). Pour chaque gène et pour chaque patient, nous allons essayer d’obtenir 3 informations :\n",
    "\n",
    "The molecular data that interests us here are of two kinds:\n",
    "- Genomic data: DNA mutations\n",
    "- Transcriptomic data: number of mRNA copies that each gene produces.\n",
    "\n",
    "#### Genomic data:\n",
    "\n",
    "We will be interested in genetic mutations for the 50 most mutated genes in the population of patients with breast cancer.\n",
    "\n",
    "There are two main types of mutations:\n",
    "- \"Copy Number Variation\" (CNV) type mutations and \"Single somatic Mutation\" (SSM) type mutations.\n",
    "\n",
    "CNVs are a variation in the copy number of a given gene. Indeed, a gene is a sequence of several thousand nucleotide base pairs. A CNV gain causes an increase in the number of copies of this gene, thus being able to produce more RNA transcripts and therefore potentially more proteins leading to a modification of the biology of the cell. A CNV loss will lead to a disappearance of the gene leading to a loss of the protein and therefore of its biological functionality.\n",
    "\n",
    "SSMs are so-called ponctual mutations. The ponctual mutation will change one or a few nucleotides in the DNA sequence. This will change the copied RNA and the RNA-copying ribosome will not make the correct amino acid match. Thus, the protein will be modified and therefore its functionality as well.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "case.case_id                        1.0\n",
       "cnv.chromosome                      1.0\n",
       "cnv.cnv_change                      1.0\n",
       "cnv.consequence.0.consequence_id    1.0\n",
       "cnv.consequence.0.gene.symbol       1.0\n",
       "cnv.end_position                    1.0\n",
       "cnv.gene_level_cn                   1.0\n",
       "cnv.ncbi_build                      1.0\n",
       "cnv.start_position                  1.0\n",
       "cnv.variant_status                  1.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns_to_drop = ['cnv.cnv_id', 'id']\n",
    "df_cnv.drop(columns_to_drop, axis=1, inplace=True)\n",
    "df_cnv.notnull().mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>cnv.chromosome</th>\n",
       "      <th>cnv.cnv_change</th>\n",
       "      <th>cnv.consequence.0.consequence_id</th>\n",
       "      <th>cnv.consequence.0.gene.symbol</th>\n",
       "      <th>cnv.end_position</th>\n",
       "      <th>cnv.gene_level_cn</th>\n",
       "      <th>cnv.ncbi_build</th>\n",
       "      <th>cnv.start_position</th>\n",
       "      <th>cnv.variant_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c5891c7a-fd24-4891-8f51-373967e4a6dd</td>\n",
       "      <td>17</td>\n",
       "      <td>Loss</td>\n",
       "      <td>99c68f83-9514-5628-9623-0ba6095dd90f</td>\n",
       "      <td>BCAS3</td>\n",
       "      <td>61392838</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>60677453</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>d5f2b85a-94a9-4168-bbe8-149ec71342b0</td>\n",
       "      <td>17</td>\n",
       "      <td>Gain</td>\n",
       "      <td>a7e6f054-f5a1-5e7f-a0c0-8daa7373a434</td>\n",
       "      <td>ABCA8</td>\n",
       "      <td>68955392</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>68867292</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>eafa45a4-2fd1-4fa3-860c-9d52ed382b7d</td>\n",
       "      <td>11</td>\n",
       "      <td>Gain</td>\n",
       "      <td>924c27ca-11cf-5b13-95b9-cf518abb142d</td>\n",
       "      <td>ORAOV1</td>\n",
       "      <td>69675416</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>69653076</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>dd0f8e5a-383d-4f03-bb71-0e115d4e442c</td>\n",
       "      <td>11</td>\n",
       "      <td>Gain</td>\n",
       "      <td>00461981-a326-573d-a9a1-ad82872a1aea</td>\n",
       "      <td>SHANK2</td>\n",
       "      <td>71252577</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>70467856</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>233b02f3-c4f0-4a67-9db5-e68d5cdaccb6</td>\n",
       "      <td>8</td>\n",
       "      <td>Gain</td>\n",
       "      <td>c6208422-2c76-553a-b0eb-4472e816381f</td>\n",
       "      <td>MYC</td>\n",
       "      <td>127741434</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>127735434</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           case.case_id  cnv.chromosome cnv.cnv_change  \\\n",
       "0  c5891c7a-fd24-4891-8f51-373967e4a6dd              17           Loss   \n",
       "1  d5f2b85a-94a9-4168-bbe8-149ec71342b0              17           Gain   \n",
       "2  eafa45a4-2fd1-4fa3-860c-9d52ed382b7d              11           Gain   \n",
       "3  dd0f8e5a-383d-4f03-bb71-0e115d4e442c              11           Gain   \n",
       "4  233b02f3-c4f0-4a67-9db5-e68d5cdaccb6               8           Gain   \n",
       "\n",
       "       cnv.consequence.0.consequence_id cnv.consequence.0.gene.symbol  \\\n",
       "0  99c68f83-9514-5628-9623-0ba6095dd90f                         BCAS3   \n",
       "1  a7e6f054-f5a1-5e7f-a0c0-8daa7373a434                         ABCA8   \n",
       "2  924c27ca-11cf-5b13-95b9-cf518abb142d                        ORAOV1   \n",
       "3  00461981-a326-573d-a9a1-ad82872a1aea                        SHANK2   \n",
       "4  c6208422-2c76-553a-b0eb-4472e816381f                           MYC   \n",
       "\n",
       "   cnv.end_position  cnv.gene_level_cn cnv.ncbi_build  cnv.start_position  \\\n",
       "0          61392838               True         GRCh38            60677453   \n",
       "1          68955392               True         GRCh38            68867292   \n",
       "2          69675416               True         GRCh38            69653076   \n",
       "3          71252577               True         GRCh38            70467856   \n",
       "4         127741434               True         GRCh38           127735434   \n",
       "\n",
       "  cnv.variant_status  \n",
       "0         Tumor only  \n",
       "1         Tumor only  \n",
       "2         Tumor only  \n",
       "3         Tumor only  \n",
       "4         Tumor only  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cnv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Now let's look at the SSM dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case.case_id \t 100.0\n",
      "id \t 100.0\n",
      "ssm.chromosome \t 100.0\n",
      "ssm.consequence.0.consequence_id \t 100.0\n",
      "ssm.consequence.0.transcript.gene.symbol \t 100.0\n",
      "ssm.consequence.1.consequence_id \t 91.22283766510438\n",
      "ssm.consequence.1.transcript.gene.symbol \t 91.22283766510438\n",
      "ssm.consequence.10.consequence_id \t 30.336599914784834\n",
      "ssm.consequence.10.transcript.gene.symbol \t 30.336599914784834\n",
      "ssm.consequence.11.consequence_id \t 29.825308904985086\n",
      "ssm.consequence.11.transcript.gene.symbol \t 29.825308904985086\n",
      "ssm.consequence.12.consequence_id \t 26.8001704303366\n",
      "ssm.consequence.12.transcript.gene.symbol \t 26.8001704303366\n",
      "ssm.consequence.13.consequence_id \t 26.54452492543673\n",
      "ssm.consequence.13.transcript.gene.symbol \t 26.54452492543673\n",
      "ssm.consequence.14.consequence_id \t 26.501917341286752\n",
      "ssm.consequence.14.transcript.gene.symbol \t 26.501917341286752\n",
      "ssm.consequence.15.consequence_id \t 26.41670217298679\n",
      "ssm.consequence.15.transcript.gene.symbol \t 26.41670217298679\n",
      "ssm.consequence.16.consequence_id \t 26.37409458883681\n",
      "ssm.consequence.16.transcript.gene.symbol \t 26.37409458883681\n",
      "ssm.consequence.17.consequence_id \t 17.213463996591393\n",
      "ssm.consequence.17.transcript.gene.symbol \t 17.213463996591393\n",
      "ssm.consequence.18.consequence_id \t 16.61695781849169\n",
      "ssm.consequence.18.transcript.gene.symbol \t 16.61695781849169\n",
      "ssm.consequence.19.consequence_id \t 16.233489561141884\n",
      "ssm.consequence.19.transcript.gene.symbol \t 16.233489561141884\n",
      "ssm.consequence.2.consequence_id \t 79.29271410311036\n",
      "ssm.consequence.2.transcript.gene.symbol \t 79.29271410311036\n",
      "ssm.consequence.20.consequence_id \t 16.063059224541966\n",
      "ssm.consequence.20.transcript.gene.symbol \t 16.063059224541966\n",
      "ssm.consequence.21.consequence_id \t 16.063059224541966\n",
      "ssm.consequence.21.transcript.gene.symbol \t 16.063059224541966\n",
      "ssm.consequence.22.consequence_id \t 16.063059224541966\n",
      "ssm.consequence.22.transcript.gene.symbol \t 16.063059224541966\n",
      "ssm.consequence.23.consequence_id \t 16.02045164039199\n",
      "ssm.consequence.23.transcript.gene.symbol \t 16.02045164039199\n",
      "ssm.consequence.24.consequence_id \t 16.02045164039199\n",
      "ssm.consequence.24.transcript.gene.symbol \t 16.02045164039199\n",
      "ssm.consequence.25.consequence_id \t 16.02045164039199\n",
      "ssm.consequence.25.transcript.gene.symbol \t 16.02045164039199\n",
      "ssm.consequence.26.consequence_id \t 16.02045164039199\n",
      "ssm.consequence.26.transcript.gene.symbol \t 16.02045164039199\n",
      "ssm.consequence.3.consequence_id \t 69.91904559011503\n",
      "ssm.consequence.3.transcript.gene.symbol \t 69.91904559011503\n",
      "ssm.consequence.4.consequence_id \t 63.86876864081806\n",
      "ssm.consequence.4.transcript.gene.symbol \t 63.86876864081806\n",
      "ssm.consequence.5.consequence_id \t 59.35236472092033\n",
      "ssm.consequence.5.transcript.gene.symbol \t 59.35236472092033\n",
      "ssm.consequence.6.consequence_id \t 57.09416276097146\n",
      "ssm.consequence.6.transcript.gene.symbol \t 57.09416276097146\n",
      "ssm.consequence.7.consequence_id \t 40.51981252662974\n",
      "ssm.consequence.7.transcript.gene.symbol \t 40.51981252662974\n",
      "ssm.consequence.8.consequence_id \t 34.08606731998295\n",
      "ssm.consequence.8.transcript.gene.symbol \t 34.08606731998295\n",
      "ssm.consequence.9.consequence_id \t 32.12611844908393\n",
      "ssm.consequence.9.transcript.gene.symbol \t 32.12611844908393\n",
      "ssm.cosmic_id.0 \t 70.64337452066468\n",
      "ssm.cosmic_id.1 \t 55.13421389007244\n",
      "ssm.cosmic_id.2 \t 23.988069876438008\n",
      "ssm.cosmic_id.3 \t 22.752449936088624\n",
      "ssm.cosmic_id.4 \t 11.717085641244141\n",
      "ssm.cosmic_id.5 \t 8.351086493395824\n",
      "ssm.cosmic_id.6 \t 5.794631444397103\n",
      "ssm.cosmic_id.7 \t 0.04260758414997869\n",
      "ssm.end_position \t 100.0\n",
      "ssm.genomic_dna_change \t 100.0\n",
      "ssm.mutation_subtype \t 100.0\n",
      "ssm.mutation_type \t 100.0\n",
      "ssm.ncbi_build \t 100.0\n",
      "ssm.reference_allele \t 100.0\n",
      "ssm.ssm_id \t 100.0\n",
      "ssm.start_position \t 100.0\n",
      "ssm.tumor_allele \t 100.0\n"
     ]
    }
   ],
   "source": [
    "serie = df_ssm.notnull().mean()*100\n",
    "for name,val in zip(serie.index,serie): \n",
    "    print(name,'\\t',val)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The consequence_id, as well as the cosmic_id are not useful at all. We delete them all."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>ssm.chromosome</th>\n",
       "      <th>ssm.consequence.0.transcript.gene.symbol</th>\n",
       "      <th>ssm.cosmic_id.0</th>\n",
       "      <th>ssm.end_position</th>\n",
       "      <th>ssm.genomic_dna_change</th>\n",
       "      <th>ssm.mutation_subtype</th>\n",
       "      <th>ssm.mutation_type</th>\n",
       "      <th>ssm.ncbi_build</th>\n",
       "      <th>ssm.reference_allele</th>\n",
       "      <th>ssm.ssm_id</th>\n",
       "      <th>ssm.start_position</th>\n",
       "      <th>ssm.tumor_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>d2694b09-7844-4496-865c-5d3d8cd03330</td>\n",
       "      <td>chr3</td>\n",
       "      <td>MUC4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>195780168</td>\n",
       "      <td>chr3:g.195780168A&gt;T</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>A</td>\n",
       "      <td>92dded19-d0ce-5c34-9d99-169342f9bc37</td>\n",
       "      <td>195780168</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17baef7c-d97d-4b98-ab53-503ef856523d</td>\n",
       "      <td>chr3</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>COSM446001</td>\n",
       "      <td>179203764</td>\n",
       "      <td>chr3:g.179203764A&gt;C</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>A</td>\n",
       "      <td>7ea2ecba-9bbc-5d84-8a58-2ed20f13bdf3</td>\n",
       "      <td>179203764</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>d3d545b3-457f-4389-821f-704cb24aff7f</td>\n",
       "      <td>chr17</td>\n",
       "      <td>TP53</td>\n",
       "      <td>COSM1480087</td>\n",
       "      <td>7676042</td>\n",
       "      <td>chr17:g.7676041_7676042insGA</td>\n",
       "      <td>Small insertion</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>'-</td>\n",
       "      <td>1c4dad33-40cc-58d0-9d8c-4d808cb33401</td>\n",
       "      <td>7676041</td>\n",
       "      <td>GA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>55262fcb-1b01-4480-b322-36570430c917</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ABCA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68883820</td>\n",
       "      <td>chr17:g.68883820C&gt;G</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>C</td>\n",
       "      <td>905a9d14-3a2e-5fb8-994e-48a2e5a1fa54</td>\n",
       "      <td>68883820</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>91b09291-5944-4691-a0d8-7d1ccc649b9b</td>\n",
       "      <td>chr3</td>\n",
       "      <td>MUC4</td>\n",
       "      <td>COSM3847118</td>\n",
       "      <td>195782578</td>\n",
       "      <td>chr3:g.195782578G&gt;A</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>G</td>\n",
       "      <td>eb5acb6d-ebcc-50fc-9dc0-9a7ac5b85fc4</td>\n",
       "      <td>195782578</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           case.case_id ssm.chromosome  \\\n",
       "0  d2694b09-7844-4496-865c-5d3d8cd03330           chr3   \n",
       "1  17baef7c-d97d-4b98-ab53-503ef856523d           chr3   \n",
       "2  d3d545b3-457f-4389-821f-704cb24aff7f          chr17   \n",
       "3  55262fcb-1b01-4480-b322-36570430c917          chr17   \n",
       "4  91b09291-5944-4691-a0d8-7d1ccc649b9b           chr3   \n",
       "\n",
       "  ssm.consequence.0.transcript.gene.symbol ssm.cosmic_id.0  ssm.end_position  \\\n",
       "0                                     MUC4             NaN         195780168   \n",
       "1                                   PIK3CA      COSM446001         179203764   \n",
       "2                                     TP53     COSM1480087           7676042   \n",
       "3                                    ABCA8             NaN          68883820   \n",
       "4                                     MUC4     COSM3847118         195782578   \n",
       "\n",
       "         ssm.genomic_dna_change      ssm.mutation_subtype  \\\n",
       "0           chr3:g.195780168A>T  Single base substitution   \n",
       "1           chr3:g.179203764A>C  Single base substitution   \n",
       "2  chr17:g.7676041_7676042insGA           Small insertion   \n",
       "3           chr17:g.68883820C>G  Single base substitution   \n",
       "4           chr3:g.195782578G>A  Single base substitution   \n",
       "\n",
       "         ssm.mutation_type ssm.ncbi_build ssm.reference_allele  \\\n",
       "0  Simple Somatic Mutation         GRCh38                    A   \n",
       "1  Simple Somatic Mutation         GRCh38                    A   \n",
       "2  Simple Somatic Mutation         GRCh38                   '-   \n",
       "3  Simple Somatic Mutation         GRCh38                    C   \n",
       "4  Simple Somatic Mutation         GRCh38                    G   \n",
       "\n",
       "                             ssm.ssm_id  ssm.start_position ssm.tumor_allele  \n",
       "0  92dded19-d0ce-5c34-9d99-169342f9bc37           195780168                T  \n",
       "1  7ea2ecba-9bbc-5d84-8a58-2ed20f13bdf3           179203764                C  \n",
       "2  1c4dad33-40cc-58d0-9d8c-4d808cb33401             7676041               GA  \n",
       "3  905a9d14-3a2e-5fb8-994e-48a2e5a1fa54            68883820                G  \n",
       "4  eb5acb6d-ebcc-50fc-9dc0-9a7ac5b85fc4           195782578                A  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('ce_id')], axis=1, inplace=True)\n",
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('[0-2][1-9]')], axis=1, inplace=True)\n",
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('[1-9]')], axis=1, inplace=True)\n",
    "columns_to_drop = [\"ssm.cosmid_id.0\",\"id\"]\n",
    "df_ssm.drop(\"id\", axis=1,inplace=True)\n",
    "df_ssm.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Transcriptomic data:\n",
    "\n",
    "It is extremely relevant to add the so-called “transcriptomic” data to these genomic data. They correspond to the level of expression of the gene. The transcriptome represents the amount of mRNA produced for each gene. It is therefore interesting to look at this level of gene expression because it provides more direct information on the number of proteins and their types than analysis of DNA alone."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Top 50 most mutated genes\n",
    "\n",
    "Cancer are provoked by DNA mutations. Knowing what mutation has happened, on which gene is a valuable information to predict what cure shall be given. We chose to use the top 50 mutated genes in breast cancer. This choice is totally arbitrary and is done in order to limit the input dimension and conserve most of the data's value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_score</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>451.0</td>\n",
       "      <td>PIK3CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>404.0</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>322.0</td>\n",
       "      <td>MUC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>320.0</td>\n",
       "      <td>SHANK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>318.0</td>\n",
       "      <td>ANO1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   _score  symbol\n",
       "0   451.0  PIK3CA\n",
       "1   404.0    TP53\n",
       "2   322.0    MUC4\n",
       "3   320.0  SHANK2\n",
       "4   318.0    ANO1"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Whole dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/gdc_final_database.csv', sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>case_id</th>\n",
       "      <th>demographic.age_at_index</th>\n",
       "      <th>demographic.days_to_birth</th>\n",
       "      <th>demographic.ethnicity</th>\n",
       "      <th>demographic.gender</th>\n",
       "      <th>demographic.race</th>\n",
       "      <th>demographic.vital_status</th>\n",
       "      <th>demographic.year_of_birth</th>\n",
       "      <th>diagnoses.0.age_at_diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>arn.HTseq_counts.TG</th>\n",
       "      <th>arn.HTseq_counts.TP53</th>\n",
       "      <th>arn.HTseq_counts.TPCN2</th>\n",
       "      <th>arn.HTseq_counts.TRPS1</th>\n",
       "      <th>arn.HTseq_counts.TTN</th>\n",
       "      <th>ssm.importance.AC108925.1</th>\n",
       "      <th>ssm.importance.FADD</th>\n",
       "      <th>ssm.importance.FGF19</th>\n",
       "      <th>ssm.importance.DNAJB6P5</th>\n",
       "      <th>arn.HTseq_counts.AC108925.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>50.0</td>\n",
       "      <td>-18345.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>18345.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2997.0</td>\n",
       "      <td>655.0</td>\n",
       "      <td>8320.0</td>\n",
       "      <td>341.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>56.0</td>\n",
       "      <td>-20713.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1954.0</td>\n",
       "      <td>20713.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19.0</td>\n",
       "      <td>6779.0</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>13940.0</td>\n",
       "      <td>306.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>61.0</td>\n",
       "      <td>-22497.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1951.0</td>\n",
       "      <td>22497.0</td>\n",
       "      <td>...</td>\n",
       "      <td>56.0</td>\n",
       "      <td>4543.0</td>\n",
       "      <td>3168.0</td>\n",
       "      <td>15170.0</td>\n",
       "      <td>1334.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>71.0</td>\n",
       "      <td>-26274.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1939.0</td>\n",
       "      <td>26274.0</td>\n",
       "      <td>...</td>\n",
       "      <td>391.0</td>\n",
       "      <td>7822.0</td>\n",
       "      <td>1206.0</td>\n",
       "      <td>212884.0</td>\n",
       "      <td>19495.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>01263518-5f7c-49dc-8d7e-84b0c03a6a63</td>\n",
       "      <td>76.0</td>\n",
       "      <td>-27942.0</td>\n",
       "      <td>not reported</td>\n",
       "      <td>female</td>\n",
       "      <td>not reported</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1933.0</td>\n",
       "      <td>27942.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20.0</td>\n",
       "      <td>5374.0</td>\n",
       "      <td>2801.0</td>\n",
       "      <td>40581.0</td>\n",
       "      <td>244.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 198 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0                               case_id  demographic.age_at_index  \\\n",
       "0           0  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                      50.0   \n",
       "1           1  00a2d166-78c9-4687-a195-3d6315c27574                      56.0   \n",
       "2           2  00b11ca8-8540-4a3d-b602-ec754b00230b                      61.0   \n",
       "3           3  011b9b2d-ebe5-42bf-9662-d922faccc7a1                      71.0   \n",
       "4           4  01263518-5f7c-49dc-8d7e-84b0c03a6a63                      76.0   \n",
       "\n",
       "   demographic.days_to_birth   demographic.ethnicity demographic.gender  \\\n",
       "0                   -18345.0  not hispanic or latino             female   \n",
       "1                   -20713.0  not hispanic or latino             female   \n",
       "2                   -22497.0  not hispanic or latino             female   \n",
       "3                   -26274.0  not hispanic or latino             female   \n",
       "4                   -27942.0            not reported             female   \n",
       "\n",
       "  demographic.race demographic.vital_status  demographic.year_of_birth  \\\n",
       "0            white                    Alive                     1955.0   \n",
       "1            white                    Alive                     1954.0   \n",
       "2            white                    Alive                     1951.0   \n",
       "3            white                    Alive                     1939.0   \n",
       "4     not reported                    Alive                     1933.0   \n",
       "\n",
       "   diagnoses.0.age_at_diagnosis  ... arn.HTseq_counts.TG  \\\n",
       "0                       18345.0  ...                18.0   \n",
       "1                       20713.0  ...                19.0   \n",
       "2                       22497.0  ...                56.0   \n",
       "3                       26274.0  ...               391.0   \n",
       "4                       27942.0  ...                20.0   \n",
       "\n",
       "  arn.HTseq_counts.TP53 arn.HTseq_counts.TPCN2 arn.HTseq_counts.TRPS1  \\\n",
       "0                2997.0                  655.0                 8320.0   \n",
       "1                6779.0                 1123.0                13940.0   \n",
       "2                4543.0                 3168.0                15170.0   \n",
       "3                7822.0                 1206.0               212884.0   \n",
       "4                5374.0                 2801.0                40581.0   \n",
       "\n",
       "  arn.HTseq_counts.TTN ssm.importance.AC108925.1  ssm.importance.FADD  \\\n",
       "0                341.0                       0.0                  0.0   \n",
       "1                306.0                       0.0                  0.0   \n",
       "2               1334.0                       0.0                  0.0   \n",
       "3              19495.0                       0.0                  0.0   \n",
       "4                244.0                       0.0                  0.0   \n",
       "\n",
       "   ssm.importance.FGF19 ssm.importance.DNAJB6P5 arn.HTseq_counts.AC108925.1  \n",
       "0                   0.0                     0.0                         0.0  \n",
       "1                   0.0                     0.0                         0.0  \n",
       "2                   0.0                     0.0                         0.0  \n",
       "3                   0.0                     0.0                         0.0  \n",
       "4                   0.0                     0.0                         0.0  \n",
       "\n",
       "[5 rows x 198 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unnamed: 0 \t 100.0\n",
      "case_id \t 100.0\n",
      "demographic.age_at_index \t 79.62291515591008\n",
      "demographic.days_to_birth \t 78.60768672951414\n",
      "demographic.ethnicity \t 99.92748368382887\n",
      "demographic.gender \t 99.92748368382887\n",
      "demographic.race \t 99.92748368382887\n",
      "demographic.vital_status \t 99.92748368382887\n",
      "demographic.year_of_birth \t 79.4778825235678\n",
      "diagnoses.0.age_at_diagnosis \t 96.44670050761421\n",
      "diagnoses.0.ajcc_pathologic_m \t 79.62291515591008\n",
      "diagnoses.0.ajcc_pathologic_n \t 79.62291515591008\n",
      "diagnoses.0.ajcc_pathologic_stage \t 88.46990572878897\n",
      "diagnoses.0.ajcc_pathologic_t \t 79.62291515591008\n",
      "diagnoses.0.ajcc_staging_system_edition \t 69.39811457577954\n",
      "diagnoses.0.classification_of_tumor \t 89.2675852066715\n",
      "diagnoses.0.days_to_diagnosis \t 79.69543147208121\n",
      "diagnoses.0.days_to_last_follow_up \t 72.08121827411168\n",
      "diagnoses.0.last_known_disease_status \t 89.2675852066715\n",
      "diagnoses.0.morphology \t 99.92748368382887\n",
      "diagnoses.0.primary_diagnosis \t 99.92748368382887\n",
      "diagnoses.0.prior_malignancy \t 88.5424220449601\n",
      "diagnoses.0.prior_treatment \t 79.69543147208121\n",
      "diagnoses.0.progression_or_recurrence \t 89.2675852066715\n",
      "diagnoses.0.site_of_resection_or_biopsy \t 99.92748368382887\n",
      "diagnoses.0.synchronous_malignancy \t 79.62291515591008\n",
      "diagnoses.0.tissue_or_organ_of_origin \t 99.92748368382887\n",
      "diagnoses.0.treatments.0.treatment_or_therapy \t 79.69543147208121\n",
      "diagnoses.0.treatments.0.treatment_type \t 79.69543147208121\n",
      "diagnoses.0.treatments.1.treatment_or_therapy \t 79.69543147208121\n",
      "diagnoses.0.treatments.1.treatment_type \t 79.69543147208121\n",
      "diagnoses.0.tumor_grade \t 89.34010152284264\n",
      "diagnoses.0.tumor_stage \t 89.2675852066715\n",
      "diagnoses.0.year_of_diagnosis \t 79.4778825235678\n",
      "exposures.0.alcohol_history \t 79.62291515591008\n",
      "id \t 100.0\n",
      "samples.0.is_ffpe \t 90.35532994923858\n",
      "samples.0.oct_embedded \t 79.76794778825236\n",
      "samples.0.sample_type \t 100.0\n",
      "samples.0.tissue_type \t 100.0\n",
      "samples.0.tumor_descriptor \t 0.0\n",
      "samples.1.is_ffpe \t 89.34010152284264\n",
      "samples.1.oct_embedded \t 79.69543147208121\n",
      "samples.1.sample_type \t 99.0572878897752\n",
      "samples.1.tissue_type \t 99.0572878897752\n",
      "samples.2.is_ffpe \t 86.22189992748368\n",
      "samples.2.oct_embedded \t 76.93981145757796\n",
      "samples.2.sample_type \t 87.81725888324873\n",
      "samples.2.tissue_type \t 87.81725888324873\n",
      "ssm.importance.ABCA10 \t 58.66569978245105\n",
      "ssm.importance.ABCA6 \t 58.66569978245105\n",
      "ssm.importance.ABCA8 \t 58.66569978245105\n",
      "ssm.importance.ABCA9 \t 58.66569978245105\n",
      "ssm.importance.AMZ2 \t 58.66569978245105\n",
      "ssm.importance.ANO1 \t 58.66569978245105\n",
      "ssm.importance.ARSG \t 58.66569978245105\n",
      "ssm.importance.BCAS3 \t 58.66569978245105\n",
      "ssm.importance.BPTF \t 58.66569978245105\n",
      "ssm.importance.CCND1 \t 58.66569978245105\n",
      "ssm.importance.CEP112 \t 58.66569978245105\n",
      "ssm.importance.CPT1A \t 58.66569978245105\n",
      "ssm.importance.CSMD3 \t 58.66569978245105\n",
      "ssm.importance.CTTN \t 58.66569978245105\n",
      "ssm.importance.DNAH17 \t 58.66569978245105\n",
      "ssm.importance.FGF3 \t 58.66569978245105\n",
      "ssm.importance.FGF4 \t 58.66569978245105\n",
      "ssm.importance.FNTA \t 58.66569978245105\n",
      "ssm.importance.HELZ \t 58.66569978245105\n",
      "ssm.importance.IGHMBP2 \t 58.66569978245105\n",
      "ssm.importance.KPNA2 \t 58.66569978245105\n",
      "ssm.importance.MRGPRD \t 58.66569978245105\n",
      "ssm.importance.MRGPRF \t 58.66569978245105\n",
      "ssm.importance.MUC4 \t 58.66569978245105\n",
      "ssm.importance.MYC \t 58.66569978245105\n",
      "ssm.importance.MYEOV \t 58.66569978245105\n",
      "ssm.importance.NOL11 \t 58.66569978245105\n",
      "ssm.importance.NOTCH2 \t 58.66569978245105\n",
      "ssm.importance.ORAOV1 \t 58.66569978245105\n",
      "ssm.importance.PIK3CA \t 58.66569978245105\n",
      "ssm.importance.PITPNC1 \t 58.66569978245105\n",
      "ssm.importance.POU5F1B \t 58.66569978245105\n",
      "ssm.importance.PPFIA1 \t 58.66569978245105\n",
      "ssm.importance.PRKAR1A \t 58.66569978245105\n",
      "ssm.importance.PRKCA \t 58.66569978245105\n",
      "ssm.importance.RNF213 \t 58.66569978245105\n",
      "ssm.importance.RYR2 \t 58.66569978245105\n",
      "ssm.importance.SDK2 \t 58.66569978245105\n",
      "ssm.importance.SHANK2 \t 58.66569978245105\n",
      "ssm.importance.SLC16A6 \t 58.66569978245105\n",
      "ssm.importance.TANC2 \t 58.66569978245105\n",
      "ssm.importance.TG \t 58.66569978245105\n",
      "ssm.importance.TP53 \t 58.66569978245105\n",
      "ssm.importance.TPCN2 \t 58.66569978245105\n",
      "ssm.importance.TRPS1 \t 58.66569978245105\n",
      "ssm.importance.TTN \t 58.66569978245105\n",
      "cnv.change.ABCA10 \t 57.868020304568525\n",
      "cnv.change.ABCA6 \t 57.868020304568525\n",
      "cnv.change.ABCA8 \t 57.868020304568525\n",
      "cnv.change.ABCA9 \t 57.868020304568525\n",
      "cnv.change.AC108925.1 \t 57.868020304568525\n",
      "cnv.change.AMZ2 \t 57.868020304568525\n",
      "cnv.change.ANO1 \t 57.868020304568525\n",
      "cnv.change.AP001888.1 \t 57.868020304568525\n",
      "cnv.change.ARSG \t 57.868020304568525\n",
      "cnv.change.BCAS3 \t 57.868020304568525\n",
      "cnv.change.BPTF \t 57.868020304568525\n",
      "cnv.change.CCND1 \t 57.868020304568525\n",
      "cnv.change.CEP112 \t 57.868020304568525\n",
      "cnv.change.CPT1A \t 57.868020304568525\n",
      "cnv.change.CSMD3 \t 57.868020304568525\n",
      "cnv.change.CTTN \t 57.868020304568525\n",
      "cnv.change.DNAH17 \t 57.868020304568525\n",
      "cnv.change.FADD \t 57.868020304568525\n",
      "cnv.change.FGF19 \t 57.868020304568525\n",
      "cnv.change.FGF3 \t 57.868020304568525\n",
      "cnv.change.FGF4 \t 57.868020304568525\n",
      "cnv.change.FNTA \t 57.868020304568525\n",
      "cnv.change.HELZ \t 57.868020304568525\n",
      "cnv.change.IGHMBP2 \t 57.868020304568525\n",
      "cnv.change.KPNA2 \t 57.868020304568525\n",
      "cnv.change.MRGPRD \t 57.868020304568525\n",
      "cnv.change.MRGPRF \t 57.868020304568525\n",
      "cnv.change.MUC4 \t 57.868020304568525\n",
      "cnv.change.MYC \t 57.868020304568525\n",
      "cnv.change.MYEOV \t 57.868020304568525\n",
      "cnv.change.NOL11 \t 57.868020304568525\n",
      "cnv.change.NOTCH2 \t 57.868020304568525\n",
      "cnv.change.ORAOV1 \t 57.868020304568525\n",
      "cnv.change.PIK3CA \t 57.868020304568525\n",
      "cnv.change.PITPNC1 \t 57.868020304568525\n",
      "cnv.change.POU5F1B \t 57.868020304568525\n",
      "cnv.change.PPFIA1 \t 57.868020304568525\n",
      "cnv.change.PRKAR1A \t 57.868020304568525\n",
      "cnv.change.PRKCA \t 57.868020304568525\n",
      "cnv.change.RNF213 \t 57.868020304568525\n",
      "cnv.change.RYR2 \t 57.868020304568525\n",
      "cnv.change.SDK2 \t 57.868020304568525\n",
      "cnv.change.SHANK2 \t 57.868020304568525\n",
      "cnv.change.SLC16A6 \t 57.868020304568525\n",
      "cnv.change.TANC2 \t 57.868020304568525\n",
      "cnv.change.TG \t 57.868020304568525\n",
      "cnv.change.TP53 \t 57.868020304568525\n",
      "cnv.change.TPCN2 \t 57.868020304568525\n",
      "cnv.change.TRPS1 \t 57.868020304568525\n",
      "cnv.change.TTN \t 57.868020304568525\n",
      "arn.HTseq_counts.ABCA10 \t 99.56490210297318\n",
      "arn.HTseq_counts.ABCA6 \t 99.56490210297318\n",
      "arn.HTseq_counts.ABCA8 \t 99.56490210297318\n",
      "arn.HTseq_counts.ABCA9 \t 99.56490210297318\n",
      "arn.HTseq_counts.AMZ2 \t 99.56490210297318\n",
      "arn.HTseq_counts.ANO1 \t 99.56490210297318\n",
      "arn.HTseq_counts.ARSG \t 99.56490210297318\n",
      "arn.HTseq_counts.BCAS3 \t 99.56490210297318\n",
      "arn.HTseq_counts.BPTF \t 99.56490210297318\n",
      "arn.HTseq_counts.CCND1 \t 99.56490210297318\n",
      "arn.HTseq_counts.CEP112 \t 99.56490210297318\n",
      "arn.HTseq_counts.CPT1A \t 99.56490210297318\n",
      "arn.HTseq_counts.CSMD3 \t 99.56490210297318\n",
      "arn.HTseq_counts.CTTN \t 99.56490210297318\n",
      "arn.HTseq_counts.DNAH17 \t 99.56490210297318\n",
      "arn.HTseq_counts.DNAJB6P5 \t 100.0\n",
      "arn.HTseq_counts.FADD \t 100.0\n",
      "arn.HTseq_counts.FGF19 \t 100.0\n",
      "arn.HTseq_counts.FGF3 \t 99.56490210297318\n",
      "arn.HTseq_counts.FGF4 \t 99.56490210297318\n",
      "arn.HTseq_counts.FNTA \t 99.56490210297318\n",
      "arn.HTseq_counts.HELZ \t 99.56490210297318\n",
      "arn.HTseq_counts.IGHMBP2 \t 99.56490210297318\n",
      "arn.HTseq_counts.KPNA2 \t 99.56490210297318\n",
      "arn.HTseq_counts.MRGPRD \t 99.56490210297318\n",
      "arn.HTseq_counts.MRGPRF \t 99.56490210297318\n",
      "arn.HTseq_counts.MUC4 \t 99.56490210297318\n",
      "arn.HTseq_counts.MYC \t 99.56490210297318\n",
      "arn.HTseq_counts.MYEOV \t 99.56490210297318\n",
      "arn.HTseq_counts.NOL11 \t 99.56490210297318\n",
      "arn.HTseq_counts.NOTCH2 \t 99.56490210297318\n",
      "arn.HTseq_counts.PIK3CA \t 99.56490210297318\n",
      "arn.HTseq_counts.PITPNC1 \t 99.56490210297318\n",
      "arn.HTseq_counts.POU5F1B \t 99.56490210297318\n",
      "arn.HTseq_counts.PPFIA1 \t 99.56490210297318\n",
      "arn.HTseq_counts.PRKAR1A \t 99.56490210297318\n",
      "arn.HTseq_counts.PRKCA \t 99.56490210297318\n",
      "arn.HTseq_counts.RNF213 \t 99.56490210297318\n",
      "arn.HTseq_counts.RYR2 \t 99.56490210297318\n",
      "arn.HTseq_counts.SDK2 \t 99.56490210297318\n",
      "arn.HTseq_counts.SHANK2 \t 99.56490210297318\n",
      "arn.HTseq_counts.SLC16A6 \t 99.56490210297318\n",
      "arn.HTseq_counts.TANC2 \t 99.56490210297318\n",
      "arn.HTseq_counts.TG \t 99.56490210297318\n",
      "arn.HTseq_counts.TP53 \t 99.56490210297318\n",
      "arn.HTseq_counts.TPCN2 \t 99.56490210297318\n",
      "arn.HTseq_counts.TRPS1 \t 99.56490210297318\n",
      "arn.HTseq_counts.TTN \t 99.56490210297318\n",
      "ssm.importance.AC108925.1 \t 100.0\n",
      "ssm.importance.FADD \t 100.0\n",
      "ssm.importance.FGF19 \t 100.0\n",
      "ssm.importance.DNAJB6P5 \t 100.0\n",
      "arn.HTseq_counts.AC108925.1 \t 100.0\n"
     ]
    }
   ],
   "source": [
    "serie = df.notnull().mean()*100\n",
    "for name,val in zip(serie.index,serie): \n",
    "    print(name,'\\t',val)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One can wonder why there are so many columns. Since we have chosen to have the importance of the CNV and SSM for all 50 genes, it adds us automatically a hundred columns. The one hot-encoding rises the dimension gthe problem."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge of the dataframes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Limitation of this challenge:\n",
    "\n",
    "The data has been simplified for easier analysis. There are many different types of genomic mutations, involving different biological changes. To simplify this aspect, we produced a relatively arbitrary intermediate score to quantify the importance of the mutations for each gene.\n",
    "\n",
    "@Maxime: insert code with score and explanation of the choice of scoring.\n",
    "\n",
    "There are more different RNAs than genes because the process of \"alternative RNA splicing\" adds a degree of freedom for a gene to encode several different proteins. We have considered here all the mRNAs of a gene as coming from a single transcript.\n",
    "\n",
    "@Maxime: insert code and explanation.\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exploratory Data Analysis\n",
    "\n",
    "Pour que le challenger puisse naviguer dans ces données, nous avons préalablement collecté et formaté ces caractéristiques dans un dataframe de grande dimension (dim*dim). Pour plus d'informations sur l'obtention de ces données: voir la partie **Data loading**.\n",
    "\n",
    "## Clinical \n",
    "\n",
    "Number of cases\n",
    "Disease Type (histology)\n",
    "  - number of each subtypes (can we split ductal and lobular ?)\n",
    "  - Age at Diagnosis for each subtype → statistical difference for each type ? (p-value)\n",
    "  - Survival for each disease type → statistical difference for each type ? (p-value)\n",
    "  - General treatment for each subtype ? (Treatment Type in GDC) → normally we just have 1098 cases for which we have pharmaceutical therapy/radiation therapy\n",
    "      Surgery ?\n",
    "      Radiotherapie?\n",
    "      Chemotherapy ?\n",
    "      Which drugs are used ?\n",
    "\n",
    "\n",
    "## Genomic Data :\n",
    "\n",
    "50 most mutated genes\n",
    "  General number of mutations for each histological subtypes\n",
    "  Are the three most mutated genes known for being involved with each cancer ? \n",
    "  Is the presence of these mutations correlated with bad survival ?\n",
    "Let's be a little bit more precise : many types of mutations\n",
    "\n",
    "    SSM\n",
    "      Types of SSM → Explication Louis. pie chart by histo types ?\n",
    "      frequency of SSM (by histological type ?)\n",
    "      People with more than one SSM in the same gene → we’ll have to deal with that.\n",
    "      Other ?\n",
    "    CNV\n",
    "      Types of CNV → Explication Louis. Pie chart by histo types ?\n",
    "      frequency of CNV (by histological type ?)\n",
    "      Comparison SSM/CNV\n",
    "      \n",
    "Within the mutations : compare top 3 for CNV (in frequency) vs top 3 for SSM → is there a difference ? is it known ? (biblio check Louis)\n",
    "\n",
    "As we explained in the scientifical background → the impact of genetic mutations on cancer biology mostly depends on protein modification. Having information on the amount of RNA should give additional information (especially in the case of CNV -> is an increase in the number of copies associated with an increase in the number of transcripts?)\n",
    "\n",
    "## Transcriptomic Data\n",
    "\n",
    "for how many people did we get the transcriptomic data ?\n",
    "Correlation (Heatmap) for CNV gain/loss and increase/loss number of transcript for these genes. \n",
    "→ focus on HER2 : (HER2 amplification is super important in breast cancer treatment → Target chimio if amplified)\n",
    "correlation for each histological subtype.\n",
    "Difference in terms of survival (how to plot that precisely ?\n",
    "\n",
    "\n",
    "# Submission\n",
    "\n",
    "Travail de prédiction :\n",
    "\n",
    "## Best objective : Prediction of the most effective treatment in patients with breast cancer\n",
    "## Side objective : Histological category prediction\n",
    "## Side objective : Survival prediction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
