{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cancer prediction - starting_kit\n",
    "This notebook can be consulted as an HTML page following this link.\n",
    "\n",
    "\n",
    "This notebook aims at defining the theoretical understanding of cancer prediction and its correlation with genes. \n",
    "\n",
    "In addition to the medical knowldege, the data set which will be used will also be presented and illustrated. \n",
    "\n",
    "Last, an initial machine learning example will be provided in order to initiate your contribution to the challenge by proving you with an basic working solution.\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In 2018 there have been 18million new cases of cancer and 10 millions death. About 20% of men will have a cancer and 17% of the woman. 1 man over 8 will die of cancer and 1 woman over 11.\n",
    "Living with a cancer five year after its diagnosis is called prevalence. 50 million in the world are in this case.\n",
    "\n",
    "The rise of cancer in our societies has multiple origins : demographic growth, the ageing of the population, social and economical factors. The type of cancers shifts with the development of countries from cancers linked to poverty to cancers associated with industrialized way of life. \n",
    "\n",
    "Europe concentrates 23.4% of the cancers and 20.3% of the death due to cancer even though it only represents 9% of the worldwide population. United States and Canada together count for 21% of the cancer cases, 14% of the death with 13.3% of the population. \n",
    "\n",
    "\n",
    "A cancer is caused by a mutation of the DNA, thus, the cell’s reproduction becomes out of control. Many mutation can have the same effect, resulting in many different types of cancer. That is why cancer cannot can be cured as simply as other pathologies. \n",
    "\n",
    "One of the major aspects of treating a cancer is detecting is, knowing which mutation of the DNA it is, what are the specific consequences of those mutations on the cells which then gives the doctor an indication about how to treat it. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Paragraphe à propos de la récupération des datas par Joël**\n",
    "Expliquer que gdc_cancer est une aggrégation de multiples db ce qui donne des données à la fois larges mais aussi clairsemées par endroit. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Treatment of the data**\n",
    "\n",
    "The data obtained through the above process gave us 5 datasets :\n",
    "- df_case: General informations about the patient, age, sexe, ethnicity, multiple risk factors, diagnosis informations, what treatment has been given, the stage of the tumor, caracteristics of the tumor...etc\n",
    "- cf_cnv: \n",
    "- dg_gene: Top 50 of the most mutated genes\n",
    "- df_ssm: Simple somatic mutation. It gives the type of mutations, the allele of reference, the genomic DNA change...etc\n",
    "\n",
    "As the gdc cancer is an aggregation of about 15 databases, there are many valuable informations, but most of the time it is very sparse. Thus, most fields in our datasets are empty. However, this does not imply that we cannot extract value from this data. First, we shall explore, clean\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "E:\\Programmes\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3146: DtypeWarning: Columns (0,1,2,3,4,5,6,7,8,9,10,12,13,14,15,16,17,18,19,20,21,22,23,25,28,29,51,53,54,55,56,57,60,61,62,87,97,119,139,140,148,151,154,156,157,161,162,163,165,166,168,171,175,176,177,185,186,187,189,190,192,199,200,201,209,210,211,213,214,216,219,223,224,225,233,234,235,237,238,239,241,242,243,244,245,246,247,248,249,251,258,259,260,268,269,270,272,273,275,282,283,284,292,293,294,296,297,299,302,306,307,316,317,318,320,321,323,326,330,331,340,342,344,345,347,353,354,355,364,365,366,368,369,371,374,378,379,388,389,390,392,393,410,416,419,428,429,437,440,441,444,446,448,449,450,452,455,467,470,472,473,488,491,492,503,506,508,509,510,511,514,515,519,523,524,527,528,539,542,544,545,546,547,550,551,555,559,560,563,564,575,578,580,581,582,583,586,587,591,595,596,599,600,611,614,616,618,619,622,623,627,631,632,635,636,647,654,655,658,659,663,667,668,672,683,690,691,694,695,699,703,704,707,708,719,726,727,730,731,735,739,740,743,744,755,762,763,766,767,771,775,777,778,779,780,781) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# We import the data that has already been downloaded\n",
    "df_case = pd.read_csv('data/case.tsv', sep='\\t')\n",
    "df_cnv = pd.read_csv('data/cnv.tsv', sep='\\t')\n",
    "df_gene = pd.read_csv('data/gene.tsv', sep='\\t')\n",
    "df_ssm = pd.read_csv('data/ssm.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: case\n",
    "\n",
    "Information on each case such as age, alcohol, tobacco, BMI (Body Mass Index), most of the values are not availale (NaN)\n",
    "\n",
    "36 columns and 9115 rows.\n",
    "\n",
    "The case_id column will be the key for the merge with the other dataframes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, let's start by looking at how filled the dataset is. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotations.0.annotation_id \t 0.003181568842567197\n",
      "annotations.0.case_id \t 0.003181568842567197\n",
      "annotations.0.category \t 0.003181568842567197\n",
      "annotations.0.classification \t 0.003181568842567197\n",
      "annotations.0.created_datetime \t 0.003181568842567197\n",
      "annotations.0.entity_id \t 0.003181568842567197\n",
      "annotations.0.entity_submitter_id \t 0.003181568842567197\n",
      "annotations.0.entity_type \t 0.003181568842567197\n",
      "annotations.0.notes \t 0.003181568842567197\n",
      "annotations.0.state \t 0.003181568842567197\n",
      "annotations.0.status \t 0.003181568842567197\n",
      "annotations.0.submitter_id \t 0.003181568842567197\n",
      "annotations.0.updated_datetime \t 0.003181568842567197\n",
      "annotations.1.annotation_id \t 0.0006582556226001097\n",
      "annotations.1.case_id \t 0.0006582556226001097\n",
      "annotations.1.category \t 0.0006582556226001097\n",
      "annotations.1.classification \t 0.0006582556226001097\n",
      "annotations.1.created_datetime \t 0.0006582556226001097\n",
      "annotations.1.entity_id \t 0.0006582556226001097\n",
      "annotations.1.entity_submitter_id \t 0.0006582556226001097\n",
      "annotations.1.entity_type \t 0.0006582556226001097\n",
      "annotations.1.notes \t 0.0006582556226001097\n",
      "annotations.1.state \t 0.0006582556226001097\n",
      "annotations.1.status \t 0.0006582556226001097\n",
      "annotations.1.submitter_id \t 0.0006582556226001097\n",
      "annotations.1.updated_datetime \t 0.0006582556226001097\n",
      "case_id \t 1.0\n",
      "demographic.age_at_index \t 0.6760285244103127\n",
      "demographic.age_is_obfuscated \t 0.00010970927043335162\n",
      "demographic.cause_of_death \t 0.00010970927043335162\n",
      "demographic.cause_of_death_source \t 0.0\n",
      "demographic.country_of_residence_at_enrollment \t 0.0\n",
      "demographic.created_datetime \t 0.8794295117937466\n",
      "demographic.days_to_birth \t 0.6744925946242457\n",
      "demographic.days_to_death \t 0.016675809105869445\n",
      "demographic.demographic_id \t 0.9998902907295667\n",
      "demographic.ethnicity \t 0.9998902907295667\n",
      "demographic.gender \t 0.9998902907295667\n",
      "demographic.occupation_duration_years \t 0.0\n",
      "demographic.premature_at_birth \t 0.0\n",
      "demographic.race \t 0.9998902907295667\n",
      "demographic.state \t 0.9998902907295667\n",
      "demographic.submitter_id \t 0.9998902907295667\n",
      "demographic.updated_datetime \t 0.9998902907295667\n",
      "demographic.vital_status \t 0.9998902907295667\n",
      "demographic.weeks_gestation_at_birth \t 0.0\n",
      "demographic.year_of_birth \t 0.12024136039495337\n",
      "demographic.year_of_death \t 0.011409764125068568\n",
      "diagnoses.0.age_at_diagnosis \t 0.4365331870543061\n",
      "diagnoses.0.ajcc_clinical_m \t 0.0\n",
      "diagnoses.0.ajcc_clinical_n \t 0.0\n",
      "diagnoses.0.ajcc_clinical_stage \t 0.00010970927043335162\n",
      "diagnoses.0.ajcc_clinical_t \t 0.0\n",
      "diagnoses.0.ajcc_pathologic_m \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_n \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_stage \t 0.13395501919912234\n",
      "diagnoses.0.ajcc_pathologic_t \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_staging_system_edition \t 0.1049917718047175\n",
      "diagnoses.0.anaplasia_present \t 0.0\n",
      "diagnoses.0.anaplasia_present_type \t 0.0\n",
      "diagnoses.0.ann_arbor_b_symptoms \t 0.00010970927043335162\n",
      "diagnoses.0.ann_arbor_clinical_stage \t 0.00010970927043335162\n",
      "diagnoses.0.ann_arbor_extranodal_involvement \t 0.00010970927043335162\n",
      "diagnoses.0.ann_arbor_pathologic_stage \t 0.0\n",
      "diagnoses.0.best_overall_response \t 0.0\n",
      "diagnoses.0.breslow_thickness \t 0.0\n",
      "diagnoses.0.burkitt_lymphoma_clinical_variant \t 0.0\n",
      "diagnoses.0.child_pugh_classification \t 0.0\n",
      "diagnoses.0.circumferential_resection_margin \t 0.0\n",
      "diagnoses.0.classification_of_tumor \t 0.4185408667032364\n",
      "diagnoses.0.cog_liver_stage \t 0.0\n",
      "diagnoses.0.cog_neuroblastoma_risk_group \t 0.0\n",
      "diagnoses.0.cog_renal_stage \t 0.0\n",
      "diagnoses.0.cog_rhabdomyosarcoma_risk_group \t 0.0\n",
      "diagnoses.0.created_datetime \t 0.8794295117937466\n",
      "diagnoses.0.days_to_best_overall_response \t 0.0\n",
      "diagnoses.0.days_to_diagnosis \t 0.12057048820625343\n",
      "diagnoses.0.days_to_last_follow_up \t 0.10905101481075151\n",
      "diagnoses.0.days_to_last_known_disease_status \t 0.0\n",
      "diagnoses.0.days_to_recurrence \t 0.0\n",
      "diagnoses.0.diagnosis_id \t 0.9998902907295667\n",
      "diagnoses.0.enneking_msts_grade \t 0.0\n",
      "diagnoses.0.enneking_msts_metastasis \t 0.0\n",
      "diagnoses.0.enneking_msts_stage \t 0.0\n",
      "diagnoses.0.enneking_msts_tumor_site \t 0.0\n",
      "diagnoses.0.esophageal_columnar_dysplasia_degree \t 0.0\n",
      "diagnoses.0.esophageal_columnar_metaplasia_present \t 0.0\n",
      "diagnoses.0.figo_stage \t 0.014701042238069117\n",
      "diagnoses.0.figo_staging_edition_year \t 0.0\n",
      "diagnoses.0.first_symptom_prior_to_diagnosis \t 0.0\n",
      "diagnoses.0.gastric_esophageal_junction_involvement \t 0.0\n",
      "diagnoses.0.gleason_grade_group \t 0.0\n",
      "diagnoses.0.gleason_grade_tertiary \t 0.0\n",
      "diagnoses.0.gleason_patterns_percent \t 0.0\n",
      "diagnoses.0.goblet_cells_columnar_mucosa_present \t 0.0\n",
      "diagnoses.0.greatest_tumor_dimension \t 0.0\n",
      "diagnoses.0.gross_tumor_weight \t 0.0\n",
      "diagnoses.0.icd_10_code \t 0.12057048820625343\n",
      "diagnoses.0.igcccg_stage \t 0.0\n",
      "diagnoses.0.inpc_grade \t 0.0\n",
      "diagnoses.0.inpc_histologic_group \t 0.0\n",
      "diagnoses.0.inrg_stage \t 0.0\n",
      "diagnoses.0.inss_stage \t 0.0\n",
      "diagnoses.0.international_prognostic_index \t 0.0\n",
      "diagnoses.0.irs_group \t 0.0\n",
      "diagnoses.0.irs_stage \t 0.0\n",
      "diagnoses.0.ishak_fibrosis_score \t 0.0\n",
      "diagnoses.0.iss_stage \t 0.0\n",
      "diagnoses.0.largest_extrapelvic_peritoneal_focus \t 0.0\n",
      "diagnoses.0.last_known_disease_status \t 0.975973669775096\n",
      "diagnoses.0.laterality \t 0.0\n",
      "diagnoses.0.lymph_node_involved_site \t 0.0\n",
      "diagnoses.0.lymph_nodes_positive \t 0.0\n",
      "diagnoses.0.lymph_nodes_tested \t 0.0\n",
      "diagnoses.0.lymphatic_invasion_present \t 0.0\n",
      "diagnoses.0.margin_distance \t 0.0\n",
      "diagnoses.0.margins_involved_site \t 0.0\n",
      "diagnoses.0.masaoka_stage \t 0.0\n",
      "diagnoses.0.medulloblastoma_molecular_classification \t 0.0\n",
      "diagnoses.0.metastasis_at_diagnosis \t 0.00010970927043335162\n",
      "diagnoses.0.metastasis_at_diagnosis_site \t 0.0\n",
      "diagnoses.0.method_of_diagnosis \t 0.0\n",
      "diagnoses.0.micropapillary_features \t 0.0\n",
      "diagnoses.0.mitosis_karyorrhexis_index \t 0.0\n",
      "diagnoses.0.mitotic_count \t 0.0\n",
      "diagnoses.0.morphology \t 0.9998902907295667\n",
      "diagnoses.0.non_nodal_regional_disease \t 0.0\n",
      "diagnoses.0.non_nodal_tumor_deposits \t 0.0\n",
      "diagnoses.0.ovarian_specimen_status \t 0.0\n",
      "diagnoses.0.ovarian_surface_involvement \t 0.0\n",
      "diagnoses.0.papillary_renal_cell_type \t 0.0\n",
      "diagnoses.0.percent_tumor_invasion \t 0.0\n",
      "diagnoses.0.perineural_invasion_present \t 0.0\n",
      "diagnoses.0.peripancreatic_lymph_nodes_positive \t 0.0\n",
      "diagnoses.0.peripancreatic_lymph_nodes_tested \t 0.0\n",
      "diagnoses.0.peritoneal_fluid_cytological_status \t 0.0\n",
      "diagnoses.0.pregnant_at_diagnosis \t 0.0\n",
      "diagnoses.0.primary_diagnosis \t 0.9998902907295667\n",
      "diagnoses.0.primary_gleason_grade \t 0.0\n",
      "diagnoses.0.prior_malignancy \t 0.13395501919912234\n",
      "diagnoses.0.prior_treatment \t 0.12057048820625343\n",
      "diagnoses.0.progression_or_recurrence \t 0.975973669775096\n",
      "diagnoses.0.residual_disease \t 0.0\n",
      "diagnoses.0.secondary_gleason_grade \t 0.0\n",
      "diagnoses.0.site_of_resection_or_biopsy \t 0.9998902907295667\n",
      "diagnoses.0.state \t 0.9998902907295667\n",
      "diagnoses.0.submitter_id \t 0.9998902907295667\n",
      "diagnoses.0.supratentorial_localization \t 0.0\n",
      "diagnoses.0.synchronous_malignancy \t 0.12046077893582008\n",
      "diagnoses.0.tissue_or_organ_of_origin \t 0.9998902907295667\n",
      "diagnoses.0.transglottic_extension \t 0.0\n",
      "diagnoses.0.treatments.0.created_datetime \t 0.06023038946791004\n",
      "diagnoses.0.treatments.0.days_to_treatment_end \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.0.state \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.submitter_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.updated_datetime \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.1.created_datetime \t 0.060449808008776744\n",
      "diagnoses.0.treatments.1.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.1.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.1.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.1.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.1.state \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.submitter_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.updated_datetime \t 0.12057048820625343\n",
      "diagnoses.0.treatments.10.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.10.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.10.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.10.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.10.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.10.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.10.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.2.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.2.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.2.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.2.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.2.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.2.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.4.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.4.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.4.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.4.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.4.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.4.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.5.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.5.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.5.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.5.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.5.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.5.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.6.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.6.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.6.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.6.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.6.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.6.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.7.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.7.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.7.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.7.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.7.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.treatment_intent_type \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.8.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.8.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.8.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.8.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.8.regimen_or_line_of_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.9.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.9.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.9.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.9.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.9.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.9.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.tumor_confined_to_organ_of_origin \t 0.0\n",
      "diagnoses.0.tumor_depth \t 0.0\n",
      "diagnoses.0.tumor_focality \t 0.0\n",
      "diagnoses.0.tumor_grade \t 0.9760833790455293\n",
      "diagnoses.0.tumor_largest_dimension_diameter \t 0.0\n",
      "diagnoses.0.tumor_regression_grade \t 0.0\n",
      "diagnoses.0.tumor_stage \t 0.975973669775096\n",
      "diagnoses.0.updated_datetime \t 0.9998902907295667\n",
      "diagnoses.0.vascular_invasion_present \t 0.0\n",
      "diagnoses.0.vascular_invasion_type \t 0.0\n",
      "diagnoses.0.weiss_assessment_score \t 0.0\n",
      "diagnoses.0.wilms_tumor_histologic_subtype \t 0.0\n",
      "diagnoses.0.year_of_diagnosis \t 0.12024136039495337\n",
      "exposures.0.age_at_onset \t 0.0\n",
      "exposures.0.alcohol_days_per_week \t 0.0\n",
      "exposures.0.alcohol_drinks_per_day \t 0.0\n",
      "exposures.0.alcohol_history \t 0.12046077893582008\n",
      "exposures.0.alcohol_intensity \t 0.0\n",
      "exposures.0.asbestos_exposure \t 0.0\n",
      "exposures.0.bmi \t 0.00010970927043335162\n",
      "exposures.0.cigarettes_per_day \t 0.0\n",
      "exposures.0.coal_dust_exposure \t 0.0\n",
      "exposures.0.created_datetime \t 0.00010970927043335162\n",
      "exposures.0.environmental_tobacco_smoke_exposure \t 0.0\n",
      "exposures.0.exposure_duration \t 0.0\n",
      "exposures.0.exposure_id \t 0.12057048820625343\n",
      "exposures.0.exposure_type \t 0.0\n",
      "exposures.0.height \t 0.00010970927043335162\n",
      "exposures.0.marijuana_use_per_week \t 0.0\n",
      "exposures.0.pack_years_smoked \t 0.0\n",
      "exposures.0.radon_exposure \t 0.0\n",
      "exposures.0.respirable_crystalline_silica_exposure \t 0.0\n",
      "exposures.0.secondhand_smoke_as_child \t 0.0\n",
      "exposures.0.smoking_frequency \t 0.0\n",
      "exposures.0.state \t 0.12057048820625343\n",
      "exposures.0.submitter_id \t 0.12057048820625343\n",
      "exposures.0.time_between_waking_and_first_smoke \t 0.0\n",
      "exposures.0.tobacco_smoking_onset_year \t 0.0\n",
      "exposures.0.tobacco_smoking_quit_year \t 0.0\n",
      "exposures.0.tobacco_smoking_status \t 0.00010970927043335162\n",
      "exposures.0.tobacco_use_per_day \t 0.0\n",
      "exposures.0.type_of_smoke_exposure \t 0.0\n",
      "exposures.0.type_of_tobacco_used \t 0.0\n",
      "exposures.0.updated_datetime \t 0.12057048820625343\n",
      "exposures.0.weight \t 0.00010970927043335162\n",
      "exposures.0.years_smoked \t 0.0\n",
      "family_histories.0.created_datetime \t 0.00010970927043335162\n",
      "family_histories.0.family_history_id \t 0.00010970927043335162\n",
      "family_histories.0.relationship_age_at_diagnosis \t 0.0\n",
      "family_histories.0.relationship_gender \t 0.0\n",
      "family_histories.0.relationship_primary_diagnosis \t 0.00010970927043335162\n",
      "family_histories.0.relationship_type \t 0.0\n",
      "family_histories.0.relative_with_cancer_history \t 0.00010970927043335162\n",
      "family_histories.0.relatives_with_cancer_history_count \t 0.0\n",
      "family_histories.0.state \t 0.00010970927043335162\n",
      "family_histories.0.submitter_id \t 0.00010970927043335162\n",
      "family_histories.0.updated_datetime \t 0.00010970927043335162\n",
      "id \t 1.0\n",
      "samples.0.biospecimen_anatomic_site \t 0.023806911684037303\n",
      "samples.0.biospecimen_laterality \t 0.0\n",
      "samples.0.catalog_reference \t 0.0\n",
      "samples.0.composition \t 0.007021393307734504\n",
      "samples.0.created_datetime \t 0.9190345584201866\n",
      "samples.0.current_weight \t 0.0\n",
      "samples.0.days_to_collection \t 0.08107515085024684\n",
      "samples.0.days_to_sample_procurement \t 0.03982446516730664\n",
      "samples.0.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.0.distance_normal_to_tumor \t 0.0\n",
      "samples.0.distributor_reference \t 0.0\n",
      "samples.0.freezing_method \t 0.0\n",
      "samples.0.growth_rate \t 0.0\n",
      "samples.0.initial_weight \t 0.04256719692814043\n",
      "samples.0.intermediate_dimension \t 0.0\n",
      "samples.0.is_ffpe \t 0.14448710916072408\n",
      "samples.0.longest_dimension \t 0.0\n",
      "samples.0.method_of_sample_procurement \t 0.0\n",
      "samples.0.oct_embedded \t 0.12068019747668678\n",
      "samples.0.passage_count \t 0.0\n",
      "samples.0.pathology_report_uuid \t 0.03817882611080636\n",
      "samples.0.preservation_method \t 0.03982446516730664\n",
      "samples.0.sample_id \t 1.0\n",
      "samples.0.sample_type \t 1.0\n",
      "samples.0.sample_type_id \t 0.1353812397147559\n",
      "samples.0.shortest_dimension \t 0.0\n",
      "samples.0.state \t 1.0\n",
      "samples.0.submitter_id \t 1.0\n",
      "samples.0.time_between_clamping_and_freezing \t 0.0\n",
      "samples.0.time_between_excision_and_freezing \t 0.0\n",
      "samples.0.tissue_collection_type \t 0.0\n",
      "samples.0.tissue_type \t 1.0\n",
      "samples.0.tumor_code \t 0.0\n",
      "samples.0.tumor_code_id \t 0.0\n",
      "samples.0.tumor_descriptor \t 0.28337904552934723\n",
      "samples.0.updated_datetime \t 1.0\n",
      "samples.1.biospecimen_anatomic_site \t 0.022270981897970377\n",
      "samples.1.biospecimen_laterality \t 0.0\n",
      "samples.1.catalog_reference \t 0.0\n",
      "samples.1.composition \t 0.007679648930334613\n",
      "samples.1.created_datetime \t 0.07383433900164564\n",
      "samples.1.current_weight \t 0.0\n",
      "samples.1.days_to_collection \t 0.08381788261108064\n",
      "samples.1.days_to_sample_procurement \t 0.03664289632473944\n",
      "samples.1.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.1.distance_normal_to_tumor \t 0.0\n",
      "samples.1.distributor_reference \t 0.0\n",
      "samples.1.freezing_method \t 0.0\n",
      "samples.1.growth_rate \t 0.0\n",
      "samples.1.initial_weight \t 0.05013713658804169\n",
      "samples.1.intermediate_dimension \t 0.0\n",
      "samples.1.is_ffpe \t 0.1428414701042238\n",
      "samples.1.longest_dimension \t 0.0\n",
      "samples.1.method_of_sample_procurement \t 0.0\n",
      "samples.1.oct_embedded \t 0.12057048820625343\n",
      "samples.1.passage_count \t 0.0\n",
      "samples.1.pathology_report_uuid \t 0.04509051014810751\n",
      "samples.1.preservation_method \t 0.03664289632473944\n",
      "samples.1.sample_id \t 0.15765222161272627\n",
      "samples.1.sample_type \t 0.15765222161272627\n",
      "samples.1.sample_type_id \t 0.1353812397147559\n",
      "samples.1.shortest_dimension \t 0.0\n",
      "samples.1.state \t 0.15765222161272627\n",
      "samples.1.submitter_id \t 0.15765222161272627\n",
      "samples.1.time_between_clamping_and_freezing \t 0.0\n",
      "samples.1.time_between_excision_and_freezing \t 0.0\n",
      "samples.1.tissue_collection_type \t 0.0\n",
      "samples.1.tissue_type \t 0.15765222161272627\n",
      "samples.1.tumor_code \t 0.0\n",
      "samples.1.tumor_code_id \t 0.0\n",
      "samples.1.tumor_descriptor \t 0.0\n",
      "samples.1.updated_datetime \t 0.15765222161272627\n",
      "samples.2.biospecimen_anatomic_site \t 0.01634668129456939\n",
      "samples.2.biospecimen_laterality \t 0.0\n",
      "samples.2.catalog_reference \t 0.0\n",
      "samples.2.composition \t 0.004388370817334065\n",
      "samples.2.created_datetime \t 0.05573230938014262\n",
      "samples.2.current_weight \t 0.0\n",
      "samples.2.days_to_collection \t 0.07931980252331322\n",
      "samples.2.days_to_sample_procurement \t 0.036752605595172794\n",
      "samples.2.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.2.distance_normal_to_tumor \t 0.0\n",
      "samples.2.distributor_reference \t 0.0\n",
      "samples.2.freezing_method \t 0.0\n",
      "samples.2.growth_rate \t 0.0\n",
      "samples.2.initial_weight \t 0.04201865057597367\n",
      "samples.2.intermediate_dimension \t 0.0\n",
      "samples.2.is_ffpe \t 0.13274821722435545\n",
      "samples.2.longest_dimension \t 0.0\n",
      "samples.2.method_of_sample_procurement \t 0.0\n",
      "samples.2.oct_embedded \t 0.11640153592978607\n",
      "samples.2.passage_count \t 0.0\n",
      "samples.2.pathology_report_uuid \t 0.03708173340647285\n",
      "samples.2.preservation_method \t 0.036752605595172794\n",
      "samples.2.sample_id \t 0.13527153044432255\n",
      "samples.2.sample_type \t 0.13527153044432255\n",
      "samples.2.sample_type_id \t 0.11892484914975315\n",
      "samples.2.shortest_dimension \t 0.0\n",
      "samples.2.state \t 0.13527153044432255\n",
      "samples.2.submitter_id \t 0.13527153044432255\n",
      "samples.2.time_between_clamping_and_freezing \t 0.0\n",
      "samples.2.time_between_excision_and_freezing \t 0.0\n",
      "samples.2.tissue_collection_type \t 0.0\n",
      "samples.2.tissue_type \t 0.13527153044432255\n",
      "samples.2.tumor_code \t 0.0\n",
      "samples.2.tumor_code_id \t 0.0\n",
      "samples.2.tumor_descriptor \t 0.0\n",
      "samples.2.updated_datetime \t 0.13527153044432255\n",
      "samples.3.biospecimen_anatomic_site \t 0.006253428414701042\n",
      "samples.3.biospecimen_laterality \t 0.0\n",
      "samples.3.catalog_reference \t 0.0\n",
      "samples.3.composition \t 0.0012068019747668679\n",
      "samples.3.created_datetime \t 0.009654415798134943\n",
      "samples.3.current_weight \t 0.0\n",
      "samples.3.days_to_collection \t 0.0072408118486012064\n",
      "samples.3.days_to_sample_procurement \t 0.0034009873834339004\n",
      "samples.3.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.3.distance_normal_to_tumor \t 0.0\n",
      "samples.3.distributor_reference \t 0.0\n",
      "samples.3.freezing_method \t 0.0\n",
      "samples.3.growth_rate \t 0.0\n",
      "samples.3.initial_weight \t 0.004388370817334065\n",
      "samples.3.intermediate_dimension \t 0.0\n",
      "samples.3.is_ffpe \t 0.01689522764673615\n",
      "samples.3.longest_dimension \t 0.0\n",
      "samples.3.method_of_sample_procurement \t 0.0\n",
      "samples.3.oct_embedded \t 0.010641799232035107\n",
      "samples.3.passage_count \t 0.0\n",
      "samples.3.pathology_report_uuid \t 0.002523313219967087\n",
      "samples.3.preservation_method \t 0.0034009873834339004\n",
      "samples.3.sample_id \t 0.01689522764673615\n",
      "samples.3.sample_type \t 0.01689522764673615\n",
      "samples.3.sample_type_id \t 0.010641799232035107\n",
      "samples.3.shortest_dimension \t 0.0\n",
      "samples.3.state \t 0.01689522764673615\n",
      "samples.3.submitter_id \t 0.01689522764673615\n",
      "samples.3.time_between_clamping_and_freezing \t 0.0\n",
      "samples.3.time_between_excision_and_freezing \t 0.0\n",
      "samples.3.tissue_collection_type \t 0.0\n",
      "samples.3.tissue_type \t 0.01689522764673615\n",
      "samples.3.tumor_code \t 0.0\n",
      "samples.3.tumor_code_id \t 0.0\n",
      "samples.3.tumor_descriptor \t 0.0\n",
      "samples.3.updated_datetime \t 0.01689522764673615\n",
      "samples.4.biospecimen_anatomic_site \t 0.004168952276467361\n",
      "samples.4.biospecimen_laterality \t 0.0\n",
      "samples.4.catalog_reference \t 0.0\n",
      "samples.4.composition \t 0.0007679648930334613\n",
      "samples.4.created_datetime \t 0.004168952276467361\n",
      "samples.4.current_weight \t 0.0\n",
      "samples.4.days_to_collection \t 0.0004388370817334065\n",
      "samples.4.days_to_sample_procurement \t 0.0\n",
      "samples.4.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.4.distance_normal_to_tumor \t 0.0\n",
      "samples.4.distributor_reference \t 0.0\n",
      "samples.4.freezing_method \t 0.0\n",
      "samples.4.growth_rate \t 0.0\n",
      "samples.4.initial_weight \t 0.00021941854086670324\n",
      "samples.4.intermediate_dimension \t 0.0\n",
      "samples.4.is_ffpe \t 0.004607789358200768\n",
      "samples.4.longest_dimension \t 0.0\n",
      "samples.4.method_of_sample_procurement \t 0.0\n",
      "samples.4.oct_embedded \t 0.0004388370817334065\n",
      "samples.4.passage_count \t 0.0\n",
      "samples.4.pathology_report_uuid \t 0.00021941854086670324\n",
      "samples.4.preservation_method \t 0.0\n",
      "samples.4.sample_id \t 0.004607789358200768\n",
      "samples.4.sample_type \t 0.004607789358200768\n",
      "samples.4.sample_type_id \t 0.0004388370817334065\n",
      "samples.4.shortest_dimension \t 0.0\n",
      "samples.4.state \t 0.004607789358200768\n",
      "samples.4.submitter_id \t 0.004607789358200768\n",
      "samples.4.time_between_clamping_and_freezing \t 0.0\n",
      "samples.4.time_between_excision_and_freezing \t 0.0\n",
      "samples.4.tissue_collection_type \t 0.0\n",
      "samples.4.tissue_type \t 0.004607789358200768\n",
      "samples.4.tumor_code \t 0.0\n",
      "samples.4.tumor_code_id \t 0.0\n",
      "samples.4.tumor_descriptor \t 0.0\n",
      "samples.4.updated_datetime \t 0.004607789358200768\n",
      "samples.5.biospecimen_anatomic_site \t 0.0015359297860669226\n",
      "samples.5.biospecimen_laterality \t 0.0\n",
      "samples.5.catalog_reference \t 0.0\n",
      "samples.5.composition \t 0.0006582556226001097\n",
      "samples.5.created_datetime \t 0.0015359297860669226\n",
      "samples.5.current_weight \t 0.0\n",
      "samples.5.days_to_collection \t 0.0\n",
      "samples.5.days_to_sample_procurement \t 0.0\n",
      "samples.5.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.5.distance_normal_to_tumor \t 0.0\n",
      "samples.5.distributor_reference \t 0.0\n",
      "samples.5.freezing_method \t 0.0\n",
      "samples.5.growth_rate \t 0.0\n",
      "samples.5.initial_weight \t 0.0\n",
      "samples.5.intermediate_dimension \t 0.0\n",
      "samples.5.is_ffpe \t 0.0015359297860669226\n",
      "samples.5.longest_dimension \t 0.0\n",
      "samples.5.method_of_sample_procurement \t 0.0\n",
      "samples.5.oct_embedded \t 0.0\n",
      "samples.5.passage_count \t 0.0\n",
      "samples.5.pathology_report_uuid \t 0.0\n",
      "samples.5.preservation_method \t 0.0\n",
      "samples.5.sample_id \t 0.0015359297860669226\n",
      "samples.5.sample_type \t 0.0015359297860669226\n",
      "samples.5.sample_type_id \t 0.0\n",
      "samples.5.shortest_dimension \t 0.0\n",
      "samples.5.state \t 0.0015359297860669226\n",
      "samples.5.submitter_id \t 0.0015359297860669226\n",
      "samples.5.time_between_clamping_and_freezing \t 0.0\n",
      "samples.5.time_between_excision_and_freezing \t 0.0\n",
      "samples.5.tissue_collection_type \t 0.0\n",
      "samples.5.tissue_type \t 0.0015359297860669226\n",
      "samples.5.tumor_code \t 0.0\n",
      "samples.5.tumor_code_id \t 0.0\n",
      "samples.5.tumor_descriptor \t 0.0\n",
      "samples.5.updated_datetime \t 0.0015359297860669226\n",
      "samples.6.biospecimen_anatomic_site \t 0.0009873834339001646\n",
      "samples.6.biospecimen_laterality \t 0.0\n",
      "samples.6.catalog_reference \t 0.0\n",
      "samples.6.composition \t 0.0\n",
      "samples.6.created_datetime \t 0.0009873834339001646\n",
      "samples.6.current_weight \t 0.0\n",
      "samples.6.days_to_collection \t 0.0\n",
      "samples.6.days_to_sample_procurement \t 0.0\n",
      "samples.6.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.6.distance_normal_to_tumor \t 0.0\n",
      "samples.6.distributor_reference \t 0.0\n",
      "samples.6.freezing_method \t 0.0\n",
      "samples.6.growth_rate \t 0.0\n",
      "samples.6.initial_weight \t 0.0\n",
      "samples.6.intermediate_dimension \t 0.0\n",
      "samples.6.is_ffpe \t 0.0009873834339001646\n",
      "samples.6.longest_dimension \t 0.0\n",
      "samples.6.method_of_sample_procurement \t 0.0\n",
      "samples.6.oct_embedded \t 0.0\n",
      "samples.6.passage_count \t 0.0\n",
      "samples.6.pathology_report_uuid \t 0.0\n",
      "samples.6.preservation_method \t 0.0\n",
      "samples.6.sample_id \t 0.0009873834339001646\n",
      "samples.6.sample_type \t 0.0009873834339001646\n",
      "samples.6.sample_type_id \t 0.0\n",
      "samples.6.shortest_dimension \t 0.0\n",
      "samples.6.state \t 0.0009873834339001646\n",
      "samples.6.submitter_id \t 0.0009873834339001646\n",
      "samples.6.time_between_clamping_and_freezing \t 0.0\n",
      "samples.6.time_between_excision_and_freezing \t 0.0\n",
      "samples.6.tissue_collection_type \t 0.0\n",
      "samples.6.tissue_type \t 0.0009873834339001646\n",
      "samples.6.tumor_code \t 0.0\n",
      "samples.6.tumor_code_id \t 0.0\n",
      "samples.6.tumor_descriptor \t 0.0\n",
      "samples.6.updated_datetime \t 0.0009873834339001646\n",
      "samples.7.biospecimen_anatomic_site \t 0.00032912781130005483\n",
      "samples.7.biospecimen_laterality \t 0.0\n",
      "samples.7.catalog_reference \t 0.0\n",
      "samples.7.composition \t 0.00010970927043335162\n",
      "samples.7.created_datetime \t 0.00032912781130005483\n",
      "samples.7.current_weight \t 0.0\n",
      "samples.7.days_to_collection \t 0.0\n",
      "samples.7.days_to_sample_procurement \t 0.0\n",
      "samples.7.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.7.distance_normal_to_tumor \t 0.0\n",
      "samples.7.distributor_reference \t 0.0\n",
      "samples.7.freezing_method \t 0.0\n",
      "samples.7.growth_rate \t 0.0\n",
      "samples.7.initial_weight \t 0.0\n",
      "samples.7.intermediate_dimension \t 0.0\n",
      "samples.7.is_ffpe \t 0.00032912781130005483\n",
      "samples.7.longest_dimension \t 0.0\n",
      "samples.7.method_of_sample_procurement \t 0.0\n",
      "samples.7.oct_embedded \t 0.0\n",
      "samples.7.passage_count \t 0.0\n",
      "samples.7.pathology_report_uuid \t 0.0\n",
      "samples.7.preservation_method \t 0.0\n",
      "samples.7.sample_id \t 0.00032912781130005483\n",
      "samples.7.sample_type \t 0.00032912781130005483\n",
      "samples.7.sample_type_id \t 0.0\n",
      "samples.7.shortest_dimension \t 0.0\n",
      "samples.7.state \t 0.00032912781130005483\n",
      "samples.7.submitter_id \t 0.00032912781130005483\n",
      "samples.7.time_between_clamping_and_freezing \t 0.0\n",
      "samples.7.time_between_excision_and_freezing \t 0.0\n",
      "samples.7.tissue_collection_type \t 0.0\n",
      "samples.7.tissue_type \t 0.00032912781130005483\n",
      "samples.7.tumor_code \t 0.0\n",
      "samples.7.tumor_code_id \t 0.0\n",
      "samples.7.tumor_descriptor \t 0.0\n",
      "samples.7.updated_datetime \t 0.00032912781130005483\n",
      "samples.8.biospecimen_anatomic_site \t 0.00010970927043335162\n",
      "samples.8.biospecimen_laterality \t 0.0\n",
      "samples.8.catalog_reference \t 0.0\n",
      "samples.8.composition \t 0.00010970927043335162\n",
      "samples.8.created_datetime \t 0.00010970927043335162\n",
      "samples.8.current_weight \t 0.0\n",
      "samples.8.days_to_collection \t 0.0\n",
      "samples.8.days_to_sample_procurement \t 0.0\n",
      "samples.8.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.8.distance_normal_to_tumor \t 0.0\n",
      "samples.8.distributor_reference \t 0.0\n",
      "samples.8.freezing_method \t 0.0\n",
      "samples.8.growth_rate \t 0.0\n",
      "samples.8.initial_weight \t 0.0\n",
      "samples.8.intermediate_dimension \t 0.0\n",
      "samples.8.is_ffpe \t 0.00010970927043335162\n",
      "samples.8.longest_dimension \t 0.0\n",
      "samples.8.method_of_sample_procurement \t 0.0\n",
      "samples.8.oct_embedded \t 0.0\n",
      "samples.8.passage_count \t 0.0\n",
      "samples.8.pathology_report_uuid \t 0.0\n",
      "samples.8.preservation_method \t 0.0\n",
      "samples.8.sample_id \t 0.00010970927043335162\n",
      "samples.8.sample_type \t 0.00010970927043335162\n",
      "samples.8.sample_type_id \t 0.0\n",
      "samples.8.shortest_dimension \t 0.0\n",
      "samples.8.state \t 0.00010970927043335162\n",
      "samples.8.submitter_id \t 0.00010970927043335162\n",
      "samples.8.time_between_clamping_and_freezing \t 0.0\n",
      "samples.8.time_between_excision_and_freezing \t 0.0\n",
      "samples.8.tissue_collection_type \t 0.0\n",
      "samples.8.tissue_type \t 0.00010970927043335162\n",
      "samples.8.tumor_code \t 0.0\n",
      "samples.8.tumor_code_id \t 0.0\n",
      "samples.8.tumor_descriptor \t 0.0\n",
      "samples.8.updated_datetime \t 0.00010970927043335162\n",
      "submitter_id \t 1.0\n",
      "tissue_source_site.bcr_id \t 0.12057048820625343\n",
      "tissue_source_site.code \t 0.12057048820625343\n",
      "tissue_source_site.name \t 0.12057048820625343\n",
      "tissue_source_site.project \t 0.12057048820625343\n",
      "tissue_source_site.tissue_source_site_id \t 0.12057048820625343\n"
     ]
    }
   ],
   "source": [
    "serie = df_case.notnull().mean()\n",
    "\n",
    "for name,val in zip(serie.index,serie): \n",
    "    print(name,'\\t',val)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can see many of the columns contain almost exclusively null values. \n",
    "Thus, to have a better quality information and not empty, we chose arbitrarly to delete every column for which the filling was under 10%.\n",
    "Moreover, many columns containing ids/code are useless to our project. Indeed, knowing where the tissue has been taken from a patient will not help us to predict the treatment. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case_id \t 1.0\n",
      "demographic.age_at_index \t 0.6760285244103127\n",
      "demographic.days_to_birth \t 0.6744925946242457\n",
      "demographic.ethnicity \t 0.9998902907295667\n",
      "demographic.gender \t 0.9998902907295667\n",
      "demographic.race \t 0.9998902907295667\n",
      "demographic.vital_status \t 0.9998902907295667\n",
      "demographic.year_of_birth \t 0.12024136039495337\n",
      "diagnoses.0.age_at_diagnosis \t 0.4365331870543061\n",
      "diagnoses.0.ajcc_pathologic_m \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_n \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_stage \t 0.13395501919912234\n",
      "diagnoses.0.ajcc_pathologic_t \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_staging_system_edition \t 0.1049917718047175\n",
      "diagnoses.0.classification_of_tumor \t 0.4185408667032364\n",
      "diagnoses.0.days_to_diagnosis \t 0.12057048820625343\n",
      "diagnoses.0.days_to_last_follow_up \t 0.10905101481075151\n",
      "diagnoses.0.last_known_disease_status \t 0.975973669775096\n",
      "diagnoses.0.morphology \t 0.9998902907295667\n",
      "diagnoses.0.primary_diagnosis \t 0.9998902907295667\n",
      "diagnoses.0.prior_malignancy \t 0.13395501919912234\n",
      "diagnoses.0.prior_treatment \t 0.12057048820625343\n",
      "diagnoses.0.progression_or_recurrence \t 0.975973669775096\n",
      "diagnoses.0.site_of_resection_or_biopsy \t 0.9998902907295667\n",
      "diagnoses.0.synchronous_malignancy \t 0.12046077893582008\n",
      "diagnoses.0.tissue_or_organ_of_origin \t 0.9998902907295667\n",
      "diagnoses.0.treatments.0.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.tumor_grade \t 0.9760833790455293\n",
      "diagnoses.0.tumor_stage \t 0.975973669775096\n",
      "diagnoses.0.year_of_diagnosis \t 0.12024136039495337\n",
      "exposures.0.alcohol_history \t 0.12046077893582008\n",
      "id \t 1.0\n",
      "samples.0.is_ffpe \t 0.14448710916072408\n",
      "samples.0.oct_embedded \t 0.12068019747668678\n",
      "samples.0.sample_type \t 1.0\n",
      "samples.0.tissue_type \t 1.0\n",
      "samples.0.tumor_descriptor \t 0.28337904552934723\n",
      "samples.1.is_ffpe \t 0.1428414701042238\n",
      "samples.1.oct_embedded \t 0.12057048820625343\n",
      "samples.1.sample_type \t 0.15765222161272627\n",
      "samples.1.tissue_type \t 0.15765222161272627\n",
      "samples.2.is_ffpe \t 0.13274821722435545\n",
      "samples.2.oct_embedded \t 0.11640153592978607\n",
      "samples.2.sample_type \t 0.13527153044432255\n",
      "samples.2.tissue_type \t 0.13527153044432255\n"
     ]
    }
   ],
   "source": [
    "df_case = df_case.drop(df_case.columns[df_case.notnull().mean()<0.1], axis=1)\n",
    "columns_to_drop = ['tissue_source_site.code', 'tissue_source_site.project','tissue_source_site.name', 'diagnoses.0.treatments.1.state', 'diagnoses.0.state', 'diagnoses.0.treatments.0.state', 'exposures.0.state', 'samples.0.state', 'samples.1.state', 'samples.2.state', 'demographic.demographic_id', 'demographic.state', 'demographic.submitter_id', 'demographic.updated_datetime', 'diagnoses.0.diagnosis_id', 'diagnoses.0.icd_10_code', 'diagnoses.0.submitter_id', 'diagnoses.0.treatments.1.updated_datetime', 'diagnoses.0.treatments.1.submitter_id', 'diagnoses.0.treatments.0.updated_datetime', 'diagnoses.0.updated_datetime', 'exposures.0.exposure_id', 'exposures.0.submitter_id', 'exposures.0.updated_datetime', 'samples.0.created_datetime', 'samples.0.sample_id', 'samples.0.sample_type_id', 'samples.0.submitter_id', 'samples.0.updated_datetime', 'samples.1.sample_id', 'samples.1.sample_type_id', 'samples.1.submitter_id', 'samples.2.sample_id', 'samples.2.sample_type_id', 'samples.2.submitter_id', 'tissue_source_site.bcr_id', 'submitter_id', 'diagnoses.0.treatments.0.submitter_id', 'diagnoses.0.treatments.1.treatment_id', 'samples.2.updated_datetime', 'samples.1.updated_datetime', 'diagnoses.0.treatments.0.treatment_id', 'demographic.created_datetime', 'diagnoses.0.created_datetime', 'tissue_source_site.tissue_source_site_id']\n",
    "df_case = df_case.drop(columns_to_drop, axis=1)\n",
    "serie = df_case.notnull().mean()\n",
    "\n",
    "for name,val in zip(serie.index,serie): \n",
    "    print(name,'\\t',val)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: cnv\n",
    "Information on the nature of the gene mutation. Most of the value are strings.\n",
    "\n",
    "12 columns and 1000 rows.\n",
    "\n",
    "\n",
    "\n",
    "Once againe, we observe some useless columns such as cnv_id or id. Thus, we will drop them to lighten the dataframe and make it more readable.\n",
    "\n",
    "We end up having 10 columns and 1000 rows. \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "case.case_id                        1.0\n",
       "cnv.chromosome                      1.0\n",
       "cnv.cnv_change                      1.0\n",
       "cnv.consequence.0.consequence_id    1.0\n",
       "cnv.consequence.0.gene.symbol       1.0\n",
       "cnv.end_position                    1.0\n",
       "cnv.gene_level_cn                   1.0\n",
       "cnv.ncbi_build                      1.0\n",
       "cnv.start_position                  1.0\n",
       "cnv.variant_status                  1.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns_to_drop = ['cnv.cnv_id', 'id']\n",
    "df_cnv.drop(columns_to_drop, axis=1, inplace=True)\n",
    "df_cnv.notnull().mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>cnv.chromosome</th>\n",
       "      <th>cnv.cnv_change</th>\n",
       "      <th>cnv.consequence.0.consequence_id</th>\n",
       "      <th>cnv.consequence.0.gene.symbol</th>\n",
       "      <th>cnv.end_position</th>\n",
       "      <th>cnv.gene_level_cn</th>\n",
       "      <th>cnv.ncbi_build</th>\n",
       "      <th>cnv.start_position</th>\n",
       "      <th>cnv.variant_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c5891c7a-fd24-4891-8f51-373967e4a6dd</td>\n",
       "      <td>17</td>\n",
       "      <td>Loss</td>\n",
       "      <td>99c68f83-9514-5628-9623-0ba6095dd90f</td>\n",
       "      <td>BCAS3</td>\n",
       "      <td>61392838</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>60677453</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>d5f2b85a-94a9-4168-bbe8-149ec71342b0</td>\n",
       "      <td>17</td>\n",
       "      <td>Gain</td>\n",
       "      <td>a7e6f054-f5a1-5e7f-a0c0-8daa7373a434</td>\n",
       "      <td>ABCA8</td>\n",
       "      <td>68955392</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>68867292</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>eafa45a4-2fd1-4fa3-860c-9d52ed382b7d</td>\n",
       "      <td>11</td>\n",
       "      <td>Gain</td>\n",
       "      <td>924c27ca-11cf-5b13-95b9-cf518abb142d</td>\n",
       "      <td>ORAOV1</td>\n",
       "      <td>69675416</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>69653076</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>dd0f8e5a-383d-4f03-bb71-0e115d4e442c</td>\n",
       "      <td>11</td>\n",
       "      <td>Gain</td>\n",
       "      <td>00461981-a326-573d-a9a1-ad82872a1aea</td>\n",
       "      <td>SHANK2</td>\n",
       "      <td>71252577</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>70467856</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>233b02f3-c4f0-4a67-9db5-e68d5cdaccb6</td>\n",
       "      <td>8</td>\n",
       "      <td>Gain</td>\n",
       "      <td>c6208422-2c76-553a-b0eb-4472e816381f</td>\n",
       "      <td>MYC</td>\n",
       "      <td>127741434</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>127735434</td>\n",
       "      <td>Tumor only</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           case.case_id  cnv.chromosome cnv.cnv_change  \\\n",
       "0  c5891c7a-fd24-4891-8f51-373967e4a6dd              17           Loss   \n",
       "1  d5f2b85a-94a9-4168-bbe8-149ec71342b0              17           Gain   \n",
       "2  eafa45a4-2fd1-4fa3-860c-9d52ed382b7d              11           Gain   \n",
       "3  dd0f8e5a-383d-4f03-bb71-0e115d4e442c              11           Gain   \n",
       "4  233b02f3-c4f0-4a67-9db5-e68d5cdaccb6               8           Gain   \n",
       "\n",
       "       cnv.consequence.0.consequence_id cnv.consequence.0.gene.symbol  \\\n",
       "0  99c68f83-9514-5628-9623-0ba6095dd90f                         BCAS3   \n",
       "1  a7e6f054-f5a1-5e7f-a0c0-8daa7373a434                         ABCA8   \n",
       "2  924c27ca-11cf-5b13-95b9-cf518abb142d                        ORAOV1   \n",
       "3  00461981-a326-573d-a9a1-ad82872a1aea                        SHANK2   \n",
       "4  c6208422-2c76-553a-b0eb-4472e816381f                           MYC   \n",
       "\n",
       "   cnv.end_position  cnv.gene_level_cn cnv.ncbi_build  cnv.start_position  \\\n",
       "0          61392838               True         GRCh38            60677453   \n",
       "1          68955392               True         GRCh38            68867292   \n",
       "2          69675416               True         GRCh38            69653076   \n",
       "3          71252577               True         GRCh38            70467856   \n",
       "4         127741434               True         GRCh38           127735434   \n",
       "\n",
       "  cnv.variant_status  \n",
       "0         Tumor only  \n",
       "1         Tumor only  \n",
       "2         Tumor only  \n",
       "3         Tumor only  \n",
       "4         Tumor only  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cnv.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: gene\n",
    "\n",
    "Information on top 50 genes to studied. Most of the value are strings.\n",
    "\n",
    "2 columns and 50 rows.\n",
    "\n",
    "The **symbol** column will be the key to merge with the df_cnv dataframe.\n",
    "\n",
    "What is 'score' ?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_score</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>451.0</td>\n",
       "      <td>PIK3CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>404.0</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>322.0</td>\n",
       "      <td>MUC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>320.0</td>\n",
       "      <td>SHANK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>318.0</td>\n",
       "      <td>ANO1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>316.0</td>\n",
       "      <td>ORAOV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>314.0</td>\n",
       "      <td>CCND1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>314.0</td>\n",
       "      <td>DNAJB6P5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>310.0</td>\n",
       "      <td>FGF19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>310.0</td>\n",
       "      <td>PPFIA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   _score    symbol\n",
       "0   451.0    PIK3CA\n",
       "1   404.0      TP53\n",
       "2   322.0      MUC4\n",
       "3   320.0    SHANK2\n",
       "4   318.0      ANO1\n",
       "5   316.0    ORAOV1\n",
       "6   314.0     CCND1\n",
       "7   314.0  DNAJB6P5\n",
       "8   310.0     FGF19\n",
       "9   310.0    PPFIA1"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "_score    1.0\n",
       "symbol    1.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene.dtypes\n",
    "df_gene.notnull().mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: ssm\n",
    "\n",
    "Information on the simple somatic mutation.\n",
    "\n",
    "74 columns and 1000 rows.\n",
    "\n",
    "Once again, they are many columns to be deleted either because they are empty or because they are of no use."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case.case_id\n",
      "id\n",
      "ssm.chromosome\n",
      "ssm.consequence.0.consequence_id\n",
      "ssm.consequence.0.transcript.gene.symbol\n",
      "ssm.consequence.1.consequence_id\n",
      "ssm.consequence.1.transcript.gene.symbol\n",
      "ssm.consequence.10.consequence_id\n",
      "ssm.consequence.10.transcript.gene.symbol\n",
      "ssm.consequence.11.consequence_id\n",
      "ssm.consequence.11.transcript.gene.symbol\n",
      "ssm.consequence.12.consequence_id\n",
      "ssm.consequence.12.transcript.gene.symbol\n",
      "ssm.consequence.13.consequence_id\n",
      "ssm.consequence.13.transcript.gene.symbol\n",
      "ssm.consequence.14.consequence_id\n",
      "ssm.consequence.14.transcript.gene.symbol\n",
      "ssm.consequence.15.consequence_id\n",
      "ssm.consequence.15.transcript.gene.symbol\n",
      "ssm.consequence.16.consequence_id\n",
      "ssm.consequence.16.transcript.gene.symbol\n",
      "ssm.consequence.17.consequence_id\n",
      "ssm.consequence.17.transcript.gene.symbol\n",
      "ssm.consequence.18.consequence_id\n",
      "ssm.consequence.18.transcript.gene.symbol\n",
      "ssm.consequence.19.consequence_id\n",
      "ssm.consequence.19.transcript.gene.symbol\n",
      "ssm.consequence.2.consequence_id\n",
      "ssm.consequence.2.transcript.gene.symbol\n",
      "ssm.consequence.20.consequence_id\n",
      "ssm.consequence.20.transcript.gene.symbol\n",
      "ssm.consequence.21.consequence_id\n",
      "ssm.consequence.21.transcript.gene.symbol\n",
      "ssm.consequence.22.consequence_id\n",
      "ssm.consequence.22.transcript.gene.symbol\n",
      "ssm.consequence.23.consequence_id\n",
      "ssm.consequence.23.transcript.gene.symbol\n",
      "ssm.consequence.24.consequence_id\n",
      "ssm.consequence.24.transcript.gene.symbol\n",
      "ssm.consequence.25.consequence_id\n",
      "ssm.consequence.25.transcript.gene.symbol\n",
      "ssm.consequence.26.consequence_id\n",
      "ssm.consequence.26.transcript.gene.symbol\n",
      "ssm.consequence.3.consequence_id\n",
      "ssm.consequence.3.transcript.gene.symbol\n",
      "ssm.consequence.4.consequence_id\n",
      "ssm.consequence.4.transcript.gene.symbol\n",
      "ssm.consequence.5.consequence_id\n",
      "ssm.consequence.5.transcript.gene.symbol\n",
      "ssm.consequence.6.consequence_id\n",
      "ssm.consequence.6.transcript.gene.symbol\n",
      "ssm.consequence.7.consequence_id\n",
      "ssm.consequence.7.transcript.gene.symbol\n",
      "ssm.consequence.8.consequence_id\n",
      "ssm.consequence.8.transcript.gene.symbol\n",
      "ssm.consequence.9.consequence_id\n",
      "ssm.consequence.9.transcript.gene.symbol\n",
      "ssm.cosmic_id.0\n",
      "ssm.cosmic_id.1\n",
      "ssm.cosmic_id.2\n",
      "ssm.cosmic_id.3\n",
      "ssm.cosmic_id.4\n",
      "ssm.cosmic_id.5\n",
      "ssm.cosmic_id.6\n",
      "ssm.cosmic_id.7\n",
      "ssm.end_position\n",
      "ssm.genomic_dna_change\n",
      "ssm.mutation_subtype\n",
      "ssm.mutation_type\n",
      "ssm.ncbi_build\n",
      "ssm.reference_allele\n",
      "ssm.ssm_id\n",
      "ssm.start_position\n",
      "ssm.tumor_allele\n"
     ]
    }
   ],
   "source": [
    "for col in df_ssm.columns: \n",
    "    print(col)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are deleting the unecessary columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2347, 13)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>ssm.chromosome</th>\n",
       "      <th>ssm.consequence.0.transcript.gene.symbol</th>\n",
       "      <th>ssm.cosmic_id.0</th>\n",
       "      <th>ssm.end_position</th>\n",
       "      <th>ssm.genomic_dna_change</th>\n",
       "      <th>ssm.mutation_subtype</th>\n",
       "      <th>ssm.mutation_type</th>\n",
       "      <th>ssm.ncbi_build</th>\n",
       "      <th>ssm.reference_allele</th>\n",
       "      <th>ssm.ssm_id</th>\n",
       "      <th>ssm.start_position</th>\n",
       "      <th>ssm.tumor_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>d2694b09-7844-4496-865c-5d3d8cd03330</td>\n",
       "      <td>chr3</td>\n",
       "      <td>MUC4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>195780168</td>\n",
       "      <td>chr3:g.195780168A&gt;T</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>A</td>\n",
       "      <td>92dded19-d0ce-5c34-9d99-169342f9bc37</td>\n",
       "      <td>195780168</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17baef7c-d97d-4b98-ab53-503ef856523d</td>\n",
       "      <td>chr3</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>COSM446001</td>\n",
       "      <td>179203764</td>\n",
       "      <td>chr3:g.179203764A&gt;C</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>A</td>\n",
       "      <td>7ea2ecba-9bbc-5d84-8a58-2ed20f13bdf3</td>\n",
       "      <td>179203764</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>d3d545b3-457f-4389-821f-704cb24aff7f</td>\n",
       "      <td>chr17</td>\n",
       "      <td>TP53</td>\n",
       "      <td>COSM1480087</td>\n",
       "      <td>7676042</td>\n",
       "      <td>chr17:g.7676041_7676042insGA</td>\n",
       "      <td>Small insertion</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>'-</td>\n",
       "      <td>1c4dad33-40cc-58d0-9d8c-4d808cb33401</td>\n",
       "      <td>7676041</td>\n",
       "      <td>GA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>55262fcb-1b01-4480-b322-36570430c917</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ABCA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68883820</td>\n",
       "      <td>chr17:g.68883820C&gt;G</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>C</td>\n",
       "      <td>905a9d14-3a2e-5fb8-994e-48a2e5a1fa54</td>\n",
       "      <td>68883820</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>91b09291-5944-4691-a0d8-7d1ccc649b9b</td>\n",
       "      <td>chr3</td>\n",
       "      <td>MUC4</td>\n",
       "      <td>COSM3847118</td>\n",
       "      <td>195782578</td>\n",
       "      <td>chr3:g.195782578G&gt;A</td>\n",
       "      <td>Single base substitution</td>\n",
       "      <td>Simple Somatic Mutation</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>G</td>\n",
       "      <td>eb5acb6d-ebcc-50fc-9dc0-9a7ac5b85fc4</td>\n",
       "      <td>195782578</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           case.case_id ssm.chromosome  \\\n",
       "0  d2694b09-7844-4496-865c-5d3d8cd03330           chr3   \n",
       "1  17baef7c-d97d-4b98-ab53-503ef856523d           chr3   \n",
       "2  d3d545b3-457f-4389-821f-704cb24aff7f          chr17   \n",
       "3  55262fcb-1b01-4480-b322-36570430c917          chr17   \n",
       "4  91b09291-5944-4691-a0d8-7d1ccc649b9b           chr3   \n",
       "\n",
       "  ssm.consequence.0.transcript.gene.symbol ssm.cosmic_id.0  ssm.end_position  \\\n",
       "0                                     MUC4             NaN         195780168   \n",
       "1                                   PIK3CA      COSM446001         179203764   \n",
       "2                                     TP53     COSM1480087           7676042   \n",
       "3                                    ABCA8             NaN          68883820   \n",
       "4                                     MUC4     COSM3847118         195782578   \n",
       "\n",
       "         ssm.genomic_dna_change      ssm.mutation_subtype  \\\n",
       "0           chr3:g.195780168A>T  Single base substitution   \n",
       "1           chr3:g.179203764A>C  Single base substitution   \n",
       "2  chr17:g.7676041_7676042insGA           Small insertion   \n",
       "3           chr17:g.68883820C>G  Single base substitution   \n",
       "4           chr3:g.195782578G>A  Single base substitution   \n",
       "\n",
       "         ssm.mutation_type ssm.ncbi_build ssm.reference_allele  \\\n",
       "0  Simple Somatic Mutation         GRCh38                    A   \n",
       "1  Simple Somatic Mutation         GRCh38                    A   \n",
       "2  Simple Somatic Mutation         GRCh38                   '-   \n",
       "3  Simple Somatic Mutation         GRCh38                    C   \n",
       "4  Simple Somatic Mutation         GRCh38                    G   \n",
       "\n",
       "                             ssm.ssm_id  ssm.start_position ssm.tumor_allele  \n",
       "0  92dded19-d0ce-5c34-9d99-169342f9bc37           195780168                T  \n",
       "1  7ea2ecba-9bbc-5d84-8a58-2ed20f13bdf3           179203764                C  \n",
       "2  1c4dad33-40cc-58d0-9d8c-4d808cb33401             7676041               GA  \n",
       "3  905a9d14-3a2e-5fb8-994e-48a2e5a1fa54            68883820                G  \n",
       "4  eb5acb6d-ebcc-50fc-9dc0-9a7ac5b85fc4           195782578                A  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('ce_id')], axis=1, inplace=True)\n",
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('[0-2][1-9]')], axis=1, inplace=True)\n",
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('[1-9]')], axis=1, inplace=True)\n",
    "columns_to_drop = [\"id\"]\n",
    "df_ssm.drop(\"id\", axis=1,inplace=True)\n",
    "\n",
    "print(df_ssm.shape)\n",
    "df_ssm.head(5)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It leaves us with a 13 columns dataframe which is way more exploitable."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We shall now start to merge the dataframes to get a single one that contains all the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Single base substitution', 'Small insertion', 'Small deletion'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm[\"ssm.mutation_subtype\"].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# It starts from there:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Transcriptome (ARN)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We need 3 data tables to be able to use the transcriptomic data :\n",
    "\n",
    "* HTseq data (in ENSEMBL, transcriptome_case_id format) $\\rightarrow$ main transcriptomic data\n",
    "\n",
    "* ENSEMBL to symbol table\n",
    "\n",
    "* transcriptome_case_id to case_id table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ENSEMBL to Symbol"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get the relation table, click [this link](https://www.genenames.org/cgi-bin/download/custom?col=gd_app_sym&col=md_ensembl_id&status=Approved&status=Entry%20Withdrawn&hgnc_dbtag=on&order_by=gd_app_sym_sort&format=text&submit=submit) from [https://www.genenames.org/cgi-bin/download](https://www.genenames.org/cgi-bin/download). Symbols and ENSEMBL are two conventions in naming genes and CNV, SSM data use symbols while Transcriptomic data use ENSEMBL."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a folder ```transcriptome``` in ```./data```, put the coresponding file in it. Rename it ```ENSEMBL.txt``` then run the following code:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Approved symbol</th>\n",
       "      <th>Ensembl ID(supplied by Ensembl)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1S9T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47479</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47480</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>ENSG00000159840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47481</th>\n",
       "      <td>ZYXP1</td>\n",
       "      <td>ENSG00000274572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47482</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47483</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47484 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Approved symbol Ensembl ID(supplied by Ensembl)\n",
       "0                A1BG                 ENSG00000121410\n",
       "1            A1BG-AS1                 ENSG00000268895\n",
       "2                A1CF                 ENSG00000148584\n",
       "3               A1S9T                             NaN\n",
       "4                 A2M                 ENSG00000175899\n",
       "...               ...                             ...\n",
       "47479          ZYG11B                 ENSG00000162378\n",
       "47480             ZYX                 ENSG00000159840\n",
       "47481           ZYXP1                 ENSG00000274572\n",
       "47482           ZZEF1                 ENSG00000074755\n",
       "47483            ZZZ3                 ENSG00000036549\n",
       "\n",
       "[47484 rows x 2 columns]"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl = pd.read_csv('./data/transcriptome/ENSEMBL.txt', sep=\"\\t\", header=0)\n",
    "ensembl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We only keep the 50 more mutated genes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>symbol</th>\n",
       "      <th>Ensembl ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>ENSG00000121879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>ENSG00000145113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SHANK2</td>\n",
       "      <td>ENSG00000162105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ANO1</td>\n",
       "      <td>ENSG00000131620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CCND1</td>\n",
       "      <td>ENSG00000110092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DNAJB6P5</td>\n",
       "      <td>ENSG00000256441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DNAJB6P5</td>\n",
       "      <td>ENSG00000256441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>FGF19</td>\n",
       "      <td>ENSG00000162344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PPFIA1</td>\n",
       "      <td>ENSG00000131626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>FGF3</td>\n",
       "      <td>ENSG00000186895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>FGF4</td>\n",
       "      <td>ENSG00000075388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>FADD</td>\n",
       "      <td>ENSG00000168040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CTTN</td>\n",
       "      <td>ENSG00000085733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MYEOV</td>\n",
       "      <td>ENSG00000172927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENSG00000155657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TPCN2</td>\n",
       "      <td>ENSG00000162341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>BPTF</td>\n",
       "      <td>ENSG00000171634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>FNTA</td>\n",
       "      <td>ENSG00000168522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MRGPRF</td>\n",
       "      <td>ENSG00000172935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TANC2</td>\n",
       "      <td>ENSG00000170921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>ENSG00000198626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>RNF213</td>\n",
       "      <td>ENSG00000173821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TRPS1</td>\n",
       "      <td>ENSG00000104447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ABCA8</td>\n",
       "      <td>ENSG00000141338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MYC</td>\n",
       "      <td>ENSG00000136997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DNAH17</td>\n",
       "      <td>ENSG00000187775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ARSG</td>\n",
       "      <td>ENSG00000141337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MRGPRD</td>\n",
       "      <td>ENSG00000172938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>IGHMBP2</td>\n",
       "      <td>ENSG00000132740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>KPNA2</td>\n",
       "      <td>ENSG00000182481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SLC16A6</td>\n",
       "      <td>ENSG00000108932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>AMZ2</td>\n",
       "      <td>ENSG00000196704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>ENSG00000134250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>POU5F1B</td>\n",
       "      <td>ENSG00000212993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>ABCA10</td>\n",
       "      <td>ENSG00000154263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>BCAS3</td>\n",
       "      <td>ENSG00000141376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>ABCA6</td>\n",
       "      <td>ENSG00000154262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>HELZ</td>\n",
       "      <td>ENSG00000198265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>ENSG00000164796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PITPNC1</td>\n",
       "      <td>ENSG00000154217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CEP112</td>\n",
       "      <td>ENSG00000154240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>SDK2</td>\n",
       "      <td>ENSG00000069188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>CPT1A</td>\n",
       "      <td>ENSG00000110090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>ENSG00000154258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NOL11</td>\n",
       "      <td>ENSG00000130935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>PRKCA</td>\n",
       "      <td>ENSG00000154229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>AC108925.1</td>\n",
       "      <td>ENSG00000286010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>PRKAR1A</td>\n",
       "      <td>ENSG00000108946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>TG</td>\n",
       "      <td>ENSG00000042832</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        symbol       Ensembl ID\n",
       "0       PIK3CA  ENSG00000121879\n",
       "1         TP53  ENSG00000141510\n",
       "2         MUC4  ENSG00000145113\n",
       "3       SHANK2  ENSG00000162105\n",
       "4         ANO1  ENSG00000131620\n",
       "5        CCND1  ENSG00000110092\n",
       "6     DNAJB6P5  ENSG00000256441\n",
       "7     DNAJB6P5  ENSG00000256441\n",
       "8        FGF19  ENSG00000162344\n",
       "9       PPFIA1  ENSG00000131626\n",
       "10        FGF3  ENSG00000186895\n",
       "11        FGF4  ENSG00000075388\n",
       "12        FADD  ENSG00000168040\n",
       "13        CTTN  ENSG00000085733\n",
       "14       MYEOV  ENSG00000172927\n",
       "15         TTN  ENSG00000155657\n",
       "16       TPCN2  ENSG00000162341\n",
       "17        BPTF  ENSG00000171634\n",
       "18        FNTA  ENSG00000168522\n",
       "19      MRGPRF  ENSG00000172935\n",
       "20       TANC2  ENSG00000170921\n",
       "21        RYR2  ENSG00000198626\n",
       "22      RNF213  ENSG00000173821\n",
       "23       TRPS1  ENSG00000104447\n",
       "24       ABCA8  ENSG00000141338\n",
       "25         MYC  ENSG00000136997\n",
       "26      DNAH17  ENSG00000187775\n",
       "27        ARSG  ENSG00000141337\n",
       "28      MRGPRD  ENSG00000172938\n",
       "29     IGHMBP2  ENSG00000132740\n",
       "30       KPNA2  ENSG00000182481\n",
       "31     SLC16A6  ENSG00000108932\n",
       "32        AMZ2  ENSG00000196704\n",
       "33      NOTCH2  ENSG00000134250\n",
       "34     POU5F1B  ENSG00000212993\n",
       "35      ABCA10  ENSG00000154263\n",
       "36       BCAS3  ENSG00000141376\n",
       "37       ABCA6  ENSG00000154262\n",
       "38        HELZ  ENSG00000198265\n",
       "39       CSMD3  ENSG00000164796\n",
       "40     PITPNC1  ENSG00000154217\n",
       "41      CEP112  ENSG00000154240\n",
       "42        SDK2  ENSG00000069188\n",
       "43       CPT1A  ENSG00000110090\n",
       "44       ABCA9  ENSG00000154258\n",
       "45       NOL11  ENSG00000130935\n",
       "46       PRKCA  ENSG00000154229\n",
       "47  AC108925.1  ENSG00000286010\n",
       "48     PRKAR1A  ENSG00000108946\n",
       "49          TG  ENSG00000042832"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene_exp = (pd.merge(df_gene, ensembl, \n",
    "                       how='inner', \n",
    "                       left_on='symbol', \n",
    "                       right_on='Approved symbol')\n",
    "               .drop([\"Approved symbol\", \"_score\"], axis=1)\n",
    "               .rename(columns={\"Ensembl ID(supplied by Ensembl)\": \"Ensembl ID\"})\n",
    "              )\n",
    "df_gene_exp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We rather use a dictionnary to access rapidly the corresponding Symbol of an ENSEMBL ID:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "match_gene = {}\n",
    "for symb, ens in zip(df_gene_exp[\"symbol\"], df_gene_exp[\"Ensembl ID\"]):\n",
    "    match_gene[ens] = symb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### UUID matching"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The UUID in the transcriptomic data corresponds to different values than the one used for the general medical data. Fortunately, we can find a correspondance in a json file downloadable from [the gdc portal website](https://portal.gdc.cancer.gov/repository?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.analysis.workflow_type%22%2C%22value%22%3A%5B%22HTSeq%20-%20Counts%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.cases.primary_site%22%2C%22value%22%3A%5B%22breast%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_type%22%2C%22value%22%3A%5B%22Gene%20Expression%20Quantification%22%5D%7D%7D%5D%7D) (click on the JSON button under the pie charts).\n",
    "\n",
    "Put the file in ```./data/transcriptome``` and name it ```UUID_match.json```."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"./data/transcriptome/UUID_match.json\", \"r\") as f:\n",
    "    json_content = json.load(f)\n",
    "\n",
    "match_uuid = {}\n",
    "for dico in json_content:\n",
    "    match_uuid[dico['file_name']] = dico['cases'][0]['case_id']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Download transcriptomic data\n",
    "\n",
    "The first step consists in downloading the manifest file by clicking on the download button named **Manifest** in [this page](https://portal.gdc.cancer.gov/repository?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.analysis.workflow_type%22%2C%22value%22%3A%5B%22HTSeq%20-%20Counts%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.cases.primary_site%22%2C%22value%22%3A%5B%22breast%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_type%22%2C%22value%22%3A%5B%22Gene%20Expression%20Quantification%22%5D%7D%7D%5D%7D) of the gdc portal website. The manifest contains all the information about what is wanted to be downloaded from the database (that is to say the transcriptomic data).\n",
    "\n",
    "The second steps is to install [gdc client](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool#:~:text=The%20GDC%20provides%20a%20standard%20client-based%20mechanism%20in,or%20FASTQ%2C%20make%20up%20the%20bulk%20of%20data.) to download the data. \n",
    "\n",
    "**Important**: set \"gdc-client\" to the environnement path variables by pointing to the ```gdc-client.exe``` file.\n",
    "\n",
    "Then open a terminal and go to ```./data/transcriptome``` where is also the manifest, then run:\n",
    "\n",
    "```\n",
    "gdc-client download -m <name of manifest>.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It will created thousands of folders, and each one contains a ```.gz``` zip file containing the transcriptomic data of 1 individual."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Extract content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add useful info of 1 individual to dict 'merge'\n",
    "# it directly uses 'match_uuid' and 'match_gene' done previously\n",
    "# to already store the useful data\n",
    "def add_case_htseq(content, merge, filename):\n",
    "    for line in content:\n",
    "        line = str(line)[2:-1]\n",
    "        for ens in match_gene.keys():\n",
    "            if ens in line:\n",
    "                htseq = line.split('\\\\t')[1][:-2]\n",
    "                merge[\"case_id\"].append(match_uuid[filename])  # add UUID \n",
    "                merge[\"gene\"].append(match_gene[line.split('.')[0]])  # add gene symbol\n",
    "                merge[\"HTseq_counts\"].append(int(htseq))   # add transcriptomic data\n",
    "    return merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "# long to run\n",
    "import gzip\n",
    "import os\n",
    "\n",
    "merge_dict = {\"case_id\": list(), \"gene\": list(), \"HTseq_counts\": list()}\n",
    "# Go through all the folders\n",
    "for root, dirs, files in os.walk(\"./data/transcriptom\", topdown=False):\n",
    "    for name in files:\n",
    "        if name.endswith(\".gz\"):   # if .gz file containing the wanted data \n",
    "            with gzip.open(os.path.join(root, name), 'r') as f:\n",
    "                content = f.readlines()   # list of the lines (costly step)\n",
    "                # add useful data into merge:\n",
    "                merge_dict = add_case_htseq(content, merge_dict, name)   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_arn = pd.DataFrame(merge_dict)\n",
    "df_arn.to_csv(\"./data/arn.tsv\")   # save the dataframe (long to run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>case_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>HTseq_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>TG</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>SDK2</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>FGF4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>CTTN</td>\n",
       "      <td>11975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>14826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75211</th>\n",
       "      <td>75211</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>AMZ2</td>\n",
       "      <td>3613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75212</th>\n",
       "      <td>75212</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>HELZ</td>\n",
       "      <td>1963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75213</th>\n",
       "      <td>75213</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75214</th>\n",
       "      <td>75214</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>POU5F1B</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75215</th>\n",
       "      <td>75215</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>DNAJB6P5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75216 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0                               case_id      gene  \\\n",
       "0               0  59bf0ce4-715a-4017-bccb-c59916e78324        TG   \n",
       "1               1  59bf0ce4-715a-4017-bccb-c59916e78324      SDK2   \n",
       "2               2  59bf0ce4-715a-4017-bccb-c59916e78324      FGF4   \n",
       "3               3  59bf0ce4-715a-4017-bccb-c59916e78324      CTTN   \n",
       "4               4  59bf0ce4-715a-4017-bccb-c59916e78324     TRPS1   \n",
       "...           ...                                   ...       ...   \n",
       "75211       75211  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4      AMZ2   \n",
       "75212       75212  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4      HELZ   \n",
       "75213       75213  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4      RYR2   \n",
       "75214       75214  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4   POU5F1B   \n",
       "75215       75215  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4  DNAJB6P5   \n",
       "\n",
       "       HTseq_counts  \n",
       "0               183  \n",
       "1                98  \n",
       "2                 0  \n",
       "3             11975  \n",
       "4             14826  \n",
       "...             ...  \n",
       "75211          3613  \n",
       "75212          1963  \n",
       "75213            15  \n",
       "75214            53  \n",
       "75215             0  \n",
       "\n",
       "[75216 rows x 4 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_arn = pd.read_csv(\"./data/arn.tsv\")\n",
    "df_arn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that we have all the dataframes required (df_case, df_cnv, df_ssm, df_arn), we can join them into one."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preparing the merge"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### SSM"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To avoid all the details of mutations, we need to aggregate the SSM data for any couple (case_id, gene). To do so, we decided to use a scoring of the types of mutations in terms of their expected consequences.\n",
    "\n",
    "* **Single base substitution**: This type of mutation corresponds to the substituation of a ATGC base by an other. It can either just change the amino acid in the protein with variable changes, create or delete a stop codon that changes the size of the protein (rare), or change nothing. We attribute a score of $1$ for every such mutation.\n",
    "\n",
    "* **Single base deletion/addition**: This type of mutation has a huge impact on the amino acid encoding since, what is called a frame-shift, can change all the amino acids after the mutation in the sequence. We attribute a score of $3$ for every such mutation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def score_attribution(x):\n",
    "    if x == \"Single base substitution\":\n",
    "        return 1\n",
    "    else:\n",
    "        return 3\n",
    "\n",
    "df_ssm[\"ssm.mutation_subtype\"] = df_ssm[\"ssm.mutation_subtype\"].apply(score_attribution)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We then sum the scores of each mutation for a given (case_id, gene):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>ssm.consequence.0.transcript.gene.symbol</th>\n",
       "      <th>ssm.mutation_subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>MUC4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>TG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1837</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1838</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>TTN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1839</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1840</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1841</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>TTN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1842 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              case.case_id  \\\n",
       "0     00807dae-9f4a-4fd1-aac2-82eb11bf2afb   \n",
       "1     00807dae-9f4a-4fd1-aac2-82eb11bf2afb   \n",
       "2     00807dae-9f4a-4fd1-aac2-82eb11bf2afb   \n",
       "3     00807dae-9f4a-4fd1-aac2-82eb11bf2afb   \n",
       "4     00807dae-9f4a-4fd1-aac2-82eb11bf2afb   \n",
       "...                                    ...   \n",
       "1837  fec0da58-1047-44d2-b6d1-c18cceed43dc   \n",
       "1838  fec0da58-1047-44d2-b6d1-c18cceed43dc   \n",
       "1839  ff0f875f-113a-4c0f-8c0a-4259cd496e6d   \n",
       "1840  ff0f875f-113a-4c0f-8c0a-4259cd496e6d   \n",
       "1841  ff0f875f-113a-4c0f-8c0a-4259cd496e6d   \n",
       "\n",
       "     ssm.consequence.0.transcript.gene.symbol  ssm.mutation_subtype  \n",
       "0                                        MUC4                     2  \n",
       "1                                      PIK3CA                     1  \n",
       "2                                        RYR2                     1  \n",
       "3                                          TG                     1  \n",
       "4                                        TP53                     1  \n",
       "...                                       ...                   ...  \n",
       "1837                                    TRPS1                     1  \n",
       "1838                                      TTN                     1  \n",
       "1839                                   PIK3CA                     1  \n",
       "1840                                     TP53                     1  \n",
       "1841                                      TTN                     1  \n",
       "\n",
       "[1842 rows x 3 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm_ = (df_ssm[[\"case.case_id\",\n",
    "                  \"ssm.consequence.0.transcript.gene.symbol\",\n",
    "                  \"ssm.mutation_subtype\"]]\n",
    "          .groupby(by=[\"case.case_id\", \"ssm.consequence.0.transcript.gene.symbol\"])\n",
    "           .sum()\n",
    "         )\n",
    "\n",
    "df_ssm_.reset_index(inplace=True)  \n",
    "df_ssm_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We rename the columns to make things simpler:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>ssm.symbol</th>\n",
       "      <th>ssm.mutation_importance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>MUC4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>TG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1837</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1838</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>TTN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1839</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1840</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1841</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>TTN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1842 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              case.case_id ssm.symbol  ssm.mutation_importance\n",
       "0     00807dae-9f4a-4fd1-aac2-82eb11bf2afb       MUC4                        2\n",
       "1     00807dae-9f4a-4fd1-aac2-82eb11bf2afb     PIK3CA                        1\n",
       "2     00807dae-9f4a-4fd1-aac2-82eb11bf2afb       RYR2                        1\n",
       "3     00807dae-9f4a-4fd1-aac2-82eb11bf2afb         TG                        1\n",
       "4     00807dae-9f4a-4fd1-aac2-82eb11bf2afb       TP53                        1\n",
       "...                                    ...        ...                      ...\n",
       "1837  fec0da58-1047-44d2-b6d1-c18cceed43dc      TRPS1                        1\n",
       "1838  fec0da58-1047-44d2-b6d1-c18cceed43dc        TTN                        1\n",
       "1839  ff0f875f-113a-4c0f-8c0a-4259cd496e6d     PIK3CA                        1\n",
       "1840  ff0f875f-113a-4c0f-8c0a-4259cd496e6d       TP53                        1\n",
       "1841  ff0f875f-113a-4c0f-8c0a-4259cd496e6d        TTN                        1\n",
       "\n",
       "[1842 rows x 3 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm_ = df_ssm_.rename(columns={\"ssm.mutation_subtype\": \"ssm.mutation_importance\",\n",
    "                       \"ssm.consequence.0.transcript.gene.symbol\": \"ssm.symbol\"})\n",
    "df_ssm_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We use pivoting to obtain a one hot encoding of every gene for a given patient:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th colspan=\"20\" halign=\"left\">ssm.mutation_importance</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ssm.symbol</th>\n",
       "      <th></th>\n",
       "      <th>ABCA10</th>\n",
       "      <th>ABCA6</th>\n",
       "      <th>ABCA8</th>\n",
       "      <th>ABCA9</th>\n",
       "      <th>AMZ2</th>\n",
       "      <th>ANO1</th>\n",
       "      <th>ARSG</th>\n",
       "      <th>BCAS3</th>\n",
       "      <th>BPTF</th>\n",
       "      <th>...</th>\n",
       "      <th>RYR2</th>\n",
       "      <th>SDK2</th>\n",
       "      <th>SHANK2</th>\n",
       "      <th>SLC16A6</th>\n",
       "      <th>TANC2</th>\n",
       "      <th>TG</th>\n",
       "      <th>TP53</th>\n",
       "      <th>TPCN2</th>\n",
       "      <th>TRPS1</th>\n",
       "      <th>TTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01263518-5f7c-49dc-8d7e-84b0c03a6a63</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>fcef8cb5-fb2c-4bfb-82cd-6b9f3145182c</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>fe15c48b-116d-461e-9b40-68514730fd18</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>fe7f74b8-20f4-4471-91dc-4cca8c68e5c0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>809 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    case.case_id ssm.mutation_importance  \\\n",
       "ssm.symbol                                                        ABCA10   \n",
       "0           00807dae-9f4a-4fd1-aac2-82eb11bf2afb                     0.0   \n",
       "1           00a2d166-78c9-4687-a195-3d6315c27574                     0.0   \n",
       "2           00b11ca8-8540-4a3d-b602-ec754b00230b                     0.0   \n",
       "3           011b9b2d-ebe5-42bf-9662-d922faccc7a1                     0.0   \n",
       "4           01263518-5f7c-49dc-8d7e-84b0c03a6a63                     0.0   \n",
       "..                                           ...                     ...   \n",
       "804         fcef8cb5-fb2c-4bfb-82cd-6b9f3145182c                     0.0   \n",
       "805         fe15c48b-116d-461e-9b40-68514730fd18                     0.0   \n",
       "806         fe7f74b8-20f4-4471-91dc-4cca8c68e5c0                     0.0   \n",
       "807         fec0da58-1047-44d2-b6d1-c18cceed43dc                     0.0   \n",
       "808         ff0f875f-113a-4c0f-8c0a-4259cd496e6d                     0.0   \n",
       "\n",
       "                                                        ...                   \\\n",
       "ssm.symbol ABCA6 ABCA8 ABCA9 AMZ2 ANO1 ARSG BCAS3 BPTF  ... RYR2 SDK2 SHANK2   \n",
       "0            0.0   0.0   0.0  0.0  0.0  0.0   0.0  0.0  ...  1.0  0.0    0.0   \n",
       "1            0.0   0.0   0.0  0.0  0.0  0.0   0.0  0.0  ...  0.0  0.0    0.0   \n",
       "2            0.0   0.0   0.0  0.0  0.0  0.0   0.0  1.0  ...  0.0  0.0    0.0   \n",
       "3            0.0   1.0   0.0  0.0  0.0  0.0   0.0  0.0  ...  0.0  1.0    0.0   \n",
       "4            1.0   0.0   0.0  0.0  0.0  0.0   0.0  1.0  ...  0.0  0.0    0.0   \n",
       "..           ...   ...   ...  ...  ...  ...   ...  ...  ...  ...  ...    ...   \n",
       "804          0.0   0.0   0.0  0.0  0.0  0.0   0.0  0.0  ...  0.0  0.0    0.0   \n",
       "805          0.0   0.0   0.0  0.0  0.0  0.0   0.0  0.0  ...  0.0  0.0    0.0   \n",
       "806          0.0   0.0   0.0  0.0  0.0  0.0   0.0  0.0  ...  0.0  0.0    0.0   \n",
       "807          0.0   0.0   3.0  0.0  0.0  0.0   0.0  0.0  ...  0.0  0.0    0.0   \n",
       "808          0.0   0.0   0.0  0.0  0.0  0.0   0.0  0.0  ...  0.0  0.0    0.0   \n",
       "\n",
       "                                                     \n",
       "ssm.symbol SLC16A6 TANC2   TG TP53 TPCN2 TRPS1  TTN  \n",
       "0              0.0   0.0  1.0  1.0   0.0   0.0  0.0  \n",
       "1              0.0   0.0  0.0  0.0   0.0   0.0  0.0  \n",
       "2              0.0   0.0  0.0  0.0   0.0   0.0  0.0  \n",
       "3              0.0   0.0  0.0  0.0   0.0   0.0  0.0  \n",
       "4              0.0   0.0  0.0  1.0   0.0   0.0  0.0  \n",
       "..             ...   ...  ...  ...   ...   ...  ...  \n",
       "804            0.0   0.0  0.0  0.0   0.0   0.0  1.0  \n",
       "805            0.0   0.0  0.0  0.0   0.0   0.0  0.0  \n",
       "806            0.0   0.0  0.0  0.0   0.0   0.0  0.0  \n",
       "807            0.0   0.0  0.0  0.0   0.0   1.0  1.0  \n",
       "808            0.0   0.0  0.0  1.0   0.0   0.0  1.0  \n",
       "\n",
       "[809 rows x 47 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm_pivot = df_ssm_.pivot(index=\"case.case_id\",\n",
    "                             columns=\"ssm.symbol\",\n",
    "                             values=[\"ssm.mutation_importance\"])\n",
    "df_ssm_pivot.reset_index(inplace=True)\n",
    "df_ssm_pivot.fillna(0., inplace=True)  # We put 0 since there is no mutation\n",
    "df_ssm_pivot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Flatten the multi-indexing\n",
    "df_ssm_pivot.columns = df_ssm_pivot.columns.droplevel()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rename every column\n",
    "renames = {}\n",
    "for x in df_ssm_pivot.columns:\n",
    "#     print(\"*\", x)\n",
    "    if x == '':\n",
    "        renames[x] = 'case.case_id'\n",
    "    else:\n",
    "        renames[x] = 'ssm.importance.' + x "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ssm.symbol</th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>ssm.importance.ABCA10</th>\n",
       "      <th>ssm.importance.ABCA6</th>\n",
       "      <th>ssm.importance.ABCA8</th>\n",
       "      <th>ssm.importance.ABCA9</th>\n",
       "      <th>ssm.importance.AMZ2</th>\n",
       "      <th>ssm.importance.ANO1</th>\n",
       "      <th>ssm.importance.ARSG</th>\n",
       "      <th>ssm.importance.BCAS3</th>\n",
       "      <th>ssm.importance.BPTF</th>\n",
       "      <th>...</th>\n",
       "      <th>ssm.importance.RYR2</th>\n",
       "      <th>ssm.importance.SDK2</th>\n",
       "      <th>ssm.importance.SHANK2</th>\n",
       "      <th>ssm.importance.SLC16A6</th>\n",
       "      <th>ssm.importance.TANC2</th>\n",
       "      <th>ssm.importance.TG</th>\n",
       "      <th>ssm.importance.TP53</th>\n",
       "      <th>ssm.importance.TPCN2</th>\n",
       "      <th>ssm.importance.TRPS1</th>\n",
       "      <th>ssm.importance.TTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01263518-5f7c-49dc-8d7e-84b0c03a6a63</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>fcef8cb5-fb2c-4bfb-82cd-6b9f3145182c</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>fe15c48b-116d-461e-9b40-68514730fd18</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>fe7f74b8-20f4-4471-91dc-4cca8c68e5c0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>809 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ssm.symbol                          case.case_id  ssm.importance.ABCA10  \\\n",
       "0           00807dae-9f4a-4fd1-aac2-82eb11bf2afb                    0.0   \n",
       "1           00a2d166-78c9-4687-a195-3d6315c27574                    0.0   \n",
       "2           00b11ca8-8540-4a3d-b602-ec754b00230b                    0.0   \n",
       "3           011b9b2d-ebe5-42bf-9662-d922faccc7a1                    0.0   \n",
       "4           01263518-5f7c-49dc-8d7e-84b0c03a6a63                    0.0   \n",
       "..                                           ...                    ...   \n",
       "804         fcef8cb5-fb2c-4bfb-82cd-6b9f3145182c                    0.0   \n",
       "805         fe15c48b-116d-461e-9b40-68514730fd18                    0.0   \n",
       "806         fe7f74b8-20f4-4471-91dc-4cca8c68e5c0                    0.0   \n",
       "807         fec0da58-1047-44d2-b6d1-c18cceed43dc                    0.0   \n",
       "808         ff0f875f-113a-4c0f-8c0a-4259cd496e6d                    0.0   \n",
       "\n",
       "ssm.symbol  ssm.importance.ABCA6  ssm.importance.ABCA8  ssm.importance.ABCA9  \\\n",
       "0                            0.0                   0.0                   0.0   \n",
       "1                            0.0                   0.0                   0.0   \n",
       "2                            0.0                   0.0                   0.0   \n",
       "3                            0.0                   1.0                   0.0   \n",
       "4                            1.0                   0.0                   0.0   \n",
       "..                           ...                   ...                   ...   \n",
       "804                          0.0                   0.0                   0.0   \n",
       "805                          0.0                   0.0                   0.0   \n",
       "806                          0.0                   0.0                   0.0   \n",
       "807                          0.0                   0.0                   3.0   \n",
       "808                          0.0                   0.0                   0.0   \n",
       "\n",
       "ssm.symbol  ssm.importance.AMZ2  ssm.importance.ANO1  ssm.importance.ARSG  \\\n",
       "0                           0.0                  0.0                  0.0   \n",
       "1                           0.0                  0.0                  0.0   \n",
       "2                           0.0                  0.0                  0.0   \n",
       "3                           0.0                  0.0                  0.0   \n",
       "4                           0.0                  0.0                  0.0   \n",
       "..                          ...                  ...                  ...   \n",
       "804                         0.0                  0.0                  0.0   \n",
       "805                         0.0                  0.0                  0.0   \n",
       "806                         0.0                  0.0                  0.0   \n",
       "807                         0.0                  0.0                  0.0   \n",
       "808                         0.0                  0.0                  0.0   \n",
       "\n",
       "ssm.symbol  ssm.importance.BCAS3  ssm.importance.BPTF  ...  \\\n",
       "0                            0.0                  0.0  ...   \n",
       "1                            0.0                  0.0  ...   \n",
       "2                            0.0                  1.0  ...   \n",
       "3                            0.0                  0.0  ...   \n",
       "4                            0.0                  1.0  ...   \n",
       "..                           ...                  ...  ...   \n",
       "804                          0.0                  0.0  ...   \n",
       "805                          0.0                  0.0  ...   \n",
       "806                          0.0                  0.0  ...   \n",
       "807                          0.0                  0.0  ...   \n",
       "808                          0.0                  0.0  ...   \n",
       "\n",
       "ssm.symbol  ssm.importance.RYR2  ssm.importance.SDK2  ssm.importance.SHANK2  \\\n",
       "0                           1.0                  0.0                    0.0   \n",
       "1                           0.0                  0.0                    0.0   \n",
       "2                           0.0                  0.0                    0.0   \n",
       "3                           0.0                  1.0                    0.0   \n",
       "4                           0.0                  0.0                    0.0   \n",
       "..                          ...                  ...                    ...   \n",
       "804                         0.0                  0.0                    0.0   \n",
       "805                         0.0                  0.0                    0.0   \n",
       "806                         0.0                  0.0                    0.0   \n",
       "807                         0.0                  0.0                    0.0   \n",
       "808                         0.0                  0.0                    0.0   \n",
       "\n",
       "ssm.symbol  ssm.importance.SLC16A6  ssm.importance.TANC2  ssm.importance.TG  \\\n",
       "0                              0.0                   0.0                1.0   \n",
       "1                              0.0                   0.0                0.0   \n",
       "2                              0.0                   0.0                0.0   \n",
       "3                              0.0                   0.0                0.0   \n",
       "4                              0.0                   0.0                0.0   \n",
       "..                             ...                   ...                ...   \n",
       "804                            0.0                   0.0                0.0   \n",
       "805                            0.0                   0.0                0.0   \n",
       "806                            0.0                   0.0                0.0   \n",
       "807                            0.0                   0.0                0.0   \n",
       "808                            0.0                   0.0                0.0   \n",
       "\n",
       "ssm.symbol  ssm.importance.TP53  ssm.importance.TPCN2  ssm.importance.TRPS1  \\\n",
       "0                           1.0                   0.0                   0.0   \n",
       "1                           0.0                   0.0                   0.0   \n",
       "2                           0.0                   0.0                   0.0   \n",
       "3                           0.0                   0.0                   0.0   \n",
       "4                           1.0                   0.0                   0.0   \n",
       "..                          ...                   ...                   ...   \n",
       "804                         0.0                   0.0                   0.0   \n",
       "805                         0.0                   0.0                   0.0   \n",
       "806                         0.0                   0.0                   0.0   \n",
       "807                         0.0                   0.0                   1.0   \n",
       "808                         1.0                   0.0                   0.0   \n",
       "\n",
       "ssm.symbol  ssm.importance.TTN  \n",
       "0                          0.0  \n",
       "1                          0.0  \n",
       "2                          0.0  \n",
       "3                          0.0  \n",
       "4                          0.0  \n",
       "..                         ...  \n",
       "804                        1.0  \n",
       "805                        0.0  \n",
       "806                        0.0  \n",
       "807                        1.0  \n",
       "808                        1.0  \n",
       "\n",
       "[809 rows x 47 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm_pivot = df_ssm_pivot.rename(columns=renames)\n",
    "df_ssm_pivot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CNV"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We do the the same king of aggregation with CNV except we normally have only one cnv max per gene. We still aggregate with a 'min' to manage only one if there turns out to be 2 CNVs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>cnv.symbol</th>\n",
       "      <th>cnv.cnv_change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>001cef41-ff86-4d3f-a140-a647ac4b10a1</td>\n",
       "      <td>ANO1</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>001cef41-ff86-4d3f-a140-a647ac4b10a1</td>\n",
       "      <td>AP001888.1</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>001cef41-ff86-4d3f-a140-a647ac4b10a1</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>001cef41-ff86-4d3f-a140-a647ac4b10a1</td>\n",
       "      <td>CPT1A</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>001cef41-ff86-4d3f-a140-a647ac4b10a1</td>\n",
       "      <td>CTTN</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12876</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>ORAOV1</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12877</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>PPFIA1</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12878</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>SHANK2</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12879</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>TPCN2</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12880</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>TTN</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12881 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               case.case_id  cnv.symbol cnv.cnv_change\n",
       "0      001cef41-ff86-4d3f-a140-a647ac4b10a1        ANO1           Gain\n",
       "1      001cef41-ff86-4d3f-a140-a647ac4b10a1  AP001888.1           Gain\n",
       "2      001cef41-ff86-4d3f-a140-a647ac4b10a1       CCND1           Gain\n",
       "3      001cef41-ff86-4d3f-a140-a647ac4b10a1       CPT1A           Gain\n",
       "4      001cef41-ff86-4d3f-a140-a647ac4b10a1        CTTN           Gain\n",
       "...                                     ...         ...            ...\n",
       "12876  ff0f875f-113a-4c0f-8c0a-4259cd496e6d      ORAOV1           Gain\n",
       "12877  ff0f875f-113a-4c0f-8c0a-4259cd496e6d      PPFIA1           Gain\n",
       "12878  ff0f875f-113a-4c0f-8c0a-4259cd496e6d      SHANK2           Gain\n",
       "12879  ff0f875f-113a-4c0f-8c0a-4259cd496e6d       TPCN2           Gain\n",
       "12880  ff0f875f-113a-4c0f-8c0a-4259cd496e6d         TTN           Gain\n",
       "\n",
       "[12881 rows x 3 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cnv_ = (df_cnv[[\"case.case_id\", \"cnv.cnv_change\", \"cnv.consequence.0.gene.symbol\"]]\n",
    "           .groupby([\"case.case_id\", \"cnv.consequence.0.gene.symbol\"])\n",
    "           .aggregate('min')\n",
    "          )\n",
    "df_cnv_.reset_index(inplace=True)\n",
    "df_cnv_ = df_cnv_.rename(columns={'cnv.consequence.0.gene.symbol':\n",
    "                                 'cnv.symbol'})\n",
    "df_cnv_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pivoting:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cnv_pivot = df_cnv_.pivot(index=\"case.case_id\",\n",
    "                             columns=\"cnv.symbol\",\n",
    "                             values=[\"cnv.cnv_change\"])\n",
    "df_cnv_pivot.reset_index(inplace=True)\n",
    "df_cnv_pivot.fillna('None', inplace=True)\n",
    "df_cnv_pivot.columns = df_cnv_pivot.columns.droplevel()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Renaming:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "renames = {}\n",
    "for x in df_cnv_pivot.columns:\n",
    "#     print(\"*\", x)\n",
    "    if x == '':\n",
    "        renames[x] = 'case.case_id'\n",
    "    else:\n",
    "        renames[x] = 'cnv.change.' + x "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>cnv.symbol</th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>cnv.change.ABCA10</th>\n",
       "      <th>cnv.change.ABCA6</th>\n",
       "      <th>cnv.change.ABCA8</th>\n",
       "      <th>cnv.change.ABCA9</th>\n",
       "      <th>cnv.change.AC108925.1</th>\n",
       "      <th>cnv.change.AMZ2</th>\n",
       "      <th>cnv.change.ANO1</th>\n",
       "      <th>cnv.change.AP001888.1</th>\n",
       "      <th>cnv.change.ARSG</th>\n",
       "      <th>...</th>\n",
       "      <th>cnv.change.RYR2</th>\n",
       "      <th>cnv.change.SDK2</th>\n",
       "      <th>cnv.change.SHANK2</th>\n",
       "      <th>cnv.change.SLC16A6</th>\n",
       "      <th>cnv.change.TANC2</th>\n",
       "      <th>cnv.change.TG</th>\n",
       "      <th>cnv.change.TP53</th>\n",
       "      <th>cnv.change.TPCN2</th>\n",
       "      <th>cnv.change.TRPS1</th>\n",
       "      <th>cnv.change.TTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>001cef41-ff86-4d3f-a140-a647ac4b10a1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>793</th>\n",
       "      <td>fe15c48b-116d-461e-9b40-68514730fd18</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>794</th>\n",
       "      <td>fe2cd610-aa52-4789-ac62-7683281bb22f</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>795</th>\n",
       "      <td>fe7f74b8-20f4-4471-91dc-4cca8c68e5c0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>796</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>797</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>798 rows × 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "cnv.symbol                          case.case_id cnv.change.ABCA10  \\\n",
       "0           001cef41-ff86-4d3f-a140-a647ac4b10a1              None   \n",
       "1           00807dae-9f4a-4fd1-aac2-82eb11bf2afb              Gain   \n",
       "2           00a2d166-78c9-4687-a195-3d6315c27574              Gain   \n",
       "3           00b11ca8-8540-4a3d-b602-ec754b00230b              None   \n",
       "4           011b9b2d-ebe5-42bf-9662-d922faccc7a1              None   \n",
       "..                                           ...               ...   \n",
       "793         fe15c48b-116d-461e-9b40-68514730fd18              None   \n",
       "794         fe2cd610-aa52-4789-ac62-7683281bb22f              None   \n",
       "795         fe7f74b8-20f4-4471-91dc-4cca8c68e5c0              None   \n",
       "796         fec0da58-1047-44d2-b6d1-c18cceed43dc              None   \n",
       "797         ff0f875f-113a-4c0f-8c0a-4259cd496e6d              None   \n",
       "\n",
       "cnv.symbol cnv.change.ABCA6 cnv.change.ABCA8 cnv.change.ABCA9  \\\n",
       "0                      None             None             None   \n",
       "1                      Gain             Gain             Gain   \n",
       "2                      Gain             Gain             Gain   \n",
       "3                      None             None             None   \n",
       "4                      None             None             None   \n",
       "..                      ...              ...              ...   \n",
       "793                    None             None             None   \n",
       "794                    None             None             None   \n",
       "795                    None             None             None   \n",
       "796                    None             None             None   \n",
       "797                    None             None             None   \n",
       "\n",
       "cnv.symbol cnv.change.AC108925.1 cnv.change.AMZ2 cnv.change.ANO1  \\\n",
       "0                           None            None            Gain   \n",
       "1                           None            Gain            None   \n",
       "2                           None            Gain            None   \n",
       "3                           None            None            Gain   \n",
       "4                           None            None            None   \n",
       "..                           ...             ...             ...   \n",
       "793                         None            None            None   \n",
       "794                         None            None            Gain   \n",
       "795                         None            None            None   \n",
       "796                         None            None            Gain   \n",
       "797                         None            None            Gain   \n",
       "\n",
       "cnv.symbol cnv.change.AP001888.1 cnv.change.ARSG  ... cnv.change.RYR2  \\\n",
       "0                           Gain            None  ...            None   \n",
       "1                           None            Gain  ...            None   \n",
       "2                           None            Gain  ...            None   \n",
       "3                           Gain            None  ...            None   \n",
       "4                           None            None  ...            Gain   \n",
       "..                           ...             ...  ...             ...   \n",
       "793                         None            None  ...            None   \n",
       "794                         Gain            None  ...            None   \n",
       "795                         None            None  ...            None   \n",
       "796                         Gain            None  ...            None   \n",
       "797                         Gain            None  ...            None   \n",
       "\n",
       "cnv.symbol cnv.change.SDK2 cnv.change.SHANK2 cnv.change.SLC16A6  \\\n",
       "0                     None              Gain               None   \n",
       "1                     Gain              None               Gain   \n",
       "2                     Gain              None               Gain   \n",
       "3                     None              None               None   \n",
       "4                     None              None               None   \n",
       "..                     ...               ...                ...   \n",
       "793                   None              None               None   \n",
       "794                   None              Gain               None   \n",
       "795                   Gain              None               None   \n",
       "796                   None              Gain               None   \n",
       "797                   None              Gain               None   \n",
       "\n",
       "cnv.symbol cnv.change.TANC2 cnv.change.TG cnv.change.TP53 cnv.change.TPCN2  \\\n",
       "0                      None          None            None             Gain   \n",
       "1                      Gain          None            None             None   \n",
       "2                      Gain          None            None             None   \n",
       "3                      None          None            None             Gain   \n",
       "4                      None          None            None             None   \n",
       "..                      ...           ...             ...              ...   \n",
       "793                    None          None            None             None   \n",
       "794                    None          None            None             Gain   \n",
       "795                    Gain          None            None             None   \n",
       "796                    None          None            None             Gain   \n",
       "797                    None          None            None             Gain   \n",
       "\n",
       "cnv.symbol cnv.change.TRPS1 cnv.change.TTN  \n",
       "0                      None           None  \n",
       "1                      None           None  \n",
       "2                      None           None  \n",
       "3                      None           None  \n",
       "4                      None           None  \n",
       "..                      ...            ...  \n",
       "793                    None           None  \n",
       "794                    None           None  \n",
       "795                    None           None  \n",
       "796                    None           None  \n",
       "797                    None           Gain  \n",
       "\n",
       "[798 rows x 51 columns]"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cnv_pivot = df_cnv_pivot.rename(columns=renames)\n",
    "df_cnv_pivot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ARN"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We only need pivoting and renaming for ARN."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>case_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>HTseq_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>TG</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>SDK2</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>FGF4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>CTTN</td>\n",
       "      <td>11975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>59bf0ce4-715a-4017-bccb-c59916e78324</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>14826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75211</th>\n",
       "      <td>75211</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>AMZ2</td>\n",
       "      <td>3613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75212</th>\n",
       "      <td>75212</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>HELZ</td>\n",
       "      <td>1963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75213</th>\n",
       "      <td>75213</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75214</th>\n",
       "      <td>75214</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>POU5F1B</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75215</th>\n",
       "      <td>75215</td>\n",
       "      <td>76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4</td>\n",
       "      <td>DNAJB6P5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75216 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0                               case_id      gene  \\\n",
       "0               0  59bf0ce4-715a-4017-bccb-c59916e78324        TG   \n",
       "1               1  59bf0ce4-715a-4017-bccb-c59916e78324      SDK2   \n",
       "2               2  59bf0ce4-715a-4017-bccb-c59916e78324      FGF4   \n",
       "3               3  59bf0ce4-715a-4017-bccb-c59916e78324      CTTN   \n",
       "4               4  59bf0ce4-715a-4017-bccb-c59916e78324     TRPS1   \n",
       "...           ...                                   ...       ...   \n",
       "75211       75211  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4      AMZ2   \n",
       "75212       75212  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4      HELZ   \n",
       "75213       75213  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4      RYR2   \n",
       "75214       75214  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4   POU5F1B   \n",
       "75215       75215  76bd9bb9-9a8b-4a90-bb7c-eafe76472ee4  DNAJB6P5   \n",
       "\n",
       "       HTseq_counts  \n",
       "0               183  \n",
       "1                98  \n",
       "2                 0  \n",
       "3             11975  \n",
       "4             14826  \n",
       "...             ...  \n",
       "75211          3613  \n",
       "75212          1963  \n",
       "75213            15  \n",
       "75214            53  \n",
       "75215             0  \n",
       "\n",
       "[75216 rows x 4 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_arn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_arn_pivot = df_arn.pivot_table(index=\"case_id\",\n",
    "                             columns=\"gene\",\n",
    "                             values=\"HTseq_counts\",\n",
    "                             aggfunc='sum'\n",
    "                           )\n",
    "df_arn_pivot.reset_index(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "renames = {}\n",
    "for x in df_arn_pivot.columns:\n",
    "    if x in ['case_id']:\n",
    "        renames[x] = x\n",
    "    else:\n",
    "        renames[x] = 'arn.HTseq_counts.' + x "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>gene</th>\n",
       "      <th>case_id</th>\n",
       "      <th>arn.HTseq_counts.ABCA10</th>\n",
       "      <th>arn.HTseq_counts.ABCA6</th>\n",
       "      <th>arn.HTseq_counts.ABCA8</th>\n",
       "      <th>arn.HTseq_counts.ABCA9</th>\n",
       "      <th>arn.HTseq_counts.AMZ2</th>\n",
       "      <th>arn.HTseq_counts.ANO1</th>\n",
       "      <th>arn.HTseq_counts.ARSG</th>\n",
       "      <th>arn.HTseq_counts.BCAS3</th>\n",
       "      <th>arn.HTseq_counts.BPTF</th>\n",
       "      <th>...</th>\n",
       "      <th>arn.HTseq_counts.RYR2</th>\n",
       "      <th>arn.HTseq_counts.SDK2</th>\n",
       "      <th>arn.HTseq_counts.SHANK2</th>\n",
       "      <th>arn.HTseq_counts.SLC16A6</th>\n",
       "      <th>arn.HTseq_counts.TANC2</th>\n",
       "      <th>arn.HTseq_counts.TG</th>\n",
       "      <th>arn.HTseq_counts.TP53</th>\n",
       "      <th>arn.HTseq_counts.TPCN2</th>\n",
       "      <th>arn.HTseq_counts.TRPS1</th>\n",
       "      <th>arn.HTseq_counts.TTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>001cef41-ff86-4d3f-a140-a647ac4b10a1</td>\n",
       "      <td>46</td>\n",
       "      <td>160</td>\n",
       "      <td>197</td>\n",
       "      <td>85</td>\n",
       "      <td>5055</td>\n",
       "      <td>4188</td>\n",
       "      <td>2270</td>\n",
       "      <td>1645</td>\n",
       "      <td>5340</td>\n",
       "      <td>...</td>\n",
       "      <td>42</td>\n",
       "      <td>368</td>\n",
       "      <td>1628</td>\n",
       "      <td>2222</td>\n",
       "      <td>4110</td>\n",
       "      <td>18</td>\n",
       "      <td>3437</td>\n",
       "      <td>1057</td>\n",
       "      <td>27924</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0045349c-69d9-4306-a403-c9c1fa836644</td>\n",
       "      <td>61</td>\n",
       "      <td>127</td>\n",
       "      <td>347</td>\n",
       "      <td>194</td>\n",
       "      <td>2248</td>\n",
       "      <td>7408</td>\n",
       "      <td>823</td>\n",
       "      <td>2302</td>\n",
       "      <td>9236</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>3836</td>\n",
       "      <td>8336</td>\n",
       "      <td>165</td>\n",
       "      <td>1714</td>\n",
       "      <td>133</td>\n",
       "      <td>11130</td>\n",
       "      <td>758</td>\n",
       "      <td>12095</td>\n",
       "      <td>1219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>007df8da-3d8c-497d-b7d3-3a02596f9577</td>\n",
       "      <td>111</td>\n",
       "      <td>225</td>\n",
       "      <td>217</td>\n",
       "      <td>172</td>\n",
       "      <td>2644</td>\n",
       "      <td>1289</td>\n",
       "      <td>598</td>\n",
       "      <td>1210</td>\n",
       "      <td>2906</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>36</td>\n",
       "      <td>1146</td>\n",
       "      <td>2870</td>\n",
       "      <td>3592</td>\n",
       "      <td>16</td>\n",
       "      <td>389</td>\n",
       "      <td>401</td>\n",
       "      <td>20801</td>\n",
       "      <td>1074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>31</td>\n",
       "      <td>127</td>\n",
       "      <td>121</td>\n",
       "      <td>88</td>\n",
       "      <td>2932</td>\n",
       "      <td>1694</td>\n",
       "      <td>488</td>\n",
       "      <td>376</td>\n",
       "      <td>2349</td>\n",
       "      <td>...</td>\n",
       "      <td>83</td>\n",
       "      <td>74</td>\n",
       "      <td>1942</td>\n",
       "      <td>729</td>\n",
       "      <td>3846</td>\n",
       "      <td>18</td>\n",
       "      <td>2997</td>\n",
       "      <td>655</td>\n",
       "      <td>8320</td>\n",
       "      <td>341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>21</td>\n",
       "      <td>143</td>\n",
       "      <td>26</td>\n",
       "      <td>68</td>\n",
       "      <td>13775</td>\n",
       "      <td>2622</td>\n",
       "      <td>1024</td>\n",
       "      <td>1966</td>\n",
       "      <td>9583</td>\n",
       "      <td>...</td>\n",
       "      <td>34</td>\n",
       "      <td>266</td>\n",
       "      <td>1274</td>\n",
       "      <td>700</td>\n",
       "      <td>9455</td>\n",
       "      <td>19</td>\n",
       "      <td>6779</td>\n",
       "      <td>1123</td>\n",
       "      <td>13940</td>\n",
       "      <td>306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1368</th>\n",
       "      <td>fe2cd610-aa52-4789-ac62-7683281bb22f</td>\n",
       "      <td>57</td>\n",
       "      <td>54</td>\n",
       "      <td>20</td>\n",
       "      <td>16</td>\n",
       "      <td>4116</td>\n",
       "      <td>1046</td>\n",
       "      <td>1050</td>\n",
       "      <td>936</td>\n",
       "      <td>4704</td>\n",
       "      <td>...</td>\n",
       "      <td>2645</td>\n",
       "      <td>24</td>\n",
       "      <td>5624</td>\n",
       "      <td>1233</td>\n",
       "      <td>913</td>\n",
       "      <td>27</td>\n",
       "      <td>6768</td>\n",
       "      <td>614</td>\n",
       "      <td>25012</td>\n",
       "      <td>437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1369</th>\n",
       "      <td>fe7f74b8-20f4-4471-91dc-4cca8c68e5c0</td>\n",
       "      <td>164</td>\n",
       "      <td>367</td>\n",
       "      <td>314</td>\n",
       "      <td>299</td>\n",
       "      <td>1976</td>\n",
       "      <td>3194</td>\n",
       "      <td>462</td>\n",
       "      <td>2483</td>\n",
       "      <td>10942</td>\n",
       "      <td>...</td>\n",
       "      <td>27</td>\n",
       "      <td>405</td>\n",
       "      <td>3441</td>\n",
       "      <td>390</td>\n",
       "      <td>6497</td>\n",
       "      <td>27</td>\n",
       "      <td>4153</td>\n",
       "      <td>1563</td>\n",
       "      <td>4224</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1370</th>\n",
       "      <td>fec0da58-1047-44d2-b6d1-c18cceed43dc</td>\n",
       "      <td>92</td>\n",
       "      <td>414</td>\n",
       "      <td>894</td>\n",
       "      <td>317</td>\n",
       "      <td>4782</td>\n",
       "      <td>5776</td>\n",
       "      <td>1640</td>\n",
       "      <td>1968</td>\n",
       "      <td>4447</td>\n",
       "      <td>...</td>\n",
       "      <td>44</td>\n",
       "      <td>207</td>\n",
       "      <td>1172</td>\n",
       "      <td>951</td>\n",
       "      <td>6078</td>\n",
       "      <td>274</td>\n",
       "      <td>3132</td>\n",
       "      <td>1021</td>\n",
       "      <td>20023</td>\n",
       "      <td>263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1371</th>\n",
       "      <td>ff0f875f-113a-4c0f-8c0a-4259cd496e6d</td>\n",
       "      <td>270</td>\n",
       "      <td>675</td>\n",
       "      <td>726</td>\n",
       "      <td>509</td>\n",
       "      <td>4611</td>\n",
       "      <td>36665</td>\n",
       "      <td>695</td>\n",
       "      <td>1016</td>\n",
       "      <td>4505</td>\n",
       "      <td>...</td>\n",
       "      <td>74</td>\n",
       "      <td>231</td>\n",
       "      <td>1224</td>\n",
       "      <td>510</td>\n",
       "      <td>3770</td>\n",
       "      <td>81</td>\n",
       "      <td>6545</td>\n",
       "      <td>1381</td>\n",
       "      <td>24171</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1372</th>\n",
       "      <td>ff286f28-aff1-4de8-9a51-00b549fb7f86</td>\n",
       "      <td>10696</td>\n",
       "      <td>15045</td>\n",
       "      <td>9583</td>\n",
       "      <td>11199</td>\n",
       "      <td>8068</td>\n",
       "      <td>29236</td>\n",
       "      <td>18962</td>\n",
       "      <td>20356</td>\n",
       "      <td>141945</td>\n",
       "      <td>...</td>\n",
       "      <td>2838</td>\n",
       "      <td>2497</td>\n",
       "      <td>65652</td>\n",
       "      <td>7139</td>\n",
       "      <td>103396</td>\n",
       "      <td>6868</td>\n",
       "      <td>15514</td>\n",
       "      <td>8814</td>\n",
       "      <td>756446</td>\n",
       "      <td>89644</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1373 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "gene                               case_id  arn.HTseq_counts.ABCA10  \\\n",
       "0     001cef41-ff86-4d3f-a140-a647ac4b10a1                       46   \n",
       "1     0045349c-69d9-4306-a403-c9c1fa836644                       61   \n",
       "2     007df8da-3d8c-497d-b7d3-3a02596f9577                      111   \n",
       "3     00807dae-9f4a-4fd1-aac2-82eb11bf2afb                       31   \n",
       "4     00a2d166-78c9-4687-a195-3d6315c27574                       21   \n",
       "...                                    ...                      ...   \n",
       "1368  fe2cd610-aa52-4789-ac62-7683281bb22f                       57   \n",
       "1369  fe7f74b8-20f4-4471-91dc-4cca8c68e5c0                      164   \n",
       "1370  fec0da58-1047-44d2-b6d1-c18cceed43dc                       92   \n",
       "1371  ff0f875f-113a-4c0f-8c0a-4259cd496e6d                      270   \n",
       "1372  ff286f28-aff1-4de8-9a51-00b549fb7f86                    10696   \n",
       "\n",
       "gene  arn.HTseq_counts.ABCA6  arn.HTseq_counts.ABCA8  arn.HTseq_counts.ABCA9  \\\n",
       "0                        160                     197                      85   \n",
       "1                        127                     347                     194   \n",
       "2                        225                     217                     172   \n",
       "3                        127                     121                      88   \n",
       "4                        143                      26                      68   \n",
       "...                      ...                     ...                     ...   \n",
       "1368                      54                      20                      16   \n",
       "1369                     367                     314                     299   \n",
       "1370                     414                     894                     317   \n",
       "1371                     675                     726                     509   \n",
       "1372                   15045                    9583                   11199   \n",
       "\n",
       "gene  arn.HTseq_counts.AMZ2  arn.HTseq_counts.ANO1  arn.HTseq_counts.ARSG  \\\n",
       "0                      5055                   4188                   2270   \n",
       "1                      2248                   7408                    823   \n",
       "2                      2644                   1289                    598   \n",
       "3                      2932                   1694                    488   \n",
       "4                     13775                   2622                   1024   \n",
       "...                     ...                    ...                    ...   \n",
       "1368                   4116                   1046                   1050   \n",
       "1369                   1976                   3194                    462   \n",
       "1370                   4782                   5776                   1640   \n",
       "1371                   4611                  36665                    695   \n",
       "1372                   8068                  29236                  18962   \n",
       "\n",
       "gene  arn.HTseq_counts.BCAS3  arn.HTseq_counts.BPTF  ...  \\\n",
       "0                       1645                   5340  ...   \n",
       "1                       2302                   9236  ...   \n",
       "2                       1210                   2906  ...   \n",
       "3                        376                   2349  ...   \n",
       "4                       1966                   9583  ...   \n",
       "...                      ...                    ...  ...   \n",
       "1368                     936                   4704  ...   \n",
       "1369                    2483                  10942  ...   \n",
       "1370                    1968                   4447  ...   \n",
       "1371                    1016                   4505  ...   \n",
       "1372                   20356                 141945  ...   \n",
       "\n",
       "gene  arn.HTseq_counts.RYR2  arn.HTseq_counts.SDK2  arn.HTseq_counts.SHANK2  \\\n",
       "0                        42                    368                     1628   \n",
       "1                        33                   3836                     8336   \n",
       "2                        30                     36                     1146   \n",
       "3                        83                     74                     1942   \n",
       "4                        34                    266                     1274   \n",
       "...                     ...                    ...                      ...   \n",
       "1368                   2645                     24                     5624   \n",
       "1369                     27                    405                     3441   \n",
       "1370                     44                    207                     1172   \n",
       "1371                     74                    231                     1224   \n",
       "1372                   2838                   2497                    65652   \n",
       "\n",
       "gene  arn.HTseq_counts.SLC16A6  arn.HTseq_counts.TANC2  arn.HTseq_counts.TG  \\\n",
       "0                         2222                    4110                   18   \n",
       "1                          165                    1714                  133   \n",
       "2                         2870                    3592                   16   \n",
       "3                          729                    3846                   18   \n",
       "4                          700                    9455                   19   \n",
       "...                        ...                     ...                  ...   \n",
       "1368                      1233                     913                   27   \n",
       "1369                       390                    6497                   27   \n",
       "1370                       951                    6078                  274   \n",
       "1371                       510                    3770                   81   \n",
       "1372                      7139                  103396                 6868   \n",
       "\n",
       "gene  arn.HTseq_counts.TP53  arn.HTseq_counts.TPCN2  arn.HTseq_counts.TRPS1  \\\n",
       "0                      3437                    1057                   27924   \n",
       "1                     11130                     758                   12095   \n",
       "2                       389                     401                   20801   \n",
       "3                      2997                     655                    8320   \n",
       "4                      6779                    1123                   13940   \n",
       "...                     ...                     ...                     ...   \n",
       "1368                   6768                     614                   25012   \n",
       "1369                   4153                    1563                    4224   \n",
       "1370                   3132                    1021                   20023   \n",
       "1371                   6545                    1381                   24171   \n",
       "1372                  15514                    8814                  756446   \n",
       "\n",
       "gene  arn.HTseq_counts.TTN  \n",
       "0                      285  \n",
       "1                     1219  \n",
       "2                     1074  \n",
       "3                      341  \n",
       "4                      306  \n",
       "...                    ...  \n",
       "1368                   437  \n",
       "1369                   749  \n",
       "1370                   263  \n",
       "1371                   767  \n",
       "1372                 89644  \n",
       "\n",
       "[1373 rows x 49 columns]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_arn_pivot = df_arn_pivot.rename(columns=renames)\n",
    "df_arn_pivot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### MERGE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to keep the individuals that have at least one info on their genetics. We use 'outer' joins first between genetic tables (SSM, CNV, ARN), then 'right' on the merge with Case, where Case is on the left."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SSM + CNV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>ssm.importance.ABCA10</th>\n",
       "      <th>ssm.importance.ABCA6</th>\n",
       "      <th>ssm.importance.ABCA8</th>\n",
       "      <th>ssm.importance.ABCA9</th>\n",
       "      <th>ssm.importance.AMZ2</th>\n",
       "      <th>ssm.importance.ANO1</th>\n",
       "      <th>ssm.importance.ARSG</th>\n",
       "      <th>ssm.importance.BCAS3</th>\n",
       "      <th>ssm.importance.BPTF</th>\n",
       "      <th>...</th>\n",
       "      <th>cnv.change.RYR2</th>\n",
       "      <th>cnv.change.SDK2</th>\n",
       "      <th>cnv.change.SHANK2</th>\n",
       "      <th>cnv.change.SLC16A6</th>\n",
       "      <th>cnv.change.TANC2</th>\n",
       "      <th>cnv.change.TG</th>\n",
       "      <th>cnv.change.TP53</th>\n",
       "      <th>cnv.change.TPCN2</th>\n",
       "      <th>cnv.change.TRPS1</th>\n",
       "      <th>cnv.change.TTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01263518-5f7c-49dc-8d7e-84b0c03a6a63</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1002</th>\n",
       "      <td>f6eebd4b-b63a-4a9c-92d3-0d954a8a6655</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Loss</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>f9eb88f7-9293-46bf-ace4-a746e4ff80dc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1004</th>\n",
       "      <td>fbee40f1-d6d8-4156-8d42-36e09bb9f095</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005</th>\n",
       "      <td>fd0b1b8b-1a76-480a-b61c-d176e4e02c6b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>fe2cd610-aa52-4789-ac62-7683281bb22f</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Gain</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1007 rows × 97 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              case.case_id  ssm.importance.ABCA10  \\\n",
       "0     00807dae-9f4a-4fd1-aac2-82eb11bf2afb                    0.0   \n",
       "1     00a2d166-78c9-4687-a195-3d6315c27574                    0.0   \n",
       "2     00b11ca8-8540-4a3d-b602-ec754b00230b                    0.0   \n",
       "3     011b9b2d-ebe5-42bf-9662-d922faccc7a1                    0.0   \n",
       "4     01263518-5f7c-49dc-8d7e-84b0c03a6a63                    0.0   \n",
       "...                                    ...                    ...   \n",
       "1002  f6eebd4b-b63a-4a9c-92d3-0d954a8a6655                    NaN   \n",
       "1003  f9eb88f7-9293-46bf-ace4-a746e4ff80dc                    NaN   \n",
       "1004  fbee40f1-d6d8-4156-8d42-36e09bb9f095                    NaN   \n",
       "1005  fd0b1b8b-1a76-480a-b61c-d176e4e02c6b                    NaN   \n",
       "1006  fe2cd610-aa52-4789-ac62-7683281bb22f                    NaN   \n",
       "\n",
       "      ssm.importance.ABCA6  ssm.importance.ABCA8  ssm.importance.ABCA9  \\\n",
       "0                      0.0                   0.0                   0.0   \n",
       "1                      0.0                   0.0                   0.0   \n",
       "2                      0.0                   0.0                   0.0   \n",
       "3                      0.0                   1.0                   0.0   \n",
       "4                      1.0                   0.0                   0.0   \n",
       "...                    ...                   ...                   ...   \n",
       "1002                   NaN                   NaN                   NaN   \n",
       "1003                   NaN                   NaN                   NaN   \n",
       "1004                   NaN                   NaN                   NaN   \n",
       "1005                   NaN                   NaN                   NaN   \n",
       "1006                   NaN                   NaN                   NaN   \n",
       "\n",
       "      ssm.importance.AMZ2  ssm.importance.ANO1  ssm.importance.ARSG  \\\n",
       "0                     0.0                  0.0                  0.0   \n",
       "1                     0.0                  0.0                  0.0   \n",
       "2                     0.0                  0.0                  0.0   \n",
       "3                     0.0                  0.0                  0.0   \n",
       "4                     0.0                  0.0                  0.0   \n",
       "...                   ...                  ...                  ...   \n",
       "1002                  NaN                  NaN                  NaN   \n",
       "1003                  NaN                  NaN                  NaN   \n",
       "1004                  NaN                  NaN                  NaN   \n",
       "1005                  NaN                  NaN                  NaN   \n",
       "1006                  NaN                  NaN                  NaN   \n",
       "\n",
       "      ssm.importance.BCAS3  ssm.importance.BPTF  ...  cnv.change.RYR2  \\\n",
       "0                      0.0                  0.0  ...             None   \n",
       "1                      0.0                  0.0  ...             None   \n",
       "2                      0.0                  1.0  ...             None   \n",
       "3                      0.0                  0.0  ...             Gain   \n",
       "4                      0.0                  1.0  ...             None   \n",
       "...                    ...                  ...  ...              ...   \n",
       "1002                   NaN                  NaN  ...             None   \n",
       "1003                   NaN                  NaN  ...             None   \n",
       "1004                   NaN                  NaN  ...             None   \n",
       "1005                   NaN                  NaN  ...             None   \n",
       "1006                   NaN                  NaN  ...             None   \n",
       "\n",
       "      cnv.change.SDK2  cnv.change.SHANK2  cnv.change.SLC16A6  \\\n",
       "0                Gain               None                Gain   \n",
       "1                Gain               None                Gain   \n",
       "2                None               None                None   \n",
       "3                None               None                None   \n",
       "4                None               Gain                None   \n",
       "...               ...                ...                 ...   \n",
       "1002             None               Gain                None   \n",
       "1003             None               Gain                None   \n",
       "1004             None               Gain                None   \n",
       "1005             None               None                None   \n",
       "1006             None               Gain                None   \n",
       "\n",
       "      cnv.change.TANC2  cnv.change.TG  cnv.change.TP53  cnv.change.TPCN2  \\\n",
       "0                 Gain           None             None              None   \n",
       "1                 Gain           None             None              None   \n",
       "2                 None           None             None              Gain   \n",
       "3                 None           None             None              None   \n",
       "4                 None           None             None              Gain   \n",
       "...                ...            ...              ...               ...   \n",
       "1002              None           None             None              Loss   \n",
       "1003              None           None             None              Gain   \n",
       "1004              None           Gain             None              Gain   \n",
       "1005              None           Gain             None              None   \n",
       "1006              None           None             None              Gain   \n",
       "\n",
       "      cnv.change.TRPS1  cnv.change.TTN  \n",
       "0                 None            None  \n",
       "1                 None            None  \n",
       "2                 None            None  \n",
       "3                 None            None  \n",
       "4                 None            None  \n",
       "...                ...             ...  \n",
       "1002              None            None  \n",
       "1003              None            None  \n",
       "1004              Gain            None  \n",
       "1005              Gain            None  \n",
       "1006              None            None  \n",
       "\n",
       "[1007 rows x 97 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.merge(df_ssm_pivot, df_cnv_pivot, how='outer', left_on='case.case_id', right_on='case.case_id')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "(SSM + CNV) + ARN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>ssm.importance.ABCA10</th>\n",
       "      <th>ssm.importance.ABCA6</th>\n",
       "      <th>ssm.importance.ABCA8</th>\n",
       "      <th>ssm.importance.ABCA9</th>\n",
       "      <th>ssm.importance.AMZ2</th>\n",
       "      <th>ssm.importance.ANO1</th>\n",
       "      <th>ssm.importance.ARSG</th>\n",
       "      <th>ssm.importance.BCAS3</th>\n",
       "      <th>ssm.importance.BPTF</th>\n",
       "      <th>...</th>\n",
       "      <th>arn.HTseq_counts.RYR2</th>\n",
       "      <th>arn.HTseq_counts.SDK2</th>\n",
       "      <th>arn.HTseq_counts.SHANK2</th>\n",
       "      <th>arn.HTseq_counts.SLC16A6</th>\n",
       "      <th>arn.HTseq_counts.TANC2</th>\n",
       "      <th>arn.HTseq_counts.TG</th>\n",
       "      <th>arn.HTseq_counts.TP53</th>\n",
       "      <th>arn.HTseq_counts.TPCN2</th>\n",
       "      <th>arn.HTseq_counts.TRPS1</th>\n",
       "      <th>arn.HTseq_counts.TTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>83.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>1942.0</td>\n",
       "      <td>729.0</td>\n",
       "      <td>3846.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2997.0</td>\n",
       "      <td>655.0</td>\n",
       "      <td>8320.0</td>\n",
       "      <td>341.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>34.0</td>\n",
       "      <td>266.0</td>\n",
       "      <td>1274.0</td>\n",
       "      <td>700.0</td>\n",
       "      <td>9455.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>6779.0</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>13940.0</td>\n",
       "      <td>306.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>177.0</td>\n",
       "      <td>528.0</td>\n",
       "      <td>2830.0</td>\n",
       "      <td>880.0</td>\n",
       "      <td>5889.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>4543.0</td>\n",
       "      <td>3168.0</td>\n",
       "      <td>15170.0</td>\n",
       "      <td>1334.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>418.0</td>\n",
       "      <td>603.0</td>\n",
       "      <td>6459.0</td>\n",
       "      <td>12415.0</td>\n",
       "      <td>17324.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>7822.0</td>\n",
       "      <td>1206.0</td>\n",
       "      <td>212884.0</td>\n",
       "      <td>19495.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01263518-5f7c-49dc-8d7e-84b0c03a6a63</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>29.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>4405.0</td>\n",
       "      <td>1289.0</td>\n",
       "      <td>15729.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>5374.0</td>\n",
       "      <td>2801.0</td>\n",
       "      <td>40581.0</td>\n",
       "      <td>244.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1374</th>\n",
       "      <td>fbad4165-ac84-44ed-8688-47f93fee5957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>48.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>564.0</td>\n",
       "      <td>571.0</td>\n",
       "      <td>2093.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>1327.0</td>\n",
       "      <td>550.0</td>\n",
       "      <td>7538.0</td>\n",
       "      <td>174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1375</th>\n",
       "      <td>fc774c98-c5fa-4ba8-864d-45465e55250a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>790.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>fd7704f9-fd03-4b32-b0f4-98bd0444697b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4823.0</td>\n",
       "      <td>995.0</td>\n",
       "      <td>24148.0</td>\n",
       "      <td>2352.0</td>\n",
       "      <td>33958.0</td>\n",
       "      <td>1443.0</td>\n",
       "      <td>4487.0</td>\n",
       "      <td>4459.0</td>\n",
       "      <td>111836.0</td>\n",
       "      <td>102811.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>fe25353b-d114-43f7-8d9b-eb6f2bd61d3d</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1397.0</td>\n",
       "      <td>1193.0</td>\n",
       "      <td>11654.0</td>\n",
       "      <td>318.0</td>\n",
       "      <td>25576.0</td>\n",
       "      <td>589.0</td>\n",
       "      <td>2713.0</td>\n",
       "      <td>3141.0</td>\n",
       "      <td>100850.0</td>\n",
       "      <td>63567.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1378</th>\n",
       "      <td>ff286f28-aff1-4de8-9a51-00b549fb7f86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2838.0</td>\n",
       "      <td>2497.0</td>\n",
       "      <td>65652.0</td>\n",
       "      <td>7139.0</td>\n",
       "      <td>103396.0</td>\n",
       "      <td>6868.0</td>\n",
       "      <td>15514.0</td>\n",
       "      <td>8814.0</td>\n",
       "      <td>756446.0</td>\n",
       "      <td>89644.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1379 rows × 145 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              case.case_id  ssm.importance.ABCA10  \\\n",
       "0     00807dae-9f4a-4fd1-aac2-82eb11bf2afb                    0.0   \n",
       "1     00a2d166-78c9-4687-a195-3d6315c27574                    0.0   \n",
       "2     00b11ca8-8540-4a3d-b602-ec754b00230b                    0.0   \n",
       "3     011b9b2d-ebe5-42bf-9662-d922faccc7a1                    0.0   \n",
       "4     01263518-5f7c-49dc-8d7e-84b0c03a6a63                    0.0   \n",
       "...                                    ...                    ...   \n",
       "1374  fbad4165-ac84-44ed-8688-47f93fee5957                    NaN   \n",
       "1375  fc774c98-c5fa-4ba8-864d-45465e55250a                    NaN   \n",
       "1376  fd7704f9-fd03-4b32-b0f4-98bd0444697b                    NaN   \n",
       "1377  fe25353b-d114-43f7-8d9b-eb6f2bd61d3d                    NaN   \n",
       "1378  ff286f28-aff1-4de8-9a51-00b549fb7f86                    NaN   \n",
       "\n",
       "      ssm.importance.ABCA6  ssm.importance.ABCA8  ssm.importance.ABCA9  \\\n",
       "0                      0.0                   0.0                   0.0   \n",
       "1                      0.0                   0.0                   0.0   \n",
       "2                      0.0                   0.0                   0.0   \n",
       "3                      0.0                   1.0                   0.0   \n",
       "4                      1.0                   0.0                   0.0   \n",
       "...                    ...                   ...                   ...   \n",
       "1374                   NaN                   NaN                   NaN   \n",
       "1375                   NaN                   NaN                   NaN   \n",
       "1376                   NaN                   NaN                   NaN   \n",
       "1377                   NaN                   NaN                   NaN   \n",
       "1378                   NaN                   NaN                   NaN   \n",
       "\n",
       "      ssm.importance.AMZ2  ssm.importance.ANO1  ssm.importance.ARSG  \\\n",
       "0                     0.0                  0.0                  0.0   \n",
       "1                     0.0                  0.0                  0.0   \n",
       "2                     0.0                  0.0                  0.0   \n",
       "3                     0.0                  0.0                  0.0   \n",
       "4                     0.0                  0.0                  0.0   \n",
       "...                   ...                  ...                  ...   \n",
       "1374                  NaN                  NaN                  NaN   \n",
       "1375                  NaN                  NaN                  NaN   \n",
       "1376                  NaN                  NaN                  NaN   \n",
       "1377                  NaN                  NaN                  NaN   \n",
       "1378                  NaN                  NaN                  NaN   \n",
       "\n",
       "      ssm.importance.BCAS3  ssm.importance.BPTF  ...  arn.HTseq_counts.RYR2  \\\n",
       "0                      0.0                  0.0  ...                   83.0   \n",
       "1                      0.0                  0.0  ...                   34.0   \n",
       "2                      0.0                  1.0  ...                  177.0   \n",
       "3                      0.0                  0.0  ...                  418.0   \n",
       "4                      0.0                  1.0  ...                   29.0   \n",
       "...                    ...                  ...  ...                    ...   \n",
       "1374                   NaN                  NaN  ...                   48.0   \n",
       "1375                   NaN                  NaN  ...                    0.0   \n",
       "1376                   NaN                  NaN  ...                 4823.0   \n",
       "1377                   NaN                  NaN  ...                 1397.0   \n",
       "1378                   NaN                  NaN  ...                 2838.0   \n",
       "\n",
       "      arn.HTseq_counts.SDK2  arn.HTseq_counts.SHANK2  \\\n",
       "0                      74.0                   1942.0   \n",
       "1                     266.0                   1274.0   \n",
       "2                     528.0                   2830.0   \n",
       "3                     603.0                   6459.0   \n",
       "4                      37.0                   4405.0   \n",
       "...                     ...                      ...   \n",
       "1374                  125.0                    564.0   \n",
       "1375                    0.0                      0.0   \n",
       "1376                  995.0                  24148.0   \n",
       "1377                 1193.0                  11654.0   \n",
       "1378                 2497.0                  65652.0   \n",
       "\n",
       "      arn.HTseq_counts.SLC16A6  arn.HTseq_counts.TANC2  arn.HTseq_counts.TG  \\\n",
       "0                        729.0                  3846.0                 18.0   \n",
       "1                        700.0                  9455.0                 19.0   \n",
       "2                        880.0                  5889.0                 56.0   \n",
       "3                      12415.0                 17324.0                391.0   \n",
       "4                       1289.0                 15729.0                 20.0   \n",
       "...                        ...                     ...                  ...   \n",
       "1374                     571.0                  2093.0                 77.0   \n",
       "1375                       0.0                     0.0                 34.0   \n",
       "1376                    2352.0                 33958.0               1443.0   \n",
       "1377                     318.0                 25576.0                589.0   \n",
       "1378                    7139.0                103396.0               6868.0   \n",
       "\n",
       "      arn.HTseq_counts.TP53  arn.HTseq_counts.TPCN2  arn.HTseq_counts.TRPS1  \\\n",
       "0                    2997.0                   655.0                  8320.0   \n",
       "1                    6779.0                  1123.0                 13940.0   \n",
       "2                    4543.0                  3168.0                 15170.0   \n",
       "3                    7822.0                  1206.0                212884.0   \n",
       "4                    5374.0                  2801.0                 40581.0   \n",
       "...                     ...                     ...                     ...   \n",
       "1374                 1327.0                   550.0                  7538.0   \n",
       "1375                    9.0                     0.0                   162.0   \n",
       "1376                 4487.0                  4459.0                111836.0   \n",
       "1377                 2713.0                  3141.0                100850.0   \n",
       "1378                15514.0                  8814.0                756446.0   \n",
       "\n",
       "      arn.HTseq_counts.TTN  \n",
       "0                    341.0  \n",
       "1                    306.0  \n",
       "2                   1334.0  \n",
       "3                  19495.0  \n",
       "4                    244.0  \n",
       "...                    ...  \n",
       "1374                 174.0  \n",
       "1375                 790.0  \n",
       "1376              102811.0  \n",
       "1377               63567.0  \n",
       "1378               89644.0  \n",
       "\n",
       "[1379 rows x 145 columns]"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_arn_pivot.rename(columns={'case_id': 'case.case_id'}, inplace=True)\n",
    "df = pd.merge(df, df_arn_pivot, how='outer', left_on='case.case_id', right_on='case.case_id')\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Case + ((SSM + CNV) + ARN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>demographic.age_at_index</th>\n",
       "      <th>demographic.days_to_birth</th>\n",
       "      <th>demographic.ethnicity</th>\n",
       "      <th>demographic.gender</th>\n",
       "      <th>demographic.race</th>\n",
       "      <th>demographic.vital_status</th>\n",
       "      <th>demographic.year_of_birth</th>\n",
       "      <th>diagnoses.0.age_at_diagnosis</th>\n",
       "      <th>diagnoses.0.ajcc_pathologic_m</th>\n",
       "      <th>...</th>\n",
       "      <th>arn.HTseq_counts.RYR2</th>\n",
       "      <th>arn.HTseq_counts.SDK2</th>\n",
       "      <th>arn.HTseq_counts.SHANK2</th>\n",
       "      <th>arn.HTseq_counts.SLC16A6</th>\n",
       "      <th>arn.HTseq_counts.TANC2</th>\n",
       "      <th>arn.HTseq_counts.TG</th>\n",
       "      <th>arn.HTseq_counts.TP53</th>\n",
       "      <th>arn.HTseq_counts.TPCN2</th>\n",
       "      <th>arn.HTseq_counts.TRPS1</th>\n",
       "      <th>arn.HTseq_counts.TTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>50.0</td>\n",
       "      <td>-18345.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>18345.0</td>\n",
       "      <td>M0</td>\n",
       "      <td>...</td>\n",
       "      <td>83.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>1942.0</td>\n",
       "      <td>729.0</td>\n",
       "      <td>3846.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2997.0</td>\n",
       "      <td>655.0</td>\n",
       "      <td>8320.0</td>\n",
       "      <td>341.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>56.0</td>\n",
       "      <td>-20713.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1954.0</td>\n",
       "      <td>20713.0</td>\n",
       "      <td>M0</td>\n",
       "      <td>...</td>\n",
       "      <td>34.0</td>\n",
       "      <td>266.0</td>\n",
       "      <td>1274.0</td>\n",
       "      <td>700.0</td>\n",
       "      <td>9455.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>6779.0</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>13940.0</td>\n",
       "      <td>306.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>61.0</td>\n",
       "      <td>-22497.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1951.0</td>\n",
       "      <td>22497.0</td>\n",
       "      <td>MX</td>\n",
       "      <td>...</td>\n",
       "      <td>177.0</td>\n",
       "      <td>528.0</td>\n",
       "      <td>2830.0</td>\n",
       "      <td>880.0</td>\n",
       "      <td>5889.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>4543.0</td>\n",
       "      <td>3168.0</td>\n",
       "      <td>15170.0</td>\n",
       "      <td>1334.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>71.0</td>\n",
       "      <td>-26274.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1939.0</td>\n",
       "      <td>26274.0</td>\n",
       "      <td>MX</td>\n",
       "      <td>...</td>\n",
       "      <td>418.0</td>\n",
       "      <td>603.0</td>\n",
       "      <td>6459.0</td>\n",
       "      <td>12415.0</td>\n",
       "      <td>17324.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>7822.0</td>\n",
       "      <td>1206.0</td>\n",
       "      <td>212884.0</td>\n",
       "      <td>19495.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01263518-5f7c-49dc-8d7e-84b0c03a6a63</td>\n",
       "      <td>76.0</td>\n",
       "      <td>-27942.0</td>\n",
       "      <td>not reported</td>\n",
       "      <td>female</td>\n",
       "      <td>not reported</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1933.0</td>\n",
       "      <td>27942.0</td>\n",
       "      <td>M1</td>\n",
       "      <td>...</td>\n",
       "      <td>29.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>4405.0</td>\n",
       "      <td>1289.0</td>\n",
       "      <td>15729.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>5374.0</td>\n",
       "      <td>2801.0</td>\n",
       "      <td>40581.0</td>\n",
       "      <td>244.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1374</th>\n",
       "      <td>fbad4165-ac84-44ed-8688-47f93fee5957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not reported</td>\n",
       "      <td>not reported</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>48.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>564.0</td>\n",
       "      <td>571.0</td>\n",
       "      <td>2093.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>1327.0</td>\n",
       "      <td>550.0</td>\n",
       "      <td>7538.0</td>\n",
       "      <td>174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1375</th>\n",
       "      <td>fc774c98-c5fa-4ba8-864d-45465e55250a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13822.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>790.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>fd7704f9-fd03-4b32-b0f4-98bd0444697b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13014.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4823.0</td>\n",
       "      <td>995.0</td>\n",
       "      <td>24148.0</td>\n",
       "      <td>2352.0</td>\n",
       "      <td>33958.0</td>\n",
       "      <td>1443.0</td>\n",
       "      <td>4487.0</td>\n",
       "      <td>4459.0</td>\n",
       "      <td>111836.0</td>\n",
       "      <td>102811.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>fe25353b-d114-43f7-8d9b-eb6f2bd61d3d</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12859.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1397.0</td>\n",
       "      <td>1193.0</td>\n",
       "      <td>11654.0</td>\n",
       "      <td>318.0</td>\n",
       "      <td>25576.0</td>\n",
       "      <td>589.0</td>\n",
       "      <td>2713.0</td>\n",
       "      <td>3141.0</td>\n",
       "      <td>100850.0</td>\n",
       "      <td>63567.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1378</th>\n",
       "      <td>ff286f28-aff1-4de8-9a51-00b549fb7f86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15931.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2838.0</td>\n",
       "      <td>2497.0</td>\n",
       "      <td>65652.0</td>\n",
       "      <td>7139.0</td>\n",
       "      <td>103396.0</td>\n",
       "      <td>6868.0</td>\n",
       "      <td>15514.0</td>\n",
       "      <td>8814.0</td>\n",
       "      <td>756446.0</td>\n",
       "      <td>89644.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1379 rows × 192 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  demographic.age_at_index  \\\n",
       "0     00807dae-9f4a-4fd1-aac2-82eb11bf2afb                      50.0   \n",
       "1     00a2d166-78c9-4687-a195-3d6315c27574                      56.0   \n",
       "2     00b11ca8-8540-4a3d-b602-ec754b00230b                      61.0   \n",
       "3     011b9b2d-ebe5-42bf-9662-d922faccc7a1                      71.0   \n",
       "4     01263518-5f7c-49dc-8d7e-84b0c03a6a63                      76.0   \n",
       "...                                    ...                       ...   \n",
       "1374  fbad4165-ac84-44ed-8688-47f93fee5957                       NaN   \n",
       "1375  fc774c98-c5fa-4ba8-864d-45465e55250a                       NaN   \n",
       "1376  fd7704f9-fd03-4b32-b0f4-98bd0444697b                       NaN   \n",
       "1377  fe25353b-d114-43f7-8d9b-eb6f2bd61d3d                       NaN   \n",
       "1378  ff286f28-aff1-4de8-9a51-00b549fb7f86                       NaN   \n",
       "\n",
       "      demographic.days_to_birth   demographic.ethnicity demographic.gender  \\\n",
       "0                      -18345.0  not hispanic or latino             female   \n",
       "1                      -20713.0  not hispanic or latino             female   \n",
       "2                      -22497.0  not hispanic or latino             female   \n",
       "3                      -26274.0  not hispanic or latino             female   \n",
       "4                      -27942.0            not reported             female   \n",
       "...                         ...                     ...                ...   \n",
       "1374                        NaN            not reported       not reported   \n",
       "1375                        NaN  not hispanic or latino             female   \n",
       "1376                        NaN  not hispanic or latino             female   \n",
       "1377                        NaN  not hispanic or latino             female   \n",
       "1378                        NaN  not hispanic or latino             female   \n",
       "\n",
       "     demographic.race demographic.vital_status  demographic.year_of_birth  \\\n",
       "0               white                    Alive                     1955.0   \n",
       "1               white                    Alive                     1954.0   \n",
       "2               white                    Alive                     1951.0   \n",
       "3               white                    Alive                     1939.0   \n",
       "4        not reported                    Alive                     1933.0   \n",
       "...               ...                      ...                        ...   \n",
       "1374            white             Not Reported                        NaN   \n",
       "1375            white             Not Reported                        NaN   \n",
       "1376            white             Not Reported                        NaN   \n",
       "1377            white             Not Reported                        NaN   \n",
       "1378          Unknown             Not Reported                        NaN   \n",
       "\n",
       "      diagnoses.0.age_at_diagnosis diagnoses.0.ajcc_pathologic_m  ...  \\\n",
       "0                          18345.0                            M0  ...   \n",
       "1                          20713.0                            M0  ...   \n",
       "2                          22497.0                            MX  ...   \n",
       "3                          26274.0                            MX  ...   \n",
       "4                          27942.0                            M1  ...   \n",
       "...                            ...                           ...  ...   \n",
       "1374                           NaN                           NaN  ...   \n",
       "1375                       13822.0                           NaN  ...   \n",
       "1376                       13014.0                           NaN  ...   \n",
       "1377                       12859.0                           NaN  ...   \n",
       "1378                       15931.0                           NaN  ...   \n",
       "\n",
       "     arn.HTseq_counts.RYR2 arn.HTseq_counts.SDK2 arn.HTseq_counts.SHANK2  \\\n",
       "0                     83.0                  74.0                  1942.0   \n",
       "1                     34.0                 266.0                  1274.0   \n",
       "2                    177.0                 528.0                  2830.0   \n",
       "3                    418.0                 603.0                  6459.0   \n",
       "4                     29.0                  37.0                  4405.0   \n",
       "...                    ...                   ...                     ...   \n",
       "1374                  48.0                 125.0                   564.0   \n",
       "1375                   0.0                   0.0                     0.0   \n",
       "1376                4823.0                 995.0                 24148.0   \n",
       "1377                1397.0                1193.0                 11654.0   \n",
       "1378                2838.0                2497.0                 65652.0   \n",
       "\n",
       "     arn.HTseq_counts.SLC16A6 arn.HTseq_counts.TANC2  arn.HTseq_counts.TG  \\\n",
       "0                       729.0                 3846.0                 18.0   \n",
       "1                       700.0                 9455.0                 19.0   \n",
       "2                       880.0                 5889.0                 56.0   \n",
       "3                     12415.0                17324.0                391.0   \n",
       "4                      1289.0                15729.0                 20.0   \n",
       "...                       ...                    ...                  ...   \n",
       "1374                    571.0                 2093.0                 77.0   \n",
       "1375                      0.0                    0.0                 34.0   \n",
       "1376                   2352.0                33958.0               1443.0   \n",
       "1377                    318.0                25576.0                589.0   \n",
       "1378                   7139.0               103396.0               6868.0   \n",
       "\n",
       "      arn.HTseq_counts.TP53 arn.HTseq_counts.TPCN2 arn.HTseq_counts.TRPS1  \\\n",
       "0                    2997.0                  655.0                 8320.0   \n",
       "1                    6779.0                 1123.0                13940.0   \n",
       "2                    4543.0                 3168.0                15170.0   \n",
       "3                    7822.0                 1206.0               212884.0   \n",
       "4                    5374.0                 2801.0                40581.0   \n",
       "...                     ...                    ...                    ...   \n",
       "1374                 1327.0                  550.0                 7538.0   \n",
       "1375                    9.0                    0.0                  162.0   \n",
       "1376                 4487.0                 4459.0               111836.0   \n",
       "1377                 2713.0                 3141.0               100850.0   \n",
       "1378                15514.0                 8814.0               756446.0   \n",
       "\n",
       "     arn.HTseq_counts.TTN  \n",
       "0                   341.0  \n",
       "1                   306.0  \n",
       "2                  1334.0  \n",
       "3                 19495.0  \n",
       "4                   244.0  \n",
       "...                   ...  \n",
       "1374                174.0  \n",
       "1375                790.0  \n",
       "1376             102811.0  \n",
       "1377              63567.0  \n",
       "1378              89644.0  \n",
       "\n",
       "[1379 rows x 192 columns]"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.rename(columns={'case.case_id': 'case_id'}, inplace=True)\n",
    "df = pd.merge(df_case, df, how='right', left_on='case_id', right_on='case_id')\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One can check the columns and assess there is (almost) everything:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case_id\n",
      "demographic.age_at_index\n",
      "demographic.days_to_birth\n",
      "demographic.ethnicity\n",
      "demographic.gender\n",
      "demographic.race\n",
      "demographic.vital_status\n",
      "demographic.year_of_birth\n",
      "diagnoses.0.age_at_diagnosis\n",
      "diagnoses.0.ajcc_pathologic_m\n",
      "diagnoses.0.ajcc_pathologic_n\n",
      "diagnoses.0.ajcc_pathologic_stage\n",
      "diagnoses.0.ajcc_pathologic_t\n",
      "diagnoses.0.ajcc_staging_system_edition\n",
      "diagnoses.0.classification_of_tumor\n",
      "diagnoses.0.days_to_diagnosis\n",
      "diagnoses.0.days_to_last_follow_up\n",
      "diagnoses.0.last_known_disease_status\n",
      "diagnoses.0.morphology\n",
      "diagnoses.0.primary_diagnosis\n",
      "diagnoses.0.prior_malignancy\n",
      "diagnoses.0.prior_treatment\n",
      "diagnoses.0.progression_or_recurrence\n",
      "diagnoses.0.site_of_resection_or_biopsy\n",
      "diagnoses.0.synchronous_malignancy\n",
      "diagnoses.0.tissue_or_organ_of_origin\n",
      "diagnoses.0.treatments.0.treatment_or_therapy\n",
      "diagnoses.0.treatments.0.treatment_type\n",
      "diagnoses.0.treatments.1.treatment_or_therapy\n",
      "diagnoses.0.treatments.1.treatment_type\n",
      "diagnoses.0.tumor_grade\n",
      "diagnoses.0.tumor_stage\n",
      "diagnoses.0.year_of_diagnosis\n",
      "exposures.0.alcohol_history\n",
      "id\n",
      "samples.0.is_ffpe\n",
      "samples.0.oct_embedded\n",
      "samples.0.sample_type\n",
      "samples.0.tissue_type\n",
      "samples.0.tumor_descriptor\n",
      "samples.1.is_ffpe\n",
      "samples.1.oct_embedded\n",
      "samples.1.sample_type\n",
      "samples.1.tissue_type\n",
      "samples.2.is_ffpe\n",
      "samples.2.oct_embedded\n",
      "samples.2.sample_type\n",
      "samples.2.tissue_type\n",
      "ssm.importance.ABCA10\n",
      "ssm.importance.ABCA6\n",
      "ssm.importance.ABCA8\n",
      "ssm.importance.ABCA9\n",
      "ssm.importance.AMZ2\n",
      "ssm.importance.ANO1\n",
      "ssm.importance.ARSG\n",
      "ssm.importance.BCAS3\n",
      "ssm.importance.BPTF\n",
      "ssm.importance.CCND1\n",
      "ssm.importance.CEP112\n",
      "ssm.importance.CPT1A\n",
      "ssm.importance.CSMD3\n",
      "ssm.importance.CTTN\n",
      "ssm.importance.DNAH17\n",
      "ssm.importance.FGF3\n",
      "ssm.importance.FGF4\n",
      "ssm.importance.FNTA\n",
      "ssm.importance.HELZ\n",
      "ssm.importance.IGHMBP2\n",
      "ssm.importance.KPNA2\n",
      "ssm.importance.MRGPRD\n",
      "ssm.importance.MRGPRF\n",
      "ssm.importance.MUC4\n",
      "ssm.importance.MYC\n",
      "ssm.importance.MYEOV\n",
      "ssm.importance.NOL11\n",
      "ssm.importance.NOTCH2\n",
      "ssm.importance.ORAOV1\n",
      "ssm.importance.PIK3CA\n",
      "ssm.importance.PITPNC1\n",
      "ssm.importance.POU5F1B\n",
      "ssm.importance.PPFIA1\n",
      "ssm.importance.PRKAR1A\n",
      "ssm.importance.PRKCA\n",
      "ssm.importance.RNF213\n",
      "ssm.importance.RYR2\n",
      "ssm.importance.SDK2\n",
      "ssm.importance.SHANK2\n",
      "ssm.importance.SLC16A6\n",
      "ssm.importance.TANC2\n",
      "ssm.importance.TG\n",
      "ssm.importance.TP53\n",
      "ssm.importance.TPCN2\n",
      "ssm.importance.TRPS1\n",
      "ssm.importance.TTN\n",
      "cnv.change.ABCA10\n",
      "cnv.change.ABCA6\n",
      "cnv.change.ABCA8\n",
      "cnv.change.ABCA9\n",
      "cnv.change.AC108925.1\n",
      "cnv.change.AMZ2\n",
      "cnv.change.ANO1\n",
      "cnv.change.AP001888.1\n",
      "cnv.change.ARSG\n",
      "cnv.change.BCAS3\n",
      "cnv.change.BPTF\n",
      "cnv.change.CCND1\n",
      "cnv.change.CEP112\n",
      "cnv.change.CPT1A\n",
      "cnv.change.CSMD3\n",
      "cnv.change.CTTN\n",
      "cnv.change.DNAH17\n",
      "cnv.change.FADD\n",
      "cnv.change.FGF19\n",
      "cnv.change.FGF3\n",
      "cnv.change.FGF4\n",
      "cnv.change.FNTA\n",
      "cnv.change.HELZ\n",
      "cnv.change.IGHMBP2\n",
      "cnv.change.KPNA2\n",
      "cnv.change.MRGPRD\n",
      "cnv.change.MRGPRF\n",
      "cnv.change.MUC4\n",
      "cnv.change.MYC\n",
      "cnv.change.MYEOV\n",
      "cnv.change.NOL11\n",
      "cnv.change.NOTCH2\n",
      "cnv.change.ORAOV1\n",
      "cnv.change.PIK3CA\n",
      "cnv.change.PITPNC1\n",
      "cnv.change.POU5F1B\n",
      "cnv.change.PPFIA1\n",
      "cnv.change.PRKAR1A\n",
      "cnv.change.PRKCA\n",
      "cnv.change.RNF213\n",
      "cnv.change.RYR2\n",
      "cnv.change.SDK2\n",
      "cnv.change.SHANK2\n",
      "cnv.change.SLC16A6\n",
      "cnv.change.TANC2\n",
      "cnv.change.TG\n",
      "cnv.change.TP53\n",
      "cnv.change.TPCN2\n",
      "cnv.change.TRPS1\n",
      "cnv.change.TTN\n",
      "arn.HTseq_counts.ABCA10\n",
      "arn.HTseq_counts.ABCA6\n",
      "arn.HTseq_counts.ABCA8\n",
      "arn.HTseq_counts.ABCA9\n",
      "arn.HTseq_counts.AMZ2\n",
      "arn.HTseq_counts.ANO1\n",
      "arn.HTseq_counts.ARSG\n",
      "arn.HTseq_counts.BCAS3\n",
      "arn.HTseq_counts.BPTF\n",
      "arn.HTseq_counts.CCND1\n",
      "arn.HTseq_counts.CEP112\n",
      "arn.HTseq_counts.CPT1A\n",
      "arn.HTseq_counts.CSMD3\n",
      "arn.HTseq_counts.CTTN\n",
      "arn.HTseq_counts.DNAH17\n",
      "arn.HTseq_counts.DNAJB6P5\n",
      "arn.HTseq_counts.FADD\n",
      "arn.HTseq_counts.FGF19\n",
      "arn.HTseq_counts.FGF3\n",
      "arn.HTseq_counts.FGF4\n",
      "arn.HTseq_counts.FNTA\n",
      "arn.HTseq_counts.HELZ\n",
      "arn.HTseq_counts.IGHMBP2\n",
      "arn.HTseq_counts.KPNA2\n",
      "arn.HTseq_counts.MRGPRD\n",
      "arn.HTseq_counts.MRGPRF\n",
      "arn.HTseq_counts.MUC4\n",
      "arn.HTseq_counts.MYC\n",
      "arn.HTseq_counts.MYEOV\n",
      "arn.HTseq_counts.NOL11\n",
      "arn.HTseq_counts.NOTCH2\n",
      "arn.HTseq_counts.PIK3CA\n",
      "arn.HTseq_counts.PITPNC1\n",
      "arn.HTseq_counts.POU5F1B\n",
      "arn.HTseq_counts.PPFIA1\n",
      "arn.HTseq_counts.PRKAR1A\n",
      "arn.HTseq_counts.PRKCA\n",
      "arn.HTseq_counts.RNF213\n",
      "arn.HTseq_counts.RYR2\n",
      "arn.HTseq_counts.SDK2\n",
      "arn.HTseq_counts.SHANK2\n",
      "arn.HTseq_counts.SLC16A6\n",
      "arn.HTseq_counts.TANC2\n",
      "arn.HTseq_counts.TG\n",
      "arn.HTseq_counts.TP53\n",
      "arn.HTseq_counts.TPCN2\n",
      "arn.HTseq_counts.TRPS1\n",
      "arn.HTseq_counts.TTN\n"
     ]
    }
   ],
   "source": [
    "for col in df.columns:\n",
    "    print(col)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We may have forgotten some genes in the pivotings that were not involved in a particular dataset (CNV, SSM or ARN)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = list(df_gene[\"symbol\"].sort_values().to_numpy())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "CNV contains all the 50 genes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "50"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_cnv = [col[11:] for col in df if col.startswith('cnv.change.')]\n",
    "len(genes_cnv)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Missing genes in SSM:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'AC108925.1', 'FADD', 'FGF19', 'DNAJB6P5'}\n"
     ]
    }
   ],
   "source": [
    "genes_ssm = [col[15:] for col in df if col.startswith('ssm.importance.')]\n",
    "ssm_missing = set(genes) - set(genes_ssm)\n",
    "print(ssm_missing)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Missing genes in ARN:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'AC108925.1', 'ORAOV1'}\n"
     ]
    }
   ],
   "source": [
    "genes_arn = [col[17:] for col in df if col.startswith('arn.HTseq_counts.')]\n",
    "arn_missing = set(genes) - set(genes_arn)\n",
    "print(arn_missing)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Add the missing columns:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [],
   "source": [
    "for gene in ssm_missing:\n",
    "    col = 'ssm.importance.' + gene\n",
    "    df[col] = 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "for gene in arn_missing:\n",
    "    col = 'arn.HTseq_counts.' + gene\n",
    "    df[col] = 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"./data/gdc_final_database.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>demographic.age_at_index</th>\n",
       "      <th>demographic.days_to_birth</th>\n",
       "      <th>demographic.ethnicity</th>\n",
       "      <th>demographic.gender</th>\n",
       "      <th>demographic.race</th>\n",
       "      <th>demographic.vital_status</th>\n",
       "      <th>demographic.year_of_birth</th>\n",
       "      <th>diagnoses.0.age_at_diagnosis</th>\n",
       "      <th>diagnoses.0.ajcc_pathologic_m</th>\n",
       "      <th>...</th>\n",
       "      <th>arn.HTseq_counts.TP53</th>\n",
       "      <th>arn.HTseq_counts.TPCN2</th>\n",
       "      <th>arn.HTseq_counts.TRPS1</th>\n",
       "      <th>arn.HTseq_counts.TTN</th>\n",
       "      <th>ssm.importance.AC108925.1</th>\n",
       "      <th>ssm.importance.FADD</th>\n",
       "      <th>ssm.importance.FGF19</th>\n",
       "      <th>ssm.importance.DNAJB6P5</th>\n",
       "      <th>arn.HTseq_counts.AC108925.1</th>\n",
       "      <th>arn.HTseq_counts.ORAOV1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>50.0</td>\n",
       "      <td>-18345.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>18345.0</td>\n",
       "      <td>M0</td>\n",
       "      <td>...</td>\n",
       "      <td>2997.0</td>\n",
       "      <td>655.0</td>\n",
       "      <td>8320.0</td>\n",
       "      <td>341.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>56.0</td>\n",
       "      <td>-20713.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1954.0</td>\n",
       "      <td>20713.0</td>\n",
       "      <td>M0</td>\n",
       "      <td>...</td>\n",
       "      <td>6779.0</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>13940.0</td>\n",
       "      <td>306.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00b11ca8-8540-4a3d-b602-ec754b00230b</td>\n",
       "      <td>61.0</td>\n",
       "      <td>-22497.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1951.0</td>\n",
       "      <td>22497.0</td>\n",
       "      <td>MX</td>\n",
       "      <td>...</td>\n",
       "      <td>4543.0</td>\n",
       "      <td>3168.0</td>\n",
       "      <td>15170.0</td>\n",
       "      <td>1334.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>011b9b2d-ebe5-42bf-9662-d922faccc7a1</td>\n",
       "      <td>71.0</td>\n",
       "      <td>-26274.0</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1939.0</td>\n",
       "      <td>26274.0</td>\n",
       "      <td>MX</td>\n",
       "      <td>...</td>\n",
       "      <td>7822.0</td>\n",
       "      <td>1206.0</td>\n",
       "      <td>212884.0</td>\n",
       "      <td>19495.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01263518-5f7c-49dc-8d7e-84b0c03a6a63</td>\n",
       "      <td>76.0</td>\n",
       "      <td>-27942.0</td>\n",
       "      <td>not reported</td>\n",
       "      <td>female</td>\n",
       "      <td>not reported</td>\n",
       "      <td>Alive</td>\n",
       "      <td>1933.0</td>\n",
       "      <td>27942.0</td>\n",
       "      <td>M1</td>\n",
       "      <td>...</td>\n",
       "      <td>5374.0</td>\n",
       "      <td>2801.0</td>\n",
       "      <td>40581.0</td>\n",
       "      <td>244.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1374</th>\n",
       "      <td>fbad4165-ac84-44ed-8688-47f93fee5957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not reported</td>\n",
       "      <td>not reported</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1327.0</td>\n",
       "      <td>550.0</td>\n",
       "      <td>7538.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1375</th>\n",
       "      <td>fc774c98-c5fa-4ba8-864d-45465e55250a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13822.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>790.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>fd7704f9-fd03-4b32-b0f4-98bd0444697b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13014.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4487.0</td>\n",
       "      <td>4459.0</td>\n",
       "      <td>111836.0</td>\n",
       "      <td>102811.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>fe25353b-d114-43f7-8d9b-eb6f2bd61d3d</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>white</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12859.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2713.0</td>\n",
       "      <td>3141.0</td>\n",
       "      <td>100850.0</td>\n",
       "      <td>63567.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1378</th>\n",
       "      <td>ff286f28-aff1-4de8-9a51-00b549fb7f86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15931.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>15514.0</td>\n",
       "      <td>8814.0</td>\n",
       "      <td>756446.0</td>\n",
       "      <td>89644.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1379 rows × 198 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  demographic.age_at_index  \\\n",
       "0     00807dae-9f4a-4fd1-aac2-82eb11bf2afb                      50.0   \n",
       "1     00a2d166-78c9-4687-a195-3d6315c27574                      56.0   \n",
       "2     00b11ca8-8540-4a3d-b602-ec754b00230b                      61.0   \n",
       "3     011b9b2d-ebe5-42bf-9662-d922faccc7a1                      71.0   \n",
       "4     01263518-5f7c-49dc-8d7e-84b0c03a6a63                      76.0   \n",
       "...                                    ...                       ...   \n",
       "1374  fbad4165-ac84-44ed-8688-47f93fee5957                       NaN   \n",
       "1375  fc774c98-c5fa-4ba8-864d-45465e55250a                       NaN   \n",
       "1376  fd7704f9-fd03-4b32-b0f4-98bd0444697b                       NaN   \n",
       "1377  fe25353b-d114-43f7-8d9b-eb6f2bd61d3d                       NaN   \n",
       "1378  ff286f28-aff1-4de8-9a51-00b549fb7f86                       NaN   \n",
       "\n",
       "      demographic.days_to_birth   demographic.ethnicity demographic.gender  \\\n",
       "0                      -18345.0  not hispanic or latino             female   \n",
       "1                      -20713.0  not hispanic or latino             female   \n",
       "2                      -22497.0  not hispanic or latino             female   \n",
       "3                      -26274.0  not hispanic or latino             female   \n",
       "4                      -27942.0            not reported             female   \n",
       "...                         ...                     ...                ...   \n",
       "1374                        NaN            not reported       not reported   \n",
       "1375                        NaN  not hispanic or latino             female   \n",
       "1376                        NaN  not hispanic or latino             female   \n",
       "1377                        NaN  not hispanic or latino             female   \n",
       "1378                        NaN  not hispanic or latino             female   \n",
       "\n",
       "     demographic.race demographic.vital_status  demographic.year_of_birth  \\\n",
       "0               white                    Alive                     1955.0   \n",
       "1               white                    Alive                     1954.0   \n",
       "2               white                    Alive                     1951.0   \n",
       "3               white                    Alive                     1939.0   \n",
       "4        not reported                    Alive                     1933.0   \n",
       "...               ...                      ...                        ...   \n",
       "1374            white             Not Reported                        NaN   \n",
       "1375            white             Not Reported                        NaN   \n",
       "1376            white             Not Reported                        NaN   \n",
       "1377            white             Not Reported                        NaN   \n",
       "1378          Unknown             Not Reported                        NaN   \n",
       "\n",
       "      diagnoses.0.age_at_diagnosis diagnoses.0.ajcc_pathologic_m  ...  \\\n",
       "0                          18345.0                            M0  ...   \n",
       "1                          20713.0                            M0  ...   \n",
       "2                          22497.0                            MX  ...   \n",
       "3                          26274.0                            MX  ...   \n",
       "4                          27942.0                            M1  ...   \n",
       "...                            ...                           ...  ...   \n",
       "1374                           NaN                           NaN  ...   \n",
       "1375                       13822.0                           NaN  ...   \n",
       "1376                       13014.0                           NaN  ...   \n",
       "1377                       12859.0                           NaN  ...   \n",
       "1378                       15931.0                           NaN  ...   \n",
       "\n",
       "     arn.HTseq_counts.TP53 arn.HTseq_counts.TPCN2 arn.HTseq_counts.TRPS1  \\\n",
       "0                   2997.0                  655.0                 8320.0   \n",
       "1                   6779.0                 1123.0                13940.0   \n",
       "2                   4543.0                 3168.0                15170.0   \n",
       "3                   7822.0                 1206.0               212884.0   \n",
       "4                   5374.0                 2801.0                40581.0   \n",
       "...                    ...                    ...                    ...   \n",
       "1374                1327.0                  550.0                 7538.0   \n",
       "1375                   9.0                    0.0                  162.0   \n",
       "1376                4487.0                 4459.0               111836.0   \n",
       "1377                2713.0                 3141.0               100850.0   \n",
       "1378               15514.0                 8814.0               756446.0   \n",
       "\n",
       "     arn.HTseq_counts.TTN ssm.importance.AC108925.1  ssm.importance.FADD  \\\n",
       "0                   341.0                       0.0                  0.0   \n",
       "1                   306.0                       0.0                  0.0   \n",
       "2                  1334.0                       0.0                  0.0   \n",
       "3                 19495.0                       0.0                  0.0   \n",
       "4                   244.0                       0.0                  0.0   \n",
       "...                   ...                       ...                  ...   \n",
       "1374                174.0                       0.0                  0.0   \n",
       "1375                790.0                       0.0                  0.0   \n",
       "1376             102811.0                       0.0                  0.0   \n",
       "1377              63567.0                       0.0                  0.0   \n",
       "1378              89644.0                       0.0                  0.0   \n",
       "\n",
       "      ssm.importance.FGF19 ssm.importance.DNAJB6P5  \\\n",
       "0                      0.0                     0.0   \n",
       "1                      0.0                     0.0   \n",
       "2                      0.0                     0.0   \n",
       "3                      0.0                     0.0   \n",
       "4                      0.0                     0.0   \n",
       "...                    ...                     ...   \n",
       "1374                   0.0                     0.0   \n",
       "1375                   0.0                     0.0   \n",
       "1376                   0.0                     0.0   \n",
       "1377                   0.0                     0.0   \n",
       "1378                   0.0                     0.0   \n",
       "\n",
       "     arn.HTseq_counts.AC108925.1 arn.HTseq_counts.ORAOV1  \n",
       "0                            0.0                     0.0  \n",
       "1                            0.0                     0.0  \n",
       "2                            0.0                     0.0  \n",
       "3                            0.0                     0.0  \n",
       "4                            0.0                     0.0  \n",
       "...                          ...                     ...  \n",
       "1374                         0.0                     0.0  \n",
       "1375                         0.0                     0.0  \n",
       "1376                         0.0                     0.0  \n",
       "1377                         0.0                     0.0  \n",
       "1378                         0.0                     0.0  \n",
       "\n",
       "[1379 rows x 198 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
