{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cancer prediction - starting_kit\n",
    "This notebook can be consulted as an HTML page following this link.\n",
    "\n",
    "\n",
    "This notebook aims at defining the theoretical understanding of cancer prediction and its correlation with genes. \n",
    "\n",
    "In addition to the medical knowldege, the data set which will be used will also be presented and illustrated. \n",
    "\n",
    "Last, an initial machine learning example will be provided in order to initiate your contribution to the challenge by proving you with an basic working solution.\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In 2018 there have been 18million new cases of cancer and 10 millions death. About 20% of men will have a cancer and 17% of the woman. 1 man over 8 will die of cancer and 1 woman over 11.\n",
    "Living with a cancer five year after its diagnosis is called prevalence. 50 million in the world are in this case.\n",
    "\n",
    "The rise of cancer in our societies has multiple origins : demographic growth, the ageing of the population, social and economical factors. The type of cancers shifts with the development of countries from cancers linked to poverty to cancers associated with industrialized way of life. \n",
    "\n",
    "Europe concentrates 23.4% of the cancers and 20.3% of the death due to cancer even though it only represents 9% of the worldwide population. United States and Canada together count for 21% of the cancer cases, 14% of the death with 13.3% of the population. \n",
    "\n",
    "\n",
    "A cancer is caused by a mutation of the DNA, thus, the cellâ€™s reproduction becomes out of control. Many mutation can have the same effect, resulting in many different types of cancer. That is why cancer cannot can be cured as simply as other pathologies. \n",
    "\n",
    "One of the major aspects of treating a cancer is detecting is, knowing which mutation of the DNA it is, what are the specific consequences of those mutations on the cells which then gives the doctor an indication about how to treat it. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_case = pd.read_csv('data/case.tsv', sep='\\t')\n",
    "df_cnv = pd.read_csv('data/cnv.tsv', sep='\\t')\n",
    "df_gene = pd.read_csv('data/gene.tsv', sep='\\t')\n",
    "df_ssm = pd.read_csv('data/ssm.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: case\n",
    "\n",
    "Information on each case such as age, alcohol, tobacco, BMI (Body Mass Index), most of the values are not availale (NaN)\n",
    "\n",
    "36 columns and 9115 rows.\n",
    "\n",
    "Possible key: 'case_id' or 'id'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case.case_id\n",
      "id\n",
      "ssm.chromosome\n",
      "ssm.consequence.0.consequence_id\n",
      "ssm.consequence.0.transcript.gene.symbol\n",
      "ssm.consequence.1.consequence_id\n",
      "ssm.consequence.1.transcript.gene.symbol\n",
      "ssm.consequence.10.consequence_id\n",
      "ssm.consequence.10.transcript.gene.symbol\n",
      "ssm.consequence.11.consequence_id\n",
      "ssm.consequence.11.transcript.gene.symbol\n",
      "ssm.consequence.12.consequence_id\n",
      "ssm.consequence.12.transcript.gene.symbol\n",
      "ssm.consequence.13.consequence_id\n",
      "ssm.consequence.13.transcript.gene.symbol\n",
      "ssm.consequence.14.consequence_id\n",
      "ssm.consequence.14.transcript.gene.symbol\n",
      "ssm.consequence.15.consequence_id\n",
      "ssm.consequence.15.transcript.gene.symbol\n",
      "ssm.consequence.16.consequence_id\n",
      "ssm.consequence.16.transcript.gene.symbol\n",
      "ssm.consequence.17.consequence_id\n",
      "ssm.consequence.17.transcript.gene.symbol\n",
      "ssm.consequence.18.consequence_id\n",
      "ssm.consequence.18.transcript.gene.symbol\n",
      "ssm.consequence.19.consequence_id\n",
      "ssm.consequence.19.transcript.gene.symbol\n",
      "ssm.consequence.2.consequence_id\n",
      "ssm.consequence.2.transcript.gene.symbol\n",
      "ssm.consequence.20.consequence_id\n",
      "ssm.consequence.20.transcript.gene.symbol\n",
      "ssm.consequence.21.consequence_id\n",
      "ssm.consequence.21.transcript.gene.symbol\n",
      "ssm.consequence.22.consequence_id\n",
      "ssm.consequence.22.transcript.gene.symbol\n",
      "ssm.consequence.23.consequence_id\n",
      "ssm.consequence.23.transcript.gene.symbol\n",
      "ssm.consequence.24.consequence_id\n",
      "ssm.consequence.24.transcript.gene.symbol\n",
      "ssm.consequence.25.consequence_id\n",
      "ssm.consequence.25.transcript.gene.symbol\n",
      "ssm.consequence.26.consequence_id\n",
      "ssm.consequence.26.transcript.gene.symbol\n",
      "ssm.consequence.3.consequence_id\n",
      "ssm.consequence.3.transcript.gene.symbol\n",
      "ssm.consequence.4.consequence_id\n",
      "ssm.consequence.4.transcript.gene.symbol\n",
      "ssm.consequence.5.consequence_id\n",
      "ssm.consequence.5.transcript.gene.symbol\n",
      "ssm.consequence.6.consequence_id\n",
      "ssm.consequence.6.transcript.gene.symbol\n",
      "ssm.consequence.7.consequence_id\n",
      "ssm.consequence.7.transcript.gene.symbol\n",
      "ssm.consequence.8.consequence_id\n",
      "ssm.consequence.8.transcript.gene.symbol\n",
      "ssm.consequence.9.consequence_id\n",
      "ssm.consequence.9.transcript.gene.symbol\n",
      "ssm.cosmic_id.0\n",
      "ssm.cosmic_id.1\n",
      "ssm.cosmic_id.2\n",
      "ssm.cosmic_id.3\n",
      "ssm.cosmic_id.4\n",
      "ssm.cosmic_id.5\n",
      "ssm.cosmic_id.6\n",
      "ssm.cosmic_id.7\n",
      "ssm.end_position\n",
      "ssm.genomic_dna_change\n",
      "ssm.mutation_subtype\n",
      "ssm.mutation_type\n",
      "ssm.ncbi_build\n",
      "ssm.reference_allele\n",
      "ssm.ssm_id\n",
      "ssm.start_position\n",
      "ssm.tumor_allele\n"
     ]
    }
   ],
   "source": [
    "df_ssm.dtypes\n",
    "for col in df_ssm.columns: \n",
    "    print(col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotations.0.annotation_id \t 0.003181568842567197\n",
      "annotations.0.case_id \t 0.003181568842567197\n",
      "annotations.0.category \t 0.003181568842567197\n",
      "annotations.0.classification \t 0.003181568842567197\n",
      "annotations.0.created_datetime \t 0.003181568842567197\n",
      "annotations.0.entity_id \t 0.003181568842567197\n",
      "annotations.0.entity_submitter_id \t 0.003181568842567197\n",
      "annotations.0.entity_type \t 0.003181568842567197\n",
      "annotations.0.notes \t 0.003181568842567197\n",
      "annotations.0.state \t 0.003181568842567197\n",
      "annotations.0.status \t 0.003181568842567197\n",
      "annotations.0.submitter_id \t 0.003181568842567197\n",
      "annotations.0.updated_datetime \t 0.003181568842567197\n",
      "annotations.1.annotation_id \t 0.0006582556226001097\n",
      "annotations.1.case_id \t 0.0006582556226001097\n",
      "annotations.1.category \t 0.0006582556226001097\n",
      "annotations.1.classification \t 0.0006582556226001097\n",
      "annotations.1.created_datetime \t 0.0006582556226001097\n",
      "annotations.1.entity_id \t 0.0006582556226001097\n",
      "annotations.1.entity_submitter_id \t 0.0006582556226001097\n",
      "annotations.1.entity_type \t 0.0006582556226001097\n",
      "annotations.1.notes \t 0.0006582556226001097\n",
      "annotations.1.state \t 0.0006582556226001097\n",
      "annotations.1.status \t 0.0006582556226001097\n",
      "annotations.1.submitter_id \t 0.0006582556226001097\n",
      "annotations.1.updated_datetime \t 0.0006582556226001097\n",
      "case_id \t 1.0\n",
      "demographic.age_at_index \t 0.6760285244103127\n",
      "demographic.age_is_obfuscated \t 0.00010970927043335162\n",
      "demographic.cause_of_death \t 0.00010970927043335162\n",
      "demographic.cause_of_death_source \t 0.0\n",
      "demographic.country_of_residence_at_enrollment \t 0.0\n",
      "demographic.created_datetime \t 0.8794295117937466\n",
      "demographic.days_to_birth \t 0.6744925946242457\n",
      "demographic.days_to_death \t 0.016675809105869445\n",
      "demographic.demographic_id \t 0.9998902907295667\n",
      "demographic.ethnicity \t 0.9998902907295667\n",
      "demographic.gender \t 0.9998902907295667\n",
      "demographic.occupation_duration_years \t 0.0\n",
      "demographic.premature_at_birth \t 0.0\n",
      "demographic.race \t 0.9998902907295667\n",
      "demographic.state \t 0.9998902907295667\n",
      "demographic.submitter_id \t 0.9998902907295667\n",
      "demographic.updated_datetime \t 0.9998902907295667\n",
      "demographic.vital_status \t 0.9998902907295667\n",
      "demographic.weeks_gestation_at_birth \t 0.0\n",
      "demographic.year_of_birth \t 0.12024136039495337\n",
      "demographic.year_of_death \t 0.011409764125068568\n",
      "diagnoses.0.age_at_diagnosis \t 0.4365331870543061\n",
      "diagnoses.0.ajcc_clinical_m \t 0.0\n",
      "diagnoses.0.ajcc_clinical_n \t 0.0\n",
      "diagnoses.0.ajcc_clinical_stage \t 0.00010970927043335162\n",
      "diagnoses.0.ajcc_clinical_t \t 0.0\n",
      "diagnoses.0.ajcc_pathologic_m \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_n \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_stage \t 0.13395501919912234\n",
      "diagnoses.0.ajcc_pathologic_t \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_staging_system_edition \t 0.1049917718047175\n",
      "diagnoses.0.anaplasia_present \t 0.0\n",
      "diagnoses.0.anaplasia_present_type \t 0.0\n",
      "diagnoses.0.ann_arbor_b_symptoms \t 0.00010970927043335162\n",
      "diagnoses.0.ann_arbor_clinical_stage \t 0.00010970927043335162\n",
      "diagnoses.0.ann_arbor_extranodal_involvement \t 0.00010970927043335162\n",
      "diagnoses.0.ann_arbor_pathologic_stage \t 0.0\n",
      "diagnoses.0.best_overall_response \t 0.0\n",
      "diagnoses.0.breslow_thickness \t 0.0\n",
      "diagnoses.0.burkitt_lymphoma_clinical_variant \t 0.0\n",
      "diagnoses.0.child_pugh_classification \t 0.0\n",
      "diagnoses.0.circumferential_resection_margin \t 0.0\n",
      "diagnoses.0.classification_of_tumor \t 0.4185408667032364\n",
      "diagnoses.0.cog_liver_stage \t 0.0\n",
      "diagnoses.0.cog_neuroblastoma_risk_group \t 0.0\n",
      "diagnoses.0.cog_renal_stage \t 0.0\n",
      "diagnoses.0.cog_rhabdomyosarcoma_risk_group \t 0.0\n",
      "diagnoses.0.created_datetime \t 0.8794295117937466\n",
      "diagnoses.0.days_to_best_overall_response \t 0.0\n",
      "diagnoses.0.days_to_diagnosis \t 0.12057048820625343\n",
      "diagnoses.0.days_to_last_follow_up \t 0.10905101481075151\n",
      "diagnoses.0.days_to_last_known_disease_status \t 0.0\n",
      "diagnoses.0.days_to_recurrence \t 0.0\n",
      "diagnoses.0.diagnosis_id \t 0.9998902907295667\n",
      "diagnoses.0.enneking_msts_grade \t 0.0\n",
      "diagnoses.0.enneking_msts_metastasis \t 0.0\n",
      "diagnoses.0.enneking_msts_stage \t 0.0\n",
      "diagnoses.0.enneking_msts_tumor_site \t 0.0\n",
      "diagnoses.0.esophageal_columnar_dysplasia_degree \t 0.0\n",
      "diagnoses.0.esophageal_columnar_metaplasia_present \t 0.0\n",
      "diagnoses.0.figo_stage \t 0.014701042238069117\n",
      "diagnoses.0.figo_staging_edition_year \t 0.0\n",
      "diagnoses.0.first_symptom_prior_to_diagnosis \t 0.0\n",
      "diagnoses.0.gastric_esophageal_junction_involvement \t 0.0\n",
      "diagnoses.0.gleason_grade_group \t 0.0\n",
      "diagnoses.0.gleason_grade_tertiary \t 0.0\n",
      "diagnoses.0.gleason_patterns_percent \t 0.0\n",
      "diagnoses.0.goblet_cells_columnar_mucosa_present \t 0.0\n",
      "diagnoses.0.greatest_tumor_dimension \t 0.0\n",
      "diagnoses.0.gross_tumor_weight \t 0.0\n",
      "diagnoses.0.icd_10_code \t 0.12057048820625343\n",
      "diagnoses.0.igcccg_stage \t 0.0\n",
      "diagnoses.0.inpc_grade \t 0.0\n",
      "diagnoses.0.inpc_histologic_group \t 0.0\n",
      "diagnoses.0.inrg_stage \t 0.0\n",
      "diagnoses.0.inss_stage \t 0.0\n",
      "diagnoses.0.international_prognostic_index \t 0.0\n",
      "diagnoses.0.irs_group \t 0.0\n",
      "diagnoses.0.irs_stage \t 0.0\n",
      "diagnoses.0.ishak_fibrosis_score \t 0.0\n",
      "diagnoses.0.iss_stage \t 0.0\n",
      "diagnoses.0.largest_extrapelvic_peritoneal_focus \t 0.0\n",
      "diagnoses.0.last_known_disease_status \t 0.975973669775096\n",
      "diagnoses.0.laterality \t 0.0\n",
      "diagnoses.0.lymph_node_involved_site \t 0.0\n",
      "diagnoses.0.lymph_nodes_positive \t 0.0\n",
      "diagnoses.0.lymph_nodes_tested \t 0.0\n",
      "diagnoses.0.lymphatic_invasion_present \t 0.0\n",
      "diagnoses.0.margin_distance \t 0.0\n",
      "diagnoses.0.margins_involved_site \t 0.0\n",
      "diagnoses.0.masaoka_stage \t 0.0\n",
      "diagnoses.0.medulloblastoma_molecular_classification \t 0.0\n",
      "diagnoses.0.metastasis_at_diagnosis \t 0.00010970927043335162\n",
      "diagnoses.0.metastasis_at_diagnosis_site \t 0.0\n",
      "diagnoses.0.method_of_diagnosis \t 0.0\n",
      "diagnoses.0.micropapillary_features \t 0.0\n",
      "diagnoses.0.mitosis_karyorrhexis_index \t 0.0\n",
      "diagnoses.0.mitotic_count \t 0.0\n",
      "diagnoses.0.morphology \t 0.9998902907295667\n",
      "diagnoses.0.non_nodal_regional_disease \t 0.0\n",
      "diagnoses.0.non_nodal_tumor_deposits \t 0.0\n",
      "diagnoses.0.ovarian_specimen_status \t 0.0\n",
      "diagnoses.0.ovarian_surface_involvement \t 0.0\n",
      "diagnoses.0.papillary_renal_cell_type \t 0.0\n",
      "diagnoses.0.percent_tumor_invasion \t 0.0\n",
      "diagnoses.0.perineural_invasion_present \t 0.0\n",
      "diagnoses.0.peripancreatic_lymph_nodes_positive \t 0.0\n",
      "diagnoses.0.peripancreatic_lymph_nodes_tested \t 0.0\n",
      "diagnoses.0.peritoneal_fluid_cytological_status \t 0.0\n",
      "diagnoses.0.pregnant_at_diagnosis \t 0.0\n",
      "diagnoses.0.primary_diagnosis \t 0.9998902907295667\n",
      "diagnoses.0.primary_gleason_grade \t 0.0\n",
      "diagnoses.0.prior_malignancy \t 0.13395501919912234\n",
      "diagnoses.0.prior_treatment \t 0.12057048820625343\n",
      "diagnoses.0.progression_or_recurrence \t 0.975973669775096\n",
      "diagnoses.0.residual_disease \t 0.0\n",
      "diagnoses.0.secondary_gleason_grade \t 0.0\n",
      "diagnoses.0.site_of_resection_or_biopsy \t 0.9998902907295667\n",
      "diagnoses.0.state \t 0.9998902907295667\n",
      "diagnoses.0.submitter_id \t 0.9998902907295667\n",
      "diagnoses.0.supratentorial_localization \t 0.0\n",
      "diagnoses.0.synchronous_malignancy \t 0.12046077893582008\n",
      "diagnoses.0.tissue_or_organ_of_origin \t 0.9998902907295667\n",
      "diagnoses.0.transglottic_extension \t 0.0\n",
      "diagnoses.0.treatments.0.created_datetime \t 0.06023038946791004\n",
      "diagnoses.0.treatments.0.days_to_treatment_end \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.0.state \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.submitter_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.0.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.0.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.updated_datetime \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.1.created_datetime \t 0.060449808008776744\n",
      "diagnoses.0.treatments.1.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.1.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.1.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.1.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.1.state \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.submitter_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_id \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.1.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.1.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.updated_datetime \t 0.12057048820625343\n",
      "diagnoses.0.treatments.10.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.10.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.10.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.10.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.10.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.10.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.10.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.10.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.10.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.2.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.2.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.2.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.2.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.2.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.2.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.2.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.2.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.3.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.4.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.4.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.4.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.4.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.4.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.4.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.4.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.4.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.5.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.5.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.5.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.5.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.5.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.5.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.therapeutic_agents \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.5.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.5.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.6.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.6.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.6.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.6.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.6.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.6.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.6.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.6.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.7.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.7.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.7.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.7.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.7.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.treatment_intent_type \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.7.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.7.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.8.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.8.days_to_treatment_start \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.initial_disease_status \t 0.0\n",
      "diagnoses.0.treatments.8.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.8.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.8.regimen_or_line_of_therapy \t 0.00010970927043335162\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diagnoses.0.treatments.8.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.8.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.8.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.chemo_concurrent_to_radiation \t 0.0\n",
      "diagnoses.0.treatments.9.created_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.days_to_treatment_end \t 0.0\n",
      "diagnoses.0.treatments.9.days_to_treatment_start \t 0.0\n",
      "diagnoses.0.treatments.9.initial_disease_status \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.number_of_cycles \t 0.0\n",
      "diagnoses.0.treatments.9.reason_treatment_ended \t 0.0\n",
      "diagnoses.0.treatments.9.regimen_or_line_of_therapy \t 0.0\n",
      "diagnoses.0.treatments.9.state \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.submitter_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.therapeutic_agents \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_anatomic_site \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_arm \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_dose \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_dose_units \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_effect \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_effect_indicator \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_frequency \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_id \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_intent_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_or_therapy \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.treatment_outcome \t 0.0\n",
      "diagnoses.0.treatments.9.treatment_type \t 0.00010970927043335162\n",
      "diagnoses.0.treatments.9.updated_datetime \t 0.00010970927043335162\n",
      "diagnoses.0.tumor_confined_to_organ_of_origin \t 0.0\n",
      "diagnoses.0.tumor_depth \t 0.0\n",
      "diagnoses.0.tumor_focality \t 0.0\n",
      "diagnoses.0.tumor_grade \t 0.9760833790455293\n",
      "diagnoses.0.tumor_largest_dimension_diameter \t 0.0\n",
      "diagnoses.0.tumor_regression_grade \t 0.0\n",
      "diagnoses.0.tumor_stage \t 0.975973669775096\n",
      "diagnoses.0.updated_datetime \t 0.9998902907295667\n",
      "diagnoses.0.vascular_invasion_present \t 0.0\n",
      "diagnoses.0.vascular_invasion_type \t 0.0\n",
      "diagnoses.0.weiss_assessment_score \t 0.0\n",
      "diagnoses.0.wilms_tumor_histologic_subtype \t 0.0\n",
      "diagnoses.0.year_of_diagnosis \t 0.12024136039495337\n",
      "exposures.0.age_at_onset \t 0.0\n",
      "exposures.0.alcohol_days_per_week \t 0.0\n",
      "exposures.0.alcohol_drinks_per_day \t 0.0\n",
      "exposures.0.alcohol_history \t 0.12046077893582008\n",
      "exposures.0.alcohol_intensity \t 0.0\n",
      "exposures.0.asbestos_exposure \t 0.0\n",
      "exposures.0.bmi \t 0.00010970927043335162\n",
      "exposures.0.cigarettes_per_day \t 0.0\n",
      "exposures.0.coal_dust_exposure \t 0.0\n",
      "exposures.0.created_datetime \t 0.00010970927043335162\n",
      "exposures.0.environmental_tobacco_smoke_exposure \t 0.0\n",
      "exposures.0.exposure_duration \t 0.0\n",
      "exposures.0.exposure_id \t 0.12057048820625343\n",
      "exposures.0.exposure_type \t 0.0\n",
      "exposures.0.height \t 0.00010970927043335162\n",
      "exposures.0.marijuana_use_per_week \t 0.0\n",
      "exposures.0.pack_years_smoked \t 0.0\n",
      "exposures.0.radon_exposure \t 0.0\n",
      "exposures.0.respirable_crystalline_silica_exposure \t 0.0\n",
      "exposures.0.secondhand_smoke_as_child \t 0.0\n",
      "exposures.0.smoking_frequency \t 0.0\n",
      "exposures.0.state \t 0.12057048820625343\n",
      "exposures.0.submitter_id \t 0.12057048820625343\n",
      "exposures.0.time_between_waking_and_first_smoke \t 0.0\n",
      "exposures.0.tobacco_smoking_onset_year \t 0.0\n",
      "exposures.0.tobacco_smoking_quit_year \t 0.0\n",
      "exposures.0.tobacco_smoking_status \t 0.00010970927043335162\n",
      "exposures.0.tobacco_use_per_day \t 0.0\n",
      "exposures.0.type_of_smoke_exposure \t 0.0\n",
      "exposures.0.type_of_tobacco_used \t 0.0\n",
      "exposures.0.updated_datetime \t 0.12057048820625343\n",
      "exposures.0.weight \t 0.00010970927043335162\n",
      "exposures.0.years_smoked \t 0.0\n",
      "family_histories.0.created_datetime \t 0.00010970927043335162\n",
      "family_histories.0.family_history_id \t 0.00010970927043335162\n",
      "family_histories.0.relationship_age_at_diagnosis \t 0.0\n",
      "family_histories.0.relationship_gender \t 0.0\n",
      "family_histories.0.relationship_primary_diagnosis \t 0.00010970927043335162\n",
      "family_histories.0.relationship_type \t 0.0\n",
      "family_histories.0.relative_with_cancer_history \t 0.00010970927043335162\n",
      "family_histories.0.relatives_with_cancer_history_count \t 0.0\n",
      "family_histories.0.state \t 0.00010970927043335162\n",
      "family_histories.0.submitter_id \t 0.00010970927043335162\n",
      "family_histories.0.updated_datetime \t 0.00010970927043335162\n",
      "id \t 1.0\n",
      "samples.0.biospecimen_anatomic_site \t 0.023806911684037303\n",
      "samples.0.biospecimen_laterality \t 0.0\n",
      "samples.0.catalog_reference \t 0.0\n",
      "samples.0.composition \t 0.007021393307734504\n",
      "samples.0.created_datetime \t 0.9190345584201866\n",
      "samples.0.current_weight \t 0.0\n",
      "samples.0.days_to_collection \t 0.08107515085024684\n",
      "samples.0.days_to_sample_procurement \t 0.03982446516730664\n",
      "samples.0.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.0.distance_normal_to_tumor \t 0.0\n",
      "samples.0.distributor_reference \t 0.0\n",
      "samples.0.freezing_method \t 0.0\n",
      "samples.0.growth_rate \t 0.0\n",
      "samples.0.initial_weight \t 0.04256719692814043\n",
      "samples.0.intermediate_dimension \t 0.0\n",
      "samples.0.is_ffpe \t 0.14448710916072408\n",
      "samples.0.longest_dimension \t 0.0\n",
      "samples.0.method_of_sample_procurement \t 0.0\n",
      "samples.0.oct_embedded \t 0.12068019747668678\n",
      "samples.0.passage_count \t 0.0\n",
      "samples.0.pathology_report_uuid \t 0.03817882611080636\n",
      "samples.0.preservation_method \t 0.03982446516730664\n",
      "samples.0.sample_id \t 1.0\n",
      "samples.0.sample_type \t 1.0\n",
      "samples.0.sample_type_id \t 0.1353812397147559\n",
      "samples.0.shortest_dimension \t 0.0\n",
      "samples.0.state \t 1.0\n",
      "samples.0.submitter_id \t 1.0\n",
      "samples.0.time_between_clamping_and_freezing \t 0.0\n",
      "samples.0.time_between_excision_and_freezing \t 0.0\n",
      "samples.0.tissue_collection_type \t 0.0\n",
      "samples.0.tissue_type \t 1.0\n",
      "samples.0.tumor_code \t 0.0\n",
      "samples.0.tumor_code_id \t 0.0\n",
      "samples.0.tumor_descriptor \t 0.28337904552934723\n",
      "samples.0.updated_datetime \t 1.0\n",
      "samples.1.biospecimen_anatomic_site \t 0.022270981897970377\n",
      "samples.1.biospecimen_laterality \t 0.0\n",
      "samples.1.catalog_reference \t 0.0\n",
      "samples.1.composition \t 0.007679648930334613\n",
      "samples.1.created_datetime \t 0.07383433900164564\n",
      "samples.1.current_weight \t 0.0\n",
      "samples.1.days_to_collection \t 0.08381788261108064\n",
      "samples.1.days_to_sample_procurement \t 0.03664289632473944\n",
      "samples.1.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.1.distance_normal_to_tumor \t 0.0\n",
      "samples.1.distributor_reference \t 0.0\n",
      "samples.1.freezing_method \t 0.0\n",
      "samples.1.growth_rate \t 0.0\n",
      "samples.1.initial_weight \t 0.05013713658804169\n",
      "samples.1.intermediate_dimension \t 0.0\n",
      "samples.1.is_ffpe \t 0.1428414701042238\n",
      "samples.1.longest_dimension \t 0.0\n",
      "samples.1.method_of_sample_procurement \t 0.0\n",
      "samples.1.oct_embedded \t 0.12057048820625343\n",
      "samples.1.passage_count \t 0.0\n",
      "samples.1.pathology_report_uuid \t 0.04509051014810751\n",
      "samples.1.preservation_method \t 0.03664289632473944\n",
      "samples.1.sample_id \t 0.15765222161272627\n",
      "samples.1.sample_type \t 0.15765222161272627\n",
      "samples.1.sample_type_id \t 0.1353812397147559\n",
      "samples.1.shortest_dimension \t 0.0\n",
      "samples.1.state \t 0.15765222161272627\n",
      "samples.1.submitter_id \t 0.15765222161272627\n",
      "samples.1.time_between_clamping_and_freezing \t 0.0\n",
      "samples.1.time_between_excision_and_freezing \t 0.0\n",
      "samples.1.tissue_collection_type \t 0.0\n",
      "samples.1.tissue_type \t 0.15765222161272627\n",
      "samples.1.tumor_code \t 0.0\n",
      "samples.1.tumor_code_id \t 0.0\n",
      "samples.1.tumor_descriptor \t 0.0\n",
      "samples.1.updated_datetime \t 0.15765222161272627\n",
      "samples.2.biospecimen_anatomic_site \t 0.01634668129456939\n",
      "samples.2.biospecimen_laterality \t 0.0\n",
      "samples.2.catalog_reference \t 0.0\n",
      "samples.2.composition \t 0.004388370817334065\n",
      "samples.2.created_datetime \t 0.05573230938014262\n",
      "samples.2.current_weight \t 0.0\n",
      "samples.2.days_to_collection \t 0.07931980252331322\n",
      "samples.2.days_to_sample_procurement \t 0.036752605595172794\n",
      "samples.2.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.2.distance_normal_to_tumor \t 0.0\n",
      "samples.2.distributor_reference \t 0.0\n",
      "samples.2.freezing_method \t 0.0\n",
      "samples.2.growth_rate \t 0.0\n",
      "samples.2.initial_weight \t 0.04201865057597367\n",
      "samples.2.intermediate_dimension \t 0.0\n",
      "samples.2.is_ffpe \t 0.13274821722435545\n",
      "samples.2.longest_dimension \t 0.0\n",
      "samples.2.method_of_sample_procurement \t 0.0\n",
      "samples.2.oct_embedded \t 0.11640153592978607\n",
      "samples.2.passage_count \t 0.0\n",
      "samples.2.pathology_report_uuid \t 0.03708173340647285\n",
      "samples.2.preservation_method \t 0.036752605595172794\n",
      "samples.2.sample_id \t 0.13527153044432255\n",
      "samples.2.sample_type \t 0.13527153044432255\n",
      "samples.2.sample_type_id \t 0.11892484914975315\n",
      "samples.2.shortest_dimension \t 0.0\n",
      "samples.2.state \t 0.13527153044432255\n",
      "samples.2.submitter_id \t 0.13527153044432255\n",
      "samples.2.time_between_clamping_and_freezing \t 0.0\n",
      "samples.2.time_between_excision_and_freezing \t 0.0\n",
      "samples.2.tissue_collection_type \t 0.0\n",
      "samples.2.tissue_type \t 0.13527153044432255\n",
      "samples.2.tumor_code \t 0.0\n",
      "samples.2.tumor_code_id \t 0.0\n",
      "samples.2.tumor_descriptor \t 0.0\n",
      "samples.2.updated_datetime \t 0.13527153044432255\n",
      "samples.3.biospecimen_anatomic_site \t 0.006253428414701042\n",
      "samples.3.biospecimen_laterality \t 0.0\n",
      "samples.3.catalog_reference \t 0.0\n",
      "samples.3.composition \t 0.0012068019747668679\n",
      "samples.3.created_datetime \t 0.009654415798134943\n",
      "samples.3.current_weight \t 0.0\n",
      "samples.3.days_to_collection \t 0.0072408118486012064\n",
      "samples.3.days_to_sample_procurement \t 0.0034009873834339004\n",
      "samples.3.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.3.distance_normal_to_tumor \t 0.0\n",
      "samples.3.distributor_reference \t 0.0\n",
      "samples.3.freezing_method \t 0.0\n",
      "samples.3.growth_rate \t 0.0\n",
      "samples.3.initial_weight \t 0.004388370817334065\n",
      "samples.3.intermediate_dimension \t 0.0\n",
      "samples.3.is_ffpe \t 0.01689522764673615\n",
      "samples.3.longest_dimension \t 0.0\n",
      "samples.3.method_of_sample_procurement \t 0.0\n",
      "samples.3.oct_embedded \t 0.010641799232035107\n",
      "samples.3.passage_count \t 0.0\n",
      "samples.3.pathology_report_uuid \t 0.002523313219967087\n",
      "samples.3.preservation_method \t 0.0034009873834339004\n",
      "samples.3.sample_id \t 0.01689522764673615\n",
      "samples.3.sample_type \t 0.01689522764673615\n",
      "samples.3.sample_type_id \t 0.010641799232035107\n",
      "samples.3.shortest_dimension \t 0.0\n",
      "samples.3.state \t 0.01689522764673615\n",
      "samples.3.submitter_id \t 0.01689522764673615\n",
      "samples.3.time_between_clamping_and_freezing \t 0.0\n",
      "samples.3.time_between_excision_and_freezing \t 0.0\n",
      "samples.3.tissue_collection_type \t 0.0\n",
      "samples.3.tissue_type \t 0.01689522764673615\n",
      "samples.3.tumor_code \t 0.0\n",
      "samples.3.tumor_code_id \t 0.0\n",
      "samples.3.tumor_descriptor \t 0.0\n",
      "samples.3.updated_datetime \t 0.01689522764673615\n",
      "samples.4.biospecimen_anatomic_site \t 0.004168952276467361\n",
      "samples.4.biospecimen_laterality \t 0.0\n",
      "samples.4.catalog_reference \t 0.0\n",
      "samples.4.composition \t 0.0007679648930334613\n",
      "samples.4.created_datetime \t 0.004168952276467361\n",
      "samples.4.current_weight \t 0.0\n",
      "samples.4.days_to_collection \t 0.0004388370817334065\n",
      "samples.4.days_to_sample_procurement \t 0.0\n",
      "samples.4.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.4.distance_normal_to_tumor \t 0.0\n",
      "samples.4.distributor_reference \t 0.0\n",
      "samples.4.freezing_method \t 0.0\n",
      "samples.4.growth_rate \t 0.0\n",
      "samples.4.initial_weight \t 0.00021941854086670324\n",
      "samples.4.intermediate_dimension \t 0.0\n",
      "samples.4.is_ffpe \t 0.004607789358200768\n",
      "samples.4.longest_dimension \t 0.0\n",
      "samples.4.method_of_sample_procurement \t 0.0\n",
      "samples.4.oct_embedded \t 0.0004388370817334065\n",
      "samples.4.passage_count \t 0.0\n",
      "samples.4.pathology_report_uuid \t 0.00021941854086670324\n",
      "samples.4.preservation_method \t 0.0\n",
      "samples.4.sample_id \t 0.004607789358200768\n",
      "samples.4.sample_type \t 0.004607789358200768\n",
      "samples.4.sample_type_id \t 0.0004388370817334065\n",
      "samples.4.shortest_dimension \t 0.0\n",
      "samples.4.state \t 0.004607789358200768\n",
      "samples.4.submitter_id \t 0.004607789358200768\n",
      "samples.4.time_between_clamping_and_freezing \t 0.0\n",
      "samples.4.time_between_excision_and_freezing \t 0.0\n",
      "samples.4.tissue_collection_type \t 0.0\n",
      "samples.4.tissue_type \t 0.004607789358200768\n",
      "samples.4.tumor_code \t 0.0\n",
      "samples.4.tumor_code_id \t 0.0\n",
      "samples.4.tumor_descriptor \t 0.0\n",
      "samples.4.updated_datetime \t 0.004607789358200768\n",
      "samples.5.biospecimen_anatomic_site \t 0.0015359297860669226\n",
      "samples.5.biospecimen_laterality \t 0.0\n",
      "samples.5.catalog_reference \t 0.0\n",
      "samples.5.composition \t 0.0006582556226001097\n",
      "samples.5.created_datetime \t 0.0015359297860669226\n",
      "samples.5.current_weight \t 0.0\n",
      "samples.5.days_to_collection \t 0.0\n",
      "samples.5.days_to_sample_procurement \t 0.0\n",
      "samples.5.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.5.distance_normal_to_tumor \t 0.0\n",
      "samples.5.distributor_reference \t 0.0\n",
      "samples.5.freezing_method \t 0.0\n",
      "samples.5.growth_rate \t 0.0\n",
      "samples.5.initial_weight \t 0.0\n",
      "samples.5.intermediate_dimension \t 0.0\n",
      "samples.5.is_ffpe \t 0.0015359297860669226\n",
      "samples.5.longest_dimension \t 0.0\n",
      "samples.5.method_of_sample_procurement \t 0.0\n",
      "samples.5.oct_embedded \t 0.0\n",
      "samples.5.passage_count \t 0.0\n",
      "samples.5.pathology_report_uuid \t 0.0\n",
      "samples.5.preservation_method \t 0.0\n",
      "samples.5.sample_id \t 0.0015359297860669226\n",
      "samples.5.sample_type \t 0.0015359297860669226\n",
      "samples.5.sample_type_id \t 0.0\n",
      "samples.5.shortest_dimension \t 0.0\n",
      "samples.5.state \t 0.0015359297860669226\n",
      "samples.5.submitter_id \t 0.0015359297860669226\n",
      "samples.5.time_between_clamping_and_freezing \t 0.0\n",
      "samples.5.time_between_excision_and_freezing \t 0.0\n",
      "samples.5.tissue_collection_type \t 0.0\n",
      "samples.5.tissue_type \t 0.0015359297860669226\n",
      "samples.5.tumor_code \t 0.0\n",
      "samples.5.tumor_code_id \t 0.0\n",
      "samples.5.tumor_descriptor \t 0.0\n",
      "samples.5.updated_datetime \t 0.0015359297860669226\n",
      "samples.6.biospecimen_anatomic_site \t 0.0009873834339001646\n",
      "samples.6.biospecimen_laterality \t 0.0\n",
      "samples.6.catalog_reference \t 0.0\n",
      "samples.6.composition \t 0.0\n",
      "samples.6.created_datetime \t 0.0009873834339001646\n",
      "samples.6.current_weight \t 0.0\n",
      "samples.6.days_to_collection \t 0.0\n",
      "samples.6.days_to_sample_procurement \t 0.0\n",
      "samples.6.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.6.distance_normal_to_tumor \t 0.0\n",
      "samples.6.distributor_reference \t 0.0\n",
      "samples.6.freezing_method \t 0.0\n",
      "samples.6.growth_rate \t 0.0\n",
      "samples.6.initial_weight \t 0.0\n",
      "samples.6.intermediate_dimension \t 0.0\n",
      "samples.6.is_ffpe \t 0.0009873834339001646\n",
      "samples.6.longest_dimension \t 0.0\n",
      "samples.6.method_of_sample_procurement \t 0.0\n",
      "samples.6.oct_embedded \t 0.0\n",
      "samples.6.passage_count \t 0.0\n",
      "samples.6.pathology_report_uuid \t 0.0\n",
      "samples.6.preservation_method \t 0.0\n",
      "samples.6.sample_id \t 0.0009873834339001646\n",
      "samples.6.sample_type \t 0.0009873834339001646\n",
      "samples.6.sample_type_id \t 0.0\n",
      "samples.6.shortest_dimension \t 0.0\n",
      "samples.6.state \t 0.0009873834339001646\n",
      "samples.6.submitter_id \t 0.0009873834339001646\n",
      "samples.6.time_between_clamping_and_freezing \t 0.0\n",
      "samples.6.time_between_excision_and_freezing \t 0.0\n",
      "samples.6.tissue_collection_type \t 0.0\n",
      "samples.6.tissue_type \t 0.0009873834339001646\n",
      "samples.6.tumor_code \t 0.0\n",
      "samples.6.tumor_code_id \t 0.0\n",
      "samples.6.tumor_descriptor \t 0.0\n",
      "samples.6.updated_datetime \t 0.0009873834339001646\n",
      "samples.7.biospecimen_anatomic_site \t 0.00032912781130005483\n",
      "samples.7.biospecimen_laterality \t 0.0\n",
      "samples.7.catalog_reference \t 0.0\n",
      "samples.7.composition \t 0.00010970927043335162\n",
      "samples.7.created_datetime \t 0.00032912781130005483\n",
      "samples.7.current_weight \t 0.0\n",
      "samples.7.days_to_collection \t 0.0\n",
      "samples.7.days_to_sample_procurement \t 0.0\n",
      "samples.7.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.7.distance_normal_to_tumor \t 0.0\n",
      "samples.7.distributor_reference \t 0.0\n",
      "samples.7.freezing_method \t 0.0\n",
      "samples.7.growth_rate \t 0.0\n",
      "samples.7.initial_weight \t 0.0\n",
      "samples.7.intermediate_dimension \t 0.0\n",
      "samples.7.is_ffpe \t 0.00032912781130005483\n",
      "samples.7.longest_dimension \t 0.0\n",
      "samples.7.method_of_sample_procurement \t 0.0\n",
      "samples.7.oct_embedded \t 0.0\n",
      "samples.7.passage_count \t 0.0\n",
      "samples.7.pathology_report_uuid \t 0.0\n",
      "samples.7.preservation_method \t 0.0\n",
      "samples.7.sample_id \t 0.00032912781130005483\n",
      "samples.7.sample_type \t 0.00032912781130005483\n",
      "samples.7.sample_type_id \t 0.0\n",
      "samples.7.shortest_dimension \t 0.0\n",
      "samples.7.state \t 0.00032912781130005483\n",
      "samples.7.submitter_id \t 0.00032912781130005483\n",
      "samples.7.time_between_clamping_and_freezing \t 0.0\n",
      "samples.7.time_between_excision_and_freezing \t 0.0\n",
      "samples.7.tissue_collection_type \t 0.0\n",
      "samples.7.tissue_type \t 0.00032912781130005483\n",
      "samples.7.tumor_code \t 0.0\n",
      "samples.7.tumor_code_id \t 0.0\n",
      "samples.7.tumor_descriptor \t 0.0\n",
      "samples.7.updated_datetime \t 0.00032912781130005483\n",
      "samples.8.biospecimen_anatomic_site \t 0.00010970927043335162\n",
      "samples.8.biospecimen_laterality \t 0.0\n",
      "samples.8.catalog_reference \t 0.0\n",
      "samples.8.composition \t 0.00010970927043335162\n",
      "samples.8.created_datetime \t 0.00010970927043335162\n",
      "samples.8.current_weight \t 0.0\n",
      "samples.8.days_to_collection \t 0.0\n",
      "samples.8.days_to_sample_procurement \t 0.0\n",
      "samples.8.diagnosis_pathologically_confirmed \t 0.0\n",
      "samples.8.distance_normal_to_tumor \t 0.0\n",
      "samples.8.distributor_reference \t 0.0\n",
      "samples.8.freezing_method \t 0.0\n",
      "samples.8.growth_rate \t 0.0\n",
      "samples.8.initial_weight \t 0.0\n",
      "samples.8.intermediate_dimension \t 0.0\n",
      "samples.8.is_ffpe \t 0.00010970927043335162\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "samples.8.longest_dimension \t 0.0\n",
      "samples.8.method_of_sample_procurement \t 0.0\n",
      "samples.8.oct_embedded \t 0.0\n",
      "samples.8.passage_count \t 0.0\n",
      "samples.8.pathology_report_uuid \t 0.0\n",
      "samples.8.preservation_method \t 0.0\n",
      "samples.8.sample_id \t 0.00010970927043335162\n",
      "samples.8.sample_type \t 0.00010970927043335162\n",
      "samples.8.sample_type_id \t 0.0\n",
      "samples.8.shortest_dimension \t 0.0\n",
      "samples.8.state \t 0.00010970927043335162\n",
      "samples.8.submitter_id \t 0.00010970927043335162\n",
      "samples.8.time_between_clamping_and_freezing \t 0.0\n",
      "samples.8.time_between_excision_and_freezing \t 0.0\n",
      "samples.8.tissue_collection_type \t 0.0\n",
      "samples.8.tissue_type \t 0.00010970927043335162\n",
      "samples.8.tumor_code \t 0.0\n",
      "samples.8.tumor_code_id \t 0.0\n",
      "samples.8.tumor_descriptor \t 0.0\n",
      "samples.8.updated_datetime \t 0.00010970927043335162\n",
      "submitter_id \t 1.0\n",
      "tissue_source_site.bcr_id \t 0.12057048820625343\n",
      "tissue_source_site.code \t 0.12057048820625343\n",
      "tissue_source_site.name \t 0.12057048820625343\n",
      "tissue_source_site.project \t 0.12057048820625343\n",
      "tissue_source_site.tissue_source_site_id \t 0.12057048820625343\n"
     ]
    }
   ],
   "source": [
    "# The value on the right is the percentage of non null value par column.\n",
    "# As a consequence, this dataframe presents no real interest.\n",
    "serie = df_case.notnull().mean()\n",
    "\n",
    "for name,val in zip(serie.index,serie): \n",
    "    print(name,'\\t',val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case_id \t 1.0\n",
      "demographic.age_at_index \t 0.6760285244103127\n",
      "demographic.days_to_birth \t 0.6744925946242457\n",
      "demographic.ethnicity \t 0.9998902907295667\n",
      "demographic.gender \t 0.9998902907295667\n",
      "demographic.race \t 0.9998902907295667\n",
      "demographic.vital_status \t 0.9998902907295667\n",
      "demographic.year_of_birth \t 0.12024136039495337\n",
      "diagnoses.0.age_at_diagnosis \t 0.4365331870543061\n",
      "diagnoses.0.ajcc_pathologic_m \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_n \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_pathologic_stage \t 0.13395501919912234\n",
      "diagnoses.0.ajcc_pathologic_t \t 0.12046077893582008\n",
      "diagnoses.0.ajcc_staging_system_edition \t 0.1049917718047175\n",
      "diagnoses.0.classification_of_tumor \t 0.4185408667032364\n",
      "diagnoses.0.days_to_diagnosis \t 0.12057048820625343\n",
      "diagnoses.0.days_to_last_follow_up \t 0.10905101481075151\n",
      "diagnoses.0.last_known_disease_status \t 0.975973669775096\n",
      "diagnoses.0.morphology \t 0.9998902907295667\n",
      "diagnoses.0.primary_diagnosis \t 0.9998902907295667\n",
      "diagnoses.0.prior_malignancy \t 0.13395501919912234\n",
      "diagnoses.0.prior_treatment \t 0.12057048820625343\n",
      "diagnoses.0.progression_or_recurrence \t 0.975973669775096\n",
      "diagnoses.0.site_of_resection_or_biopsy \t 0.9998902907295667\n",
      "diagnoses.0.synchronous_malignancy \t 0.12046077893582008\n",
      "diagnoses.0.tissue_or_organ_of_origin \t 0.9998902907295667\n",
      "diagnoses.0.treatments.0.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.0.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_or_therapy \t 0.12057048820625343\n",
      "diagnoses.0.treatments.1.treatment_type \t 0.12057048820625343\n",
      "diagnoses.0.tumor_grade \t 0.9760833790455293\n",
      "diagnoses.0.tumor_stage \t 0.975973669775096\n",
      "diagnoses.0.year_of_diagnosis \t 0.12024136039495337\n",
      "exposures.0.alcohol_history \t 0.12046077893582008\n",
      "id \t 1.0\n",
      "samples.0.is_ffpe \t 0.14448710916072408\n",
      "samples.0.oct_embedded \t 0.12068019747668678\n",
      "samples.0.sample_type \t 1.0\n",
      "samples.0.tissue_type \t 1.0\n",
      "samples.0.tumor_descriptor \t 0.28337904552934723\n",
      "samples.1.is_ffpe \t 0.1428414701042238\n",
      "samples.1.oct_embedded \t 0.12057048820625343\n",
      "samples.1.sample_type \t 0.15765222161272627\n",
      "samples.1.tissue_type \t 0.15765222161272627\n",
      "samples.2.is_ffpe \t 0.13274821722435545\n",
      "samples.2.oct_embedded \t 0.11640153592978607\n",
      "samples.2.sample_type \t 0.13527153044432255\n",
      "samples.2.tissue_type \t 0.13527153044432255\n"
     ]
    }
   ],
   "source": [
    "new_df_case = df_case.drop(df_case.columns[df_case.notnull().mean()<0.1], axis=1)\n",
    "\n",
    "columns_to_drop = ['tissue_source_site.code', 'tissue_source_site.project','tissue_source_site.name', 'diagnoses.0.treatments.1.state', 'diagnoses.0.state', 'diagnoses.0.treatments.0.state', 'exposures.0.state', 'samples.0.state', 'samples.1.state', 'samples.2.state', 'demographic.demographic_id', 'demographic.state', 'demographic.submitter_id', 'demographic.updated_datetime', 'diagnoses.0.diagnosis_id', 'diagnoses.0.icd_10_code', 'diagnoses.0.submitter_id', 'diagnoses.0.treatments.1.updated_datetime', 'diagnoses.0.treatments.1.submitter_id', 'diagnoses.0.treatments.0.updated_datetime', 'diagnoses.0.updated_datetime', 'exposures.0.exposure_id', 'exposures.0.submitter_id', 'exposures.0.updated_datetime', 'samples.0.created_datetime', 'samples.0.sample_id', 'samples.0.sample_type_id', 'samples.0.submitter_id', 'samples.0.updated_datetime', 'samples.1.sample_id', 'samples.1.sample_type_id', 'samples.1.submitter_id', 'samples.2.sample_id', 'samples.2.sample_type_id', 'samples.2.submitter_id', 'tissue_source_site.bcr_id', 'submitter_id', 'diagnoses.0.treatments.0.submitter_id', 'diagnoses.0.treatments.1.treatment_id', 'samples.2.updated_datetime', 'samples.1.updated_datetime', 'diagnoses.0.treatments.0.treatment_id', 'demographic.created_datetime', 'diagnoses.0.created_datetime', 'tissue_source_site.tissue_source_site_id']\n",
    "new_df_case = new_df_case.drop(columns_to_drop, axis=1)\n",
    "#diagnoses.0.created_datetime\n",
    "# , 'diagnoses.0.treatments.0.submitter_id ',\n",
    "serie = new_df_case.notnull().mean()\n",
    "\n",
    "for name,val in zip(serie.index,serie): \n",
    "    print(name,'\\t',val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df = df_case[df_case[\"samples.1.sample_type\"].notnull()]\n",
    "#df = df.filter(regex=(\"samples.*.sample_type\"))\n",
    "#df = df_case.filter(regex=(\"samples.*.pathology_report_uuid\"))\n",
    "#â€¡df = df[df[\"samples.0.pathology_report_uuid\"].notnull()]\n",
    "#df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: cnv\n",
    "Information on the nature of the gene mutation. Most of the value are strings.\n",
    "\n",
    "12 columns and 1000 rows.\n",
    "\n",
    "Possible key: 'case.case_id' \n",
    "\n",
    "Possible link with other dataframe: 'cnv.consequence.0.gene.symbol' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "case.case_id                        1.0\n",
       "cnv.chromosome                      1.0\n",
       "cnv.cnv_change                      1.0\n",
       "cnv.cnv_id                          1.0\n",
       "cnv.consequence.0.consequence_id    1.0\n",
       "cnv.consequence.0.gene.symbol       1.0\n",
       "cnv.end_position                    1.0\n",
       "cnv.gene_level_cn                   1.0\n",
       "cnv.ncbi_build                      1.0\n",
       "cnv.start_position                  1.0\n",
       "cnv.variant_status                  1.0\n",
       "id                                  1.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cnv.notnull().mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: gene\n",
    "\n",
    "Information on top 50 genes to studied. Most of the value are strings.\n",
    "\n",
    "2 columns and 50 rows.\n",
    "\n",
    "Possible key: 'symbol'\n",
    "\n",
    "What is 'score' ?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_score</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>451.0</td>\n",
       "      <td>PIK3CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>404.0</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>322.0</td>\n",
       "      <td>MUC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>320.0</td>\n",
       "      <td>SHANK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>318.0</td>\n",
       "      <td>ANO1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>316.0</td>\n",
       "      <td>ORAOV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>314.0</td>\n",
       "      <td>CCND1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>314.0</td>\n",
       "      <td>AP001888.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>310.0</td>\n",
       "      <td>FGF19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>310.0</td>\n",
       "      <td>PPFIA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   _score      symbol\n",
       "0   451.0      PIK3CA\n",
       "1   404.0        TP53\n",
       "2   322.0        MUC4\n",
       "3   320.0      SHANK2\n",
       "4   318.0        ANO1\n",
       "5   316.0      ORAOV1\n",
       "6   314.0       CCND1\n",
       "7   314.0  AP001888.1\n",
       "8   310.0       FGF19\n",
       "9   310.0      PPFIA1"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "_score    1.0\n",
       "symbol    1.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene.dtypes\n",
    "df_gene.notnull().mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Frame: ssm\n",
    "\n",
    "Information on \n",
    "\n",
    "74 columns and 1000 rows.\n",
    "\n",
    "Possible key: 'case.case_id', 'id', \n",
    "\n",
    "Most of the values are string."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case.case_id\n",
      "id\n",
      "ssm.chromosome\n",
      "ssm.consequence.0.consequence_id\n",
      "ssm.consequence.0.transcript.gene.symbol\n",
      "ssm.consequence.1.consequence_id\n",
      "ssm.consequence.1.transcript.gene.symbol\n",
      "ssm.consequence.10.consequence_id\n",
      "ssm.consequence.10.transcript.gene.symbol\n",
      "ssm.consequence.11.consequence_id\n",
      "ssm.consequence.11.transcript.gene.symbol\n",
      "ssm.consequence.12.consequence_id\n",
      "ssm.consequence.12.transcript.gene.symbol\n",
      "ssm.consequence.13.consequence_id\n",
      "ssm.consequence.13.transcript.gene.symbol\n",
      "ssm.consequence.14.consequence_id\n",
      "ssm.consequence.14.transcript.gene.symbol\n",
      "ssm.consequence.15.consequence_id\n",
      "ssm.consequence.15.transcript.gene.symbol\n",
      "ssm.consequence.16.consequence_id\n",
      "ssm.consequence.16.transcript.gene.symbol\n",
      "ssm.consequence.17.consequence_id\n",
      "ssm.consequence.17.transcript.gene.symbol\n",
      "ssm.consequence.18.consequence_id\n",
      "ssm.consequence.18.transcript.gene.symbol\n",
      "ssm.consequence.19.consequence_id\n",
      "ssm.consequence.19.transcript.gene.symbol\n",
      "ssm.consequence.2.consequence_id\n",
      "ssm.consequence.2.transcript.gene.symbol\n",
      "ssm.consequence.20.consequence_id\n",
      "ssm.consequence.20.transcript.gene.symbol\n",
      "ssm.consequence.21.consequence_id\n",
      "ssm.consequence.21.transcript.gene.symbol\n",
      "ssm.consequence.22.consequence_id\n",
      "ssm.consequence.22.transcript.gene.symbol\n",
      "ssm.consequence.23.consequence_id\n",
      "ssm.consequence.23.transcript.gene.symbol\n",
      "ssm.consequence.24.consequence_id\n",
      "ssm.consequence.24.transcript.gene.symbol\n",
      "ssm.consequence.25.consequence_id\n",
      "ssm.consequence.25.transcript.gene.symbol\n",
      "ssm.consequence.26.consequence_id\n",
      "ssm.consequence.26.transcript.gene.symbol\n",
      "ssm.consequence.3.consequence_id\n",
      "ssm.consequence.3.transcript.gene.symbol\n",
      "ssm.consequence.4.consequence_id\n",
      "ssm.consequence.4.transcript.gene.symbol\n",
      "ssm.consequence.5.consequence_id\n",
      "ssm.consequence.5.transcript.gene.symbol\n",
      "ssm.consequence.6.consequence_id\n",
      "ssm.consequence.6.transcript.gene.symbol\n",
      "ssm.consequence.7.consequence_id\n",
      "ssm.consequence.7.transcript.gene.symbol\n",
      "ssm.consequence.8.consequence_id\n",
      "ssm.consequence.8.transcript.gene.symbol\n",
      "ssm.consequence.9.consequence_id\n",
      "ssm.consequence.9.transcript.gene.symbol\n",
      "ssm.cosmic_id.0\n",
      "ssm.cosmic_id.1\n",
      "ssm.cosmic_id.2\n",
      "ssm.cosmic_id.3\n",
      "ssm.cosmic_id.4\n",
      "ssm.cosmic_id.5\n",
      "ssm.cosmic_id.6\n",
      "ssm.cosmic_id.7\n",
      "ssm.end_position\n",
      "ssm.genomic_dna_change\n",
      "ssm.mutation_subtype\n",
      "ssm.mutation_type\n",
      "ssm.ncbi_build\n",
      "ssm.reference_allele\n",
      "ssm.ssm_id\n",
      "ssm.start_position\n",
      "ssm.tumor_allele\n"
     ]
    }
   ],
   "source": [
    "df_ssm.dtypes\n",
    "for col in df_ssm.columns: \n",
    "    print(col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case.case_id\n",
      "ssm.chromosome\n",
      "ssm.consequence.0.transcript.gene.symbol\n",
      "ssm.cosmic_id.0\n",
      "ssm.end_position\n",
      "ssm.genomic_dna_change\n",
      "ssm.mutation_subtype\n",
      "ssm.mutation_type\n",
      "ssm.ncbi_build\n",
      "ssm.reference_allele\n",
      "ssm.ssm_id\n",
      "ssm.start_position\n",
      "ssm.tumor_allele\n"
     ]
    }
   ],
   "source": [
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('ce_id')], axis=1, inplace=True)\n",
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('[0-2][1-9]')], axis=1, inplace=True)\n",
    "df_ssm.drop(df_ssm.columns[df_ssm.columns.str.contains('[1-9]')], axis=1, inplace=True)\n",
    "columns_to_drop = [\"id\"]\n",
    "df_ssm.drop(\"id\", axis=1,inplace=True)\n",
    "\n",
    "\n",
    "\n",
    "df_ssm.dtypes\n",
    "for col in df_ssm.columns: \n",
    "    print(col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2347, 13)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ssm.head()\n",
    "df_ssm.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "join() got an unexpected keyword argument 'join'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-20-8ab1e2915d51>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_df_case\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'case_id'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_cnv\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'case.case_id'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlsuffix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'_caller'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrsuffix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'_other'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"outer\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'id_caller'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_ssm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'case.case_id'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlsuffix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrsuffix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'_other'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mcolumns_to_drop\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m\"ssm.ssm_id\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"id_other\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"cnv.cnv_id\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"ssm.cosmic_id.0\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"ssm.cosmic_id.0\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"cnv.chromosome\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"ssm.end_position\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"cnv.gene_level_cn\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"ssm.ncbi_build\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"cnv.ncbi_build\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"cnv.start_position\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"cnv.variant_status\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: join() got an unexpected keyword argument 'join'"
     ]
    }
   ],
   "source": [
    "df = new_df_case.set_index('case_id').join(df_cnv.set_index('case.case_id'), lsuffix='_caller', rsuffix='_other', )\n",
    "print(df.shape)\n",
    "df = df.set_index('id_caller').join(df_ssm.set_index('case.case_id'), lsuffix='', rsuffix='_other')\n",
    "\n",
    "columns_to_drop = [\"ssm.ssm_id\", \"id_other\", \"cnv.cnv_id\", \"ssm.cosmic_id.0\", \"ssm.cosmic_id.0\",\"cnv.chromosome\",\"ssm.end_position\", \"cnv.gene_level_cn\", \"ssm.ncbi_build\", \"cnv.ncbi_build\", \"cnv.start_position\",\"cnv.variant_status\"]\n",
    "df = df.drop(columns_to_drop, axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case.case_id</th>\n",
       "      <th>cnv.chromosome</th>\n",
       "      <th>cnv.cnv_change</th>\n",
       "      <th>cnv.cnv_id</th>\n",
       "      <th>cnv.consequence.0.consequence_id</th>\n",
       "      <th>cnv.consequence.0.gene.symbol</th>\n",
       "      <th>cnv.end_position</th>\n",
       "      <th>cnv.gene_level_cn</th>\n",
       "      <th>cnv.ncbi_build</th>\n",
       "      <th>cnv.start_position</th>\n",
       "      <th>cnv.variant_status</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c5891c7a-fd24-4891-8f51-373967e4a6dd</td>\n",
       "      <td>17</td>\n",
       "      <td>Loss</td>\n",
       "      <td>0e18bd24-e7dc-5e6e-a3ba-a23fd41ac529</td>\n",
       "      <td>99c68f83-9514-5628-9623-0ba6095dd90f</td>\n",
       "      <td>BCAS3</td>\n",
       "      <td>61392838</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>60677453</td>\n",
       "      <td>Tumor only</td>\n",
       "      <td>034993b7-98a2-57df-b741-761080debf14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>d5f2b85a-94a9-4168-bbe8-149ec71342b0</td>\n",
       "      <td>17</td>\n",
       "      <td>Gain</td>\n",
       "      <td>73eb64c1-82ef-5f05-9946-a35b8a041148</td>\n",
       "      <td>a7e6f054-f5a1-5e7f-a0c0-8daa7373a434</td>\n",
       "      <td>ABCA8</td>\n",
       "      <td>68955392</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>68867292</td>\n",
       "      <td>Tumor only</td>\n",
       "      <td>309d1772-5c76-52d3-b384-32d00130c801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>eafa45a4-2fd1-4fa3-860c-9d52ed382b7d</td>\n",
       "      <td>11</td>\n",
       "      <td>Gain</td>\n",
       "      <td>10bfced2-6df4-5613-96b3-a5d70afde19c</td>\n",
       "      <td>924c27ca-11cf-5b13-95b9-cf518abb142d</td>\n",
       "      <td>ORAOV1</td>\n",
       "      <td>69675416</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>69653076</td>\n",
       "      <td>Tumor only</td>\n",
       "      <td>51a42266-c861-523b-92b6-313f5ad92b72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>dd0f8e5a-383d-4f03-bb71-0e115d4e442c</td>\n",
       "      <td>11</td>\n",
       "      <td>Gain</td>\n",
       "      <td>de0d6322-4040-570e-b9c3-94b6d565750f</td>\n",
       "      <td>00461981-a326-573d-a9a1-ad82872a1aea</td>\n",
       "      <td>SHANK2</td>\n",
       "      <td>71252577</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>70467856</td>\n",
       "      <td>Tumor only</td>\n",
       "      <td>61d42790-c539-5f6d-867b-c24bd9bb6cd6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>233b02f3-c4f0-4a67-9db5-e68d5cdaccb6</td>\n",
       "      <td>8</td>\n",
       "      <td>Gain</td>\n",
       "      <td>f0d7c53c-76e0-590c-b0ac-1aa316fe5c50</td>\n",
       "      <td>c6208422-2c76-553a-b0eb-4472e816381f</td>\n",
       "      <td>MYC</td>\n",
       "      <td>127741434</td>\n",
       "      <td>True</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>127735434</td>\n",
       "      <td>Tumor only</td>\n",
       "      <td>69cd805c-29ed-56dd-aa11-92f11956b814</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           case.case_id  cnv.chromosome cnv.cnv_change  \\\n",
       "0  c5891c7a-fd24-4891-8f51-373967e4a6dd              17           Loss   \n",
       "1  d5f2b85a-94a9-4168-bbe8-149ec71342b0              17           Gain   \n",
       "2  eafa45a4-2fd1-4fa3-860c-9d52ed382b7d              11           Gain   \n",
       "3  dd0f8e5a-383d-4f03-bb71-0e115d4e442c              11           Gain   \n",
       "4  233b02f3-c4f0-4a67-9db5-e68d5cdaccb6               8           Gain   \n",
       "\n",
       "                             cnv.cnv_id      cnv.consequence.0.consequence_id  \\\n",
       "0  0e18bd24-e7dc-5e6e-a3ba-a23fd41ac529  99c68f83-9514-5628-9623-0ba6095dd90f   \n",
       "1  73eb64c1-82ef-5f05-9946-a35b8a041148  a7e6f054-f5a1-5e7f-a0c0-8daa7373a434   \n",
       "2  10bfced2-6df4-5613-96b3-a5d70afde19c  924c27ca-11cf-5b13-95b9-cf518abb142d   \n",
       "3  de0d6322-4040-570e-b9c3-94b6d565750f  00461981-a326-573d-a9a1-ad82872a1aea   \n",
       "4  f0d7c53c-76e0-590c-b0ac-1aa316fe5c50  c6208422-2c76-553a-b0eb-4472e816381f   \n",
       "\n",
       "  cnv.consequence.0.gene.symbol  cnv.end_position  cnv.gene_level_cn  \\\n",
       "0                         BCAS3          61392838               True   \n",
       "1                         ABCA8          68955392               True   \n",
       "2                        ORAOV1          69675416               True   \n",
       "3                        SHANK2          71252577               True   \n",
       "4                           MYC         127741434               True   \n",
       "\n",
       "  cnv.ncbi_build  cnv.start_position cnv.variant_status  \\\n",
       "0         GRCh38            60677453         Tumor only   \n",
       "1         GRCh38            68867292         Tumor only   \n",
       "2         GRCh38            69653076         Tumor only   \n",
       "3         GRCh38            70467856         Tumor only   \n",
       "4         GRCh38           127735434         Tumor only   \n",
       "\n",
       "                                     id  \n",
       "0  034993b7-98a2-57df-b741-761080debf14  \n",
       "1  309d1772-5c76-52d3-b384-32d00130c801  \n",
       "2  51a42266-c861-523b-92b6-313f5ad92b72  \n",
       "3  61d42790-c539-5f6d-867b-c24bd9bb6cd6  \n",
       "4  69cd805c-29ed-56dd-aa11-92f11956b814  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cnv = df_cnv.rename({'case.case_id':'case_id'})\n",
    "new_df_case.head(5)\n",
    "df_cnv.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(9115, 48)\n",
      "(798,)\n",
      "(2347, 61)\n"
     ]
    }
   ],
   "source": [
    "print(new_df_case.shape)\n",
    "#print(df_cnv.shape)\n",
    "\n",
    "#pd.merge(left, right, on=[\"key1\", \"key2\"])\n",
    "print(df_cnv['case.case_id'].unique().shape)\n",
    "\n",
    "df = pd.merge(new_df_case,df_cnv, how='right', left_on='case_id', right_on='case.case_id')\n",
    "#print(df['case_id'].unique().shape)\n",
    "df = pd.merge(new_df_case,df_ssm, how='right', left_on='case_id', right_on='case.case_id')\n",
    "\n",
    "\n",
    "#new_df_case.set_index('case_id').join(df_cnv.set_index('case.case_id'), lsuffix='_caller', rsuffix='_other', )\n",
    "\n",
    "print(df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "case_id\n",
      "demographic.age_at_index\n",
      "demographic.days_to_birth\n",
      "demographic.ethnicity\n",
      "demographic.gender\n",
      "demographic.race\n",
      "demographic.vital_status\n",
      "demographic.year_of_birth\n",
      "diagnoses.0.age_at_diagnosis\n",
      "diagnoses.0.ajcc_pathologic_m\n",
      "diagnoses.0.ajcc_pathologic_n\n",
      "diagnoses.0.ajcc_pathologic_stage\n",
      "diagnoses.0.ajcc_pathologic_t\n",
      "diagnoses.0.ajcc_staging_system_edition\n",
      "diagnoses.0.classification_of_tumor\n",
      "diagnoses.0.days_to_diagnosis\n",
      "diagnoses.0.days_to_last_follow_up\n",
      "diagnoses.0.last_known_disease_status\n",
      "diagnoses.0.morphology\n",
      "diagnoses.0.primary_diagnosis\n",
      "diagnoses.0.prior_malignancy\n",
      "diagnoses.0.prior_treatment\n",
      "diagnoses.0.progression_or_recurrence\n",
      "diagnoses.0.site_of_resection_or_biopsy\n",
      "diagnoses.0.synchronous_malignancy\n",
      "diagnoses.0.tissue_or_organ_of_origin\n",
      "diagnoses.0.treatments.0.treatment_or_therapy\n",
      "diagnoses.0.treatments.0.treatment_type\n",
      "diagnoses.0.treatments.1.treatment_or_therapy\n",
      "diagnoses.0.treatments.1.treatment_type\n",
      "diagnoses.0.tumor_grade\n",
      "diagnoses.0.tumor_stage\n",
      "diagnoses.0.year_of_diagnosis\n",
      "exposures.0.alcohol_history\n",
      "id\n",
      "samples.0.is_ffpe\n",
      "samples.0.oct_embedded\n",
      "samples.0.sample_type\n",
      "samples.0.tissue_type\n",
      "samples.0.tumor_descriptor\n",
      "samples.1.is_ffpe\n",
      "samples.1.oct_embedded\n",
      "samples.1.sample_type\n",
      "samples.1.tissue_type\n",
      "samples.2.is_ffpe\n",
      "samples.2.oct_embedded\n",
      "samples.2.sample_type\n",
      "samples.2.tissue_type\n",
      "case.case_id\n",
      "ssm.chromosome\n",
      "ssm.consequence.0.transcript.gene.symbol\n",
      "ssm.cosmic_id.0\n",
      "ssm.end_position\n",
      "ssm.genomic_dna_change\n",
      "ssm.mutation_subtype\n",
      "ssm.mutation_type\n",
      "ssm.ncbi_build\n",
      "ssm.reference_allele\n",
      "ssm.ssm_id\n",
      "ssm.start_position\n",
      "ssm.tumor_allele\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "df.dtypes\n",
    "for col in df.columns: \n",
    "    print(col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_col_unique = [\"diagnoses.0.ajcc_pathologic_m\",'diagnoses.0.ajcc_pathologic_n',\"diagnoses.0.ajcc_pathologic_stage\",\"diagnoses.0.ajcc_pathologic_t\",\"diagnoses.0.tumor_grade\",\"diagnoses.0.tumor_stage\",\"samples.0.is_ffpe\",\"samples.1.oct_embedded\",\"cnv.chromosome\",\"cnv.cnv_change\",\"ssm.consequence.0.transcript.gene.symbol\",\"ssm.end_position\",\"ssm.ncbi_build\",\"cnv.gene_level_cn\",\"cnv.start_position\",\"cnv.variant_status\"]\n",
    "\n",
    "for column in list_col_unique:\n",
    "    print( '--------------------------------------------')\n",
    "    print(column)\n",
    "    print( '--------------------------------------------')\n",
    "    print(df[column].unique())\n",
    "    print('\\n')\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    1\n",
       "1    1\n",
       "2    1\n",
       "3    1\n",
       "4    1\n",
       "5    1\n",
       "6    2\n",
       "7    1\n",
       "8    3\n",
       "9    2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"count_na\"] = df.isnull().sum(axis=1).values\n",
    "df_new = pd.DataFrame({'case_id':df[\"count_na\"].index, 'count':df[\"count_na\"].values})\n",
    "\n",
    "df_new[\"count\"].head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.dropna(axis=0, thresh=20, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(809,)"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['case_id'].unique().shape\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
